<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Introduction to proteomics data analysis: median summarization</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.1 MS Basics</a>
    </li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.2 Identification</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1 Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2 Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1 Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Introduction to proteomics data analysis:
median summarization</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a
href="https://statomics.github.io/PDA22GTPB/">Proteomics Data Analysis
(PDA)</a></p>
<div id="background" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>This case-study is a subset of the data of the 6th study of the
Clinical Proteomic Technology Assessment for Cancer (CPTAC). In this
experiment, the authors spiked the Sigma Universal Protein Standard
mixture 1 (UPS1) containing 48 different human proteins in a protein
background of 60 ng/<span class="math inline">\(\mu\)</span>L
Saccharomyces cerevisiae strain BY4741. Two different spike-in
concentrations were used: 6A (0.25 fmol UPS1 proteins/<span
class="math inline">\(\mu\)</span>L) and 6B (0.74 fmol UPS1
proteins/<span class="math inline">\(\mu\)</span>L) [5]. We limited
ourselves to the data of LTQ-Orbitrap W at site 56. The data were
searched with MaxQuant version 1.5.2.8, and detailed search settings
were described in Goeminne et al. (2016) [1]. Three replicates are
available for each concentration.</p>
<ul>
<li>NOTE THAT MEDIAN SUMMARISATION IS SUBOPTIMAL!</li>
<li>THIS IS FOR DIDACTICAL PURPOSES ONLY.</li>
</ul>
</div>
<div id="data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data</h1>
<p>We first import the data from peptideRaws.txt file. This is the file
containing your peptideRaw-level intensities. For a MaxQuant search [6],
this peptideRaws.txt file can be found by default in the
“path_to_raw_files/combined/txt/” folder from the MaxQuant output, with
“path_to_raw_files” the folder where the raw files were saved. In this
vignette, we use a MaxQuant peptideRaws file which is a subset of the
cptac study. This data is available in the <code>msdata</code> package.
To import the data we use the <code>QFeatures</code> package.</p>
<p>We generate the object peptideRawFile with the path to the
peptideRaws.txt file. Using the <code>grepEcols</code> function, we find
the columns that contain the expression data of the peptideRaws in the
peptideRaws.txt file.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>peptidesFile <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/cptacAvsB_lab3/peptides.txt&quot;</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>ecols <span class="ot">&lt;-</span> <span class="fu">grep</span>(</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>  <span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, </span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>  <span class="fu">names</span>(<span class="fu">read.delim</span>(peptidesFile))</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>  )</span></code></pre></div>
<p>Next, we read the data and store it in QFeatures object</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="at">assayData =</span> <span class="fu">read.delim</span>(peptidesFile),</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="at">quantCols =</span> ecols,</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span></code></pre></div>
<pre><code>## Checking arguments.</code></pre>
<pre><code>## Loading data as a &#39;SummarizedExperiment&#39; object.</code></pre>
<pre><code>## Formatting sample annotations (colData).</code></pre>
<pre><code>## Formatting data as a &#39;QFeatures&#39; object.</code></pre>
<pre><code>## Setting assay rownames.</code></pre>
<p>The QFeatures object pe currently contains a single assay, named
peptideRaw.</p>
<p>We extract the column names from the peptideRaw assay and see that
this contains information about the spike-in condition.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] &quot;Intensity.6A_7&quot; &quot;Intensity.6A_8&quot; &quot;Intensity.6A_9&quot; &quot;Intensity.6B_7&quot;
## [5] &quot;Intensity.6B_8&quot; &quot;Intensity.6B_9&quot;</code></pre>
<p>We rename the colnames by dropping the “Intensity.” from the
name.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>(newNames <span class="ot">&lt;-</span>  <span class="fu">sub</span>(</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, </span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  <span class="at">replacement =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]))</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a> )</span></code></pre></div>
<pre><code>## [1] &quot;6A_7&quot; &quot;6A_8&quot; &quot;6A_9&quot; &quot;6B_7&quot; &quot;6B_8&quot; &quot;6B_9&quot;</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">renameColname</span>(pe, </span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>                    <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>,</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>                    newNames)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">renamePrimary</span>(pe, newNames)</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] &quot;6A_7&quot; &quot;6A_8&quot; &quot;6A_9&quot; &quot;6B_7&quot; &quot;6B_8&quot; &quot;6B_9&quot;</code></pre>
<p>In the following code chunk, we add the spikein condition that we can
read in the raw file name to the colData.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>condition <span class="ot">&lt;-</span>  </span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="fu">substr</span>(<span class="at">start =</span> <span class="dv">2</span>, <span class="at">stop =</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  as.factor</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>condition</span></code></pre></div>
<pre><code>## [1] A A A B B B
## Levels: A B</code></pre>
<p>We calculate how many non zero intensities we have per peptide and
this will be useful for filtering.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZero <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code></pre></div>
<p>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<div id="data-exploration" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Data exploration</h2>
<p>45% of all peptide intensities are missing and for some peptides we
do not even measure a signal in any sample.</p>
</div>
</div>
<div id="preprocessing" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Preprocessing</h1>
<p>This section preforms preprocessing for the peptide data. This
include</p>
<ul>
<li>log transformation,</li>
<li>filtering and</li>
<li>summarisation of the data.</li>
</ul>
<div id="log-transform-the-data" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Log transform the
data</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
</div>
<div id="filtering" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Filtering</h2>
<ol style="list-style-type: decimal">
<li>Handling overlapping protein groups</li>
</ol>
<p>In our approach a peptide can map to multiple proteins, as long as
there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> Proteins <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins))</span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants and peptides that map to decoy
sequences.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in one sample</li>
</ol>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> nNonZero <span class="sc">&gt;=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## &#39;nNonZero&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="fu">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 7011</code></pre>
<p>We keep 7011 peptides upon filtering.</p>
</div>
<div id="normalize-the-data-using-median-centering"
class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Normalize the data
using median centering</h2>
<p>We normalize the data by substracting the sample median from every
intensity for peptide <span class="math inline">\(p\)</span> in a sample
<span class="math inline">\(i\)</span>:</p>
<p><span class="math display">\[y_{ip}^\text{norm} = y_{ip} -
\hat\mu_i\]</span></p>
<p>with <span class="math inline">\(\hat\mu_i\)</span> the median
intensity over all observed peptides in sample <span
class="math inline">\(i\)</span>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</div>
<div id="explore-normalized-data" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Explore normalized
data</h2>
<p>Upon the normalisation the density curves are nicely registered</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideNorm&quot;</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>  assay <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>  <span class="fu">gather</span>(sample, intensity) <span class="sc">%&gt;%</span> </span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">condition =</span> <span class="fu">colData</span>(pe)[sample,<span class="st">&quot;condition&quot;</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> intensity,<span class="at">group =</span> sample,<span class="at">color =</span> condition)) <span class="sc">+</span> </span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>    <span class="fu">geom_density</span>()</span></code></pre></div>
<pre><code>## Warning: Removed 8167 rows containing non-finite outside the scale range
## (`stat_density()`).</code></pre>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>We can visualize our data using a Multi Dimensional Scaling plot, eg.
as provided by the <code>limma</code> package.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideNorm&quot;</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>  assay <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>  limma<span class="sc">::</span><span class="fu">plotMDS</span>(<span class="at">col =</span> <span class="fu">as.numeric</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>condition))</span></code></pre></div>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>The first axis in the plot is showing the leading log fold changes
(differences on the log scale) between the samples.</p>
<p>We notice that the leading differences (log FC) in the peptide data
seems to be driven by technical variability. Indeed, the samples do not
seem to be clearly separated according to the spike-in condition.</p>
</div>
<div id="summarization-to-protein-level" class="section level2"
number="3.5">
<h2><span class="header-section-number">3.5</span> Summarization to
protein level</h2>
<ul>
<li>We use median summarization in aggregateFeatures.</li>
<li>Note, that this is a suboptimal normalisation procedure!</li>
<li>By default robust summarization is used:
<code>fun = MsCoreUtils::robustSummary()</code></li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a>  <span class="at">fun =</span> matrixStats<span class="sc">::</span>colMedians)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<pre><code>## Aggregated: 1/1</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">col =</span> <span class="fu">as.numeric</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>condition))</span></code></pre></div>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>Note that the samples upon robust summarisation show a separation
according to the spike-in condition in the second dimension of the MDS
plot.</p>
</div>
</div>
<div id="data-analysis" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Data Analysis</h1>
<div id="estimation" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Estimation</h2>
<p>We model the protein level expression values using
<code>msqrob</code>. By default <code>msqrob2</code> estimates the model
parameters using robust regression.</p>
<p>We will model the data with a different group mean. The group is
incoded in the variable <code>condition</code> of the colData. We can
specify this model by using a formula with the factor condition as its
predictor: <code>formula = ~condition</code>.</p>
<p>Note, that a formula always starts with a symbol ‘~’.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span>condition)</span></code></pre></div>
</div>
<div id="inference" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Inference</h2>
<p>First, we extract the parameter names of the model by looking at the
first model. The models are stored in the row data of the assay under
the default name msqrobModels.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">getCoef</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>## (Intercept)  conditionB 
##   -2.793005    1.541958</code></pre>
<p>We can also explore the design of the model that we specified using
the the package <code>ExploreModelMatrix</code></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix)</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span>condition)<span class="sc">$</span>plotlist[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>Spike-in condition <code>A</code> is the reference class. So the mean
log2 expression for samples from condition A is ‘(Intercept). The mean
log2 expression for samples from condition B is’(Intercept)+conditionB’.
Hence, the average log2 fold change between condition b and condition a
is modelled using the parameter ‘conditionB’. Thus, we assess the
contrast ‘conditionB = 0’ with our statistical test.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;conditionB=0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;conditionB&quot;</span>))</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
</div>
<div id="plots" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Plots</h2>
<div id="volcano-plot" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Volcano-plot</h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>conditionB,</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>Note, that only 2 proteins are found to be differentially
abundant.</p>
</div>
<div id="heatmap" class="section level3" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Heatmap</h3>
<p>We first select the names of the proteins that were declared
signficant.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>conditionB <span class="sc">%&gt;%</span></span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;protein&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>  <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>  <span class="fu">pull</span>(protein)</span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]])[sigNames, ])</span></code></pre></div>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>The majority of the proteins are indeed UPS proteins. 1 yeast protein
is returned. Note, that the yeast protein indeed shows evidence for
differential abundance.</p>
</div>
<div id="boxplots" class="section level3" number="4.3.3">
<h3><span class="header-section-number">4.3.3</span> Boxplots</h3>
<p>We make boxplot of the log2 FC and stratify according to the whether
a protein is spiked or not.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>conditionB <span class="sc">%&gt;%</span></span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;protein&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">grepl</span>(<span class="st">&quot;UPS&quot;</span>,protein),<span class="at">y=</span>logFC)) <span class="sc">+</span></span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;UPS&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a>  <span class="fu">geom_segment</span>(</span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">1.5</span>,</span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a>    <span class="at">xend =</span> <span class="fl">2.5</span>,</span>
<span id="cb44-9"><a href="#cb44-9" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">log2</span>(<span class="fl">0.74</span><span class="sc">/</span><span class="fl">0.25</span>),</span>
<span id="cb44-10"><a href="#cb44-10" tabindex="-1"></a>    <span class="at">yend =</span> <span class="fu">log2</span>(<span class="fl">0.74</span><span class="sc">/</span><span class="fl">0.25</span>),</span>
<span id="cb44-11"><a href="#cb44-11" tabindex="-1"></a>    <span class="at">colour=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-12"><a href="#cb44-12" tabindex="-1"></a>  <span class="fu">geom_segment</span>(</span>
<span id="cb44-13"><a href="#cb44-13" tabindex="-1"></a>    <span class="at">x =</span> <span class="fl">0.5</span>,</span>
<span id="cb44-14"><a href="#cb44-14" tabindex="-1"></a>    <span class="at">xend =</span> <span class="fl">1.5</span>,</span>
<span id="cb44-15"><a href="#cb44-15" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">0</span>,</span>
<span id="cb44-16"><a href="#cb44-16" tabindex="-1"></a>    <span class="at">yend =</span> <span class="dv">0</span>,</span>
<span id="cb44-17"><a href="#cb44-17" tabindex="-1"></a>    <span class="at">colour=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-18"><a href="#cb44-18" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb44-19"><a href="#cb44-19" tabindex="-1"></a>    <span class="st">&quot;text&quot;</span>,</span>
<span id="cb44-20"><a href="#cb44-20" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),</span>
<span id="cb44-21"><a href="#cb44-21" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fu">log2</span>(<span class="fl">0.74</span><span class="sc">/</span><span class="fl">0.25</span>))<span class="sc">+</span>.<span class="dv">1</span>,</span>
<span id="cb44-22"><a href="#cb44-22" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">c</span>(</span>
<span id="cb44-23"><a href="#cb44-23" tabindex="-1"></a>      <span class="st">&quot;log2 FC Ecoli = 0&quot;</span>,</span>
<span id="cb44-24"><a href="#cb44-24" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">&quot;log2 FC UPS = &quot;</span>,<span class="fu">round</span>(<span class="fu">log2</span>(<span class="fl">0.74</span><span class="sc">/</span><span class="fl">0.25</span>),<span class="dv">2</span>))</span>
<span id="cb44-25"><a href="#cb44-25" tabindex="-1"></a>      ),</span>
<span id="cb44-26"><a href="#cb44-26" tabindex="-1"></a>    <span class="at">colour =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 166 rows containing non-finite outside the scale range
## (`stat_boxplot()`).</code></pre>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>What do you observe?</p>
</div>
<div id="detail-plots" class="section level3" number="4.3.4">
<h3><span class="header-section-number">4.3.4</span> Detail plots</h3>
<p>We first extract the normalized peptideRaw expression values for a
particular protein.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="cf">for</span> (protName <span class="cf">in</span> sigNames)</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>{</span>
<span id="cb46-3"><a href="#cb46-3" tabindex="-1"></a>pePlot <span class="ot">&lt;-</span> pe[protName, , <span class="fu">c</span>(<span class="st">&quot;peptideNorm&quot;</span>,<span class="st">&quot;protein&quot;</span>)]</span>
<span id="cb46-4"><a href="#cb46-4" tabindex="-1"></a>pePlotDf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">longForm</span>(pePlot))</span>
<span id="cb46-5"><a href="#cb46-5" tabindex="-1"></a>pePlotDf<span class="sc">$</span>assay <span class="ot">&lt;-</span> <span class="fu">factor</span>(pePlotDf<span class="sc">$</span>assay,</span>
<span id="cb46-6"><a href="#cb46-6" tabindex="-1"></a>                        <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;peptideNorm&quot;</span>, <span class="st">&quot;protein&quot;</span>))</span>
<span id="cb46-7"><a href="#cb46-7" tabindex="-1"></a>pePlotDf<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">colData</span>(pePlot)[pePlotDf<span class="sc">$</span>colname, <span class="st">&quot;condition&quot;</span>])</span>
<span id="cb46-8"><a href="#cb46-8" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" tabindex="-1"></a><span class="co"># plotting</span></span>
<span id="cb46-10"><a href="#cb46-10" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> pePlotDf,</span>
<span id="cb46-11"><a href="#cb46-11" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> colname, <span class="at">y =</span> value, <span class="at">group =</span> rowname)) <span class="sc">+</span></span>
<span id="cb46-12"><a href="#cb46-12" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span> </span>
<span id="cb46-13"><a href="#cb46-13" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span>  </span>
<span id="cb46-14"><a href="#cb46-14" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">70</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb46-15"><a href="#cb46-15" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="sc">~</span>assay) <span class="sc">+</span> </span>
<span id="cb46-16"><a href="#cb46-16" tabindex="-1"></a>    <span class="fu">ggtitle</span>(protName)</span>
<span id="cb46-17"><a href="#cb46-17" tabindex="-1"></a><span class="fu">print</span>(p1)</span>
<span id="cb46-18"><a href="#cb46-18" tabindex="-1"></a></span>
<span id="cb46-19"><a href="#cb46-19" tabindex="-1"></a><span class="co"># plotting 2</span></span>
<span id="cb46-20"><a href="#cb46-20" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pePlotDf, <span class="fu">aes</span>(<span class="at">x =</span> colname, <span class="at">y =</span> value, <span class="at">fill =</span> condition)) <span class="sc">+</span></span>
<span id="cb46-21"><a href="#cb46-21" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb46-22"><a href="#cb46-22" tabindex="-1"></a>  <span class="fu">geom_point</span>(</span>
<span id="cb46-23"><a href="#cb46-23" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> .<span class="dv">1</span>),</span>
<span id="cb46-24"><a href="#cb46-24" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">shape =</span> rowname)) <span class="sc">+</span></span>
<span id="cb46-25"><a href="#cb46-25" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pePlotDf)) <span class="sc">+</span></span>
<span id="cb46-26"><a href="#cb46-26" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> protName, <span class="at">x =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">y =</span> <span class="st">&quot;peptide intensity (log2)&quot;</span>) <span class="sc">+</span> </span>
<span id="cb46-27"><a href="#cb46-27" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">70</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb46-28"><a href="#cb46-28" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>assay)</span>
<span id="cb46-29"><a href="#cb46-29" tabindex="-1"></a><span class="fu">print</span>(p2)</span>
<span id="cb46-30"><a href="#cb46-30" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="cptac_median_files/figure-html/unnamed-chunk-25-1.png" width="672" /><img src="cptac_median_files/figure-html/unnamed-chunk-25-2.png" width="672" /><img src="cptac_median_files/figure-html/unnamed-chunk-25-3.png" width="672" /><img src="cptac_median_files/figure-html/unnamed-chunk-25-4.png" width="672" /></p>
<p>Note, that the yeast protein is only covered by 3 peptides. Only one
peptide is picked up in condition A. This peptide is also only once
observed in spike-in condition B. This puts a considerable burden upon
the inference and could be avoided by more stringent filtering.</p>
</div>
</div>
</div>
<div id="session-info" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Session Info</h1>
<p>With respect to reproducibility, it is highly recommended to include
a session info in your script so that readers of your output can see
your particular setup of R.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.4.0 RC (2024-04-16 r86468)
## Platform: aarch64-apple-darwin20
## Running under: macOS 15.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Europe/Brussels
## tzcode source: internal
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] ExploreModelMatrix_1.16.0   plotly_4.10.4              
##  [3] msqrob2_1.15.1              QFeatures_1.19.3           
##  [5] MultiAssayExperiment_1.34.0 SummarizedExperiment_1.34.0
##  [7] Biobase_2.64.0              GenomicRanges_1.56.1       
##  [9] GenomeInfoDb_1.40.1         IRanges_2.38.1             
## [11] S4Vectors_0.42.1            BiocGenerics_0.54.0        
## [13] generics_0.1.3              MatrixGenerics_1.16.0      
## [15] matrixStats_1.4.1           limma_3.60.6               
## [17] lubridate_1.9.3             forcats_1.0.0              
## [19] stringr_1.5.1               dplyr_1.1.4                
## [21] purrr_1.0.2                 readr_2.1.5                
## [23] tidyr_1.3.1                 tibble_3.2.1               
## [25] ggplot2_3.5.1               tidyverse_2.0.0            
## 
## loaded via a namespace (and not attached):
##  [1] rlang_1.1.4             magrittr_2.0.3          shinydashboard_0.7.2   
##  [4] clue_0.3-65             compiler_4.4.0          vctrs_0.6.5            
##  [7] reshape2_1.4.4          ProtGenerics_1.36.0     pkgconfig_2.0.3        
## [10] crayon_1.5.3            fastmap_1.2.0           XVector_0.44.0         
## [13] labeling_0.4.3          utf8_1.2.4              promises_1.3.0         
## [16] rmarkdown_2.28          tzdb_0.4.0              UCSC.utils_1.0.0       
## [19] nloptr_2.1.1            xfun_0.47               zlibbioc_1.50.0        
## [22] cachem_1.1.0            jsonlite_1.8.9          later_1.3.2            
## [25] highr_0.11              DelayedArray_0.30.1     BiocParallel_1.38.0    
## [28] parallel_4.4.0          cluster_2.1.6           R6_2.5.1               
## [31] bslib_0.8.0             stringi_1.8.4           boot_1.3-31            
## [34] jquerylib_0.1.4         Rcpp_1.0.13-1           knitr_1.48             
## [37] BiocBaseUtils_1.10.0    httpuv_1.6.15           Matrix_1.7-0           
## [40] splines_4.4.0           igraph_2.0.3            timechange_0.3.0       
## [43] tidyselect_1.2.1        rstudioapi_0.16.0       abind_1.4-8            
## [46] yaml_2.3.10             codetools_0.2-20        lattice_0.22-6         
## [49] plyr_1.8.9              shiny_1.9.1             withr_3.0.1            
## [52] evaluate_1.0.0          pillar_1.9.0            DT_0.33                
## [55] shinyjs_2.1.0           hms_1.1.3               munsell_0.5.1          
## [58] scales_1.3.0            minqa_1.2.8             xtable_1.8-4           
## [61] glue_1.8.0              lazyeval_0.2.2          tools_4.4.0            
## [64] data.table_1.17.6       lme4_1.1-35.5           cowplot_1.1.3          
## [67] grid_4.4.0              MsCoreUtils_1.16.1      colorspace_2.1-1       
## [70] nlme_3.1-166            GenomeInfoDbData_1.2.12 cli_3.6.3              
## [73] fansi_1.0.6             S4Arrays_1.4.1          viridisLite_0.4.2      
## [76] AnnotationFilter_1.28.0 gtable_0.3.5            rintrojs_0.3.4         
## [79] sass_0.4.9              digest_0.6.37           SparseArray_1.4.8      
## [82] htmlwidgets_1.6.4       farver_2.1.2            htmltools_0.5.8.1      
## [85] lifecycle_1.0.4         httr_1.4.7              mime_0.12              
## [88] statmod_1.5.0           MASS_7.3-61</code></pre>
</div>

<div id="rmd-source-code">---
title: "Introduction to proteomics data analysis: median summarization"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: flatly
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue

bibliography: msqrob2.bib

---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

This is part of the online course [Proteomics Data Analysis (PDA)](https://statomics.github.io/PDA22GTPB/)

# Background
This case-study is a subset of the data of the 6th study of the Clinical
Proteomic Technology Assessment for Cancer (CPTAC).
In this experiment, the authors spiked the Sigma Universal Protein Standard
mixture 1 (UPS1) containing 48 different human proteins in a protein background
of 60 ng/$\mu$L Saccharomyces cerevisiae strain BY4741.
Two different spike-in concentrations were used:
6A (0.25 fmol UPS1 proteins/$\mu$L) and 6B (0.74 fmol UPS1 proteins/$\mu$L) [5].
We limited ourselves to the data of LTQ-Orbitrap W at site 56.
The data were searched with MaxQuant version 1.5.2.8, and
detailed search settings were described in Goeminne et al. (2016) [1].
Three replicates are available for each concentration.


- NOTE THAT MEDIAN SUMMARISATION IS SUBOPTIMAL! 
- THIS IS FOR DIDACTICAL PURPOSES ONLY.

# Data

We first import the data from peptideRaws.txt file. This is the file containing
your peptideRaw-level intensities. For a MaxQuant search [6],
this peptideRaws.txt file can be found by default in the
"path_to_raw_files/combined/txt/" folder from the MaxQuant output,
with "path_to_raw_files" the folder where the raw files were saved.
In this vignette, we use a MaxQuant peptideRaws file which is a subset
of the cptac study. This data is available in the `msdata` package.
To import the data we use the `QFeatures` package.

We generate the object peptideRawFile with the path to the peptideRaws.txt file.
Using the `grepEcols` function, we find the columns that contain the expression
data of the peptideRaws in the peptideRaws.txt file.


```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)

peptidesFile <- "https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/cptacAvsB_lab3/peptides.txt"

ecols <- grep(
  "Intensity\\.", 
  names(read.delim(peptidesFile))
  )
```

Next, we read the data and store it in QFeatures object

```{r}
pe <- readQFeatures(
  assayData = read.delim(peptidesFile),
  fnames = 1,
  quantCols = ecols,
  name = "peptideRaw")
```

The QFeatures object pe currently contains a single assay, named peptideRaw.

We extract the column names from the peptideRaw assay and see that this contains information about the spike-in condition.

```{r}
colnames(pe[["peptideRaw"]])
```

We rename the colnames by dropping the "Intensity." from the name.

```{r}
(newNames <-  sub(
  pattern = "Intensity\\.", 
  replacement = "",
  colnames(pe[["peptideRaw"]]))
 )
```

```{r}
pe <- renameColname(pe, 
                    i = "peptideRaw",
                    newNames)
pe <- renamePrimary(pe, newNames)
colnames(pe[["peptideRaw"]])
```

In the following code chunk, we add the spikein condition that we can read in the raw file name to the colData.

```{r}
colData(pe)$condition <-  
  colnames(pe[["peptideRaw"]]) %>%
  substr(start = 2, stop = 2) %>%
  as.factor
colData(pe)$condition
```

We calculate how many non zero intensities we have per peptide and this
will be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```


Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.
```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```


## Data exploration

`r format(mean(is.na(assay(pe[["peptideRaw"]])))*100,digits=2)`% of all peptide
intensities are missing and for some peptides we do not even measure a signal
in any sample.


# Preprocessing

This section preforms preprocessing for the peptide data. 
This include 

- log transformation, 
- filtering and 
- summarisation of the data.

## Log transform the data

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

## Filtering

1. Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))
```

2. Remove reverse sequences (decoys) and contaminants

We now remove the contaminants and peptides that map to decoy sequences.

```{r}
pe <- filterFeatures(pe,~ Reverse != "+")
pe <- filterFeatures(pe,~ Potential.contaminant != "+")
```

3. Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe <- filterFeatures(pe,~ nNonZero >=2)
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.


## Normalize the data using median centering 

We normalize the data by substracting the sample median from every intensity for peptide $p$  in a sample $i$: 

$$y_{ip}^\text{norm} = y_{ip} - \hat\mu_i$$ 

with $\hat\mu_i$ the median intensity over all observed peptides in sample $i$.

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```


## Explore  normalized data

Upon the normalisation the density curves are nicely registered

```{r}
pe[["peptideNorm"]] %>% 
  assay %>%
  as.data.frame() %>%
  gather(sample, intensity) %>% 
  mutate(condition = colData(pe)[sample,"condition"]) %>%
  ggplot(aes(x = intensity,group = sample,color = condition)) + 
    geom_density()
```

We can visualize our data using a Multi Dimensional Scaling plot,
eg. as provided by the `limma` package.

```{r}
pe[["peptideNorm"]] %>% 
  assay %>%
  limma::plotMDS(col = as.numeric(colData(pe)$condition))
```

The first axis in the plot is showing the leading log fold changes
(differences on the log scale) between the samples.

We notice that the leading differences (log FC)
in the peptide data seems to be driven by technical variability.
Indeed, the samples do not seem to be clearly separated according
to the spike-in condition.


## Summarization to protein level


- We use median summarization in aggregateFeatures.
- Note, that this is a suboptimal normalisation procedure! 
- By default robust summarization is used:  `fun = MsCoreUtils::robustSummary()`

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
  i = "peptideNorm",
  fcol = "Proteins",
  na.rm = TRUE,
  name = "protein",
  fun = matrixStats::colMedians)
```



```{r}
plotMDS(assay(pe[["protein"]]), col = as.numeric(colData(pe)$condition))
```

Note that the samples upon robust summarisation show a separation according to the spike-in condition in the second dimension of the MDS plot.

# Data Analysis

## Estimation

We model the protein level expression values using `msqrob`.
By default `msqrob2` estimates the model parameters using robust regression.

We will model the data with a different group mean. 
The group is incoded in the variable `condition` of the colData. 
We can specify this model by using a formula with the factor condition as its predictor: 
`formula = ~condition`.

Note, that a formula always starts with a symbol '~'.

```{r, warning=FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~condition)
```

## Inference

First, we extract the parameter names of the model by looking at the first model. 
The models are stored in the row data of the assay under the default name msqrobModels. 

```{r}
getCoef(rowData(pe[["protein"]])$msqrobModels[[1]])
```

We can also explore the design of the model that we specified using the the package `ExploreModelMatrix` 

```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~condition)$plotlist[[1]]
```

Spike-in condition `A` is the reference class. So the mean log2 expression
for samples from condition A is '(Intercept).
The mean log2 expression for samples from condition B is '(Intercept)+conditionB'.
Hence, the average log2 fold change between condition b and
condition a is modelled using the parameter 'conditionB'.
Thus, we assess the contrast 'conditionB = 0' with our statistical test.

```{r}
L <- makeContrast("conditionB=0", parameterNames = c("conditionB"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
```


## Plots

### Volcano-plot


```{r,warning=FALSE}
volcano <- ggplot(rowData(pe[["protein"]])$conditionB,
                  aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
  geom_point(cex = 2.5) +
  scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcano
```

Note, that only `r sum(rowData(pe[["protein"]])$conditionB$adjPval < 0.05, na.rm = TRUE)` proteins are found to be differentially abundant.

### Heatmap

We first select the names of the proteins that were declared signficant.

```{r}
sigNames <- rowData(pe[["protein"]])$conditionB %>%
  rownames_to_column("protein") %>%
  filter(adjPval<0.05) %>%
  pull(protein)
heatmap(assay(pe[["protein"]])[sigNames, ])
```

The majority of the proteins are indeed UPS proteins. 
1 yeast protein is returned. 
Note, that the yeast protein indeed shows evidence for differential abundance. 

### Boxplots

We make boxplot of the log2 FC and stratify according to the whether a protein is spiked or not.

```{r}
rowData(pe[["protein"]])$conditionB %>%
  rownames_to_column(var = "protein") %>%
  ggplot(aes(x=grepl("UPS",protein),y=logFC)) +
  geom_boxplot() +
  xlab("UPS") +
  geom_segment(
    x = 1.5,
    xend = 2.5,
    y = log2(0.74/0.25),
    yend = log2(0.74/0.25),
    colour="red") +
  geom_segment(
    x = 0.5,
    xend = 1.5,
    y = 0,
    yend = 0,
    colour="red") +
  annotate(
    "text",
    x = c(1,2),
    y = c(0,log2(0.74/0.25))+.1,
    label = c(
      "log2 FC Ecoli = 0",
      paste0("log2 FC UPS = ",round(log2(0.74/0.25),2))
      ),
    colour = "red")
```

What do you observe?

### Detail plots

We first extract the normalized peptideRaw expression values for a particular protein.  


```{r, warning=FALSE, message=FALSE}
for (protName in sigNames)
{
pePlot <- pe[protName, , c("peptideNorm","protein")]
pePlotDf <- data.frame(longForm(pePlot))
pePlotDf$assay <- factor(pePlotDf$assay,
                        levels = c("peptideNorm", "protein"))
pePlotDf$condition <- as.factor(colData(pePlot)[pePlotDf$colname, "condition"])

# plotting
p1 <- ggplot(data = pePlotDf,
       aes(x = colname, y = value, group = rowname)) +
    geom_line() + 
    geom_point() +  
    theme(axis.text.x = element_text(angle = 70, hjust = 1, vjust = 0.5)) +
    facet_grid(~assay) + 
    ggtitle(protName)
print(p1)

# plotting 2
p2 <- ggplot(pePlotDf, aes(x = colname, y = value, fill = condition)) +
  geom_boxplot(outlier.shape = NA) + 
  geom_point(
    position = position_jitter(width = .1),
    aes(shape = rowname)) +
  scale_shape_manual(values = 1:nrow(pePlotDf)) +
  labs(title = protName, x = "sample", y = "peptide intensity (log2)") + 
  theme(axis.text.x = element_text(angle = 70, hjust = 1, vjust = 0.5)) +
  facet_grid(~assay)
print(p2)
}
```

Note, that the yeast protein is only covered by 3 peptides. 
Only one peptide is picked up in condition A. 
This peptide is also only once observed in spike-in condition B. 
This puts a considerable burden upon the inference and could be avoided by more stringent filtering. 

# Session Info

With respect to reproducibility, it is highly recommended to include a session info in your script so that readers of your output can see your particular setup of R. 

```{r}
sessionInfo()
```
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("cptac_median.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
