<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Introduction to proteomics data analysis: robust summarization</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.1 MS Basics</a>
    </li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.2 Identification</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1 Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2 Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1 Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Introduction to proteomics data analysis:
robust summarization</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a
href="https://statomics.github.io"
class="uri">https://statomics.github.io</a>)</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a
href="https://statomics.github.io/SGA/">Statistical Genomics Analysis
(SGA)</a></p>
<div id="background" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Background</h1>
<p>A study on the facultative pathogen Francisella tularensis was
conceived by Ramond et al. (2015) [12]. F. tularensis enters the cells
of its host by phagocytosis. The authors showed that F. tularensis is
arginine deficient and imports arginine from the host cell via an
arginine transporter, ArgP, in order to efficiently escape from the
phagosome and reach the cytosolic compartment, where it can actively
multiply. In their study, they compared the proteome of wild type F.
tularensis (WT) to ArgP-gene deleted F. tularensis (knock-out, D8). For
this exercise, we use a subset of the F. tularensis dataset where
bacterial cultures were grown in biological quadruplicate and each
biorep was run in technical triplicate on a nanoRSLC-Q Exactive PLUS
instrument. The data were searched with MaxQuant version 1.4.1.2. and
are available on the PRIDE repository: <a
href="https://www.ebi.ac.uk/pride/archive/projects/PXD001584">PXD001584</a>.</p>
</div>
<div id="data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>peptidesFile <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/master/Francisella/data/maxquant/peptides.txt&quot;</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>ecols <span class="ot">&lt;-</span> <span class="fu">grep</span>(</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>  <span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, </span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>  <span class="fu">names</span>(<span class="fu">read.delim</span>(peptidesFile))</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>  )</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>  <span class="at">assayData =</span> <span class="fu">read.delim</span>(peptidesFile),</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>  <span class="at">quantCols =</span> ecols,</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="fu">colnames</span>(pe)</span></code></pre></div>
<pre><code>## CharacterList of length 1
## [[&quot;peptideRaw&quot;]] Intensity.1WT_20_2h_n3_1 ... Intensity.3D8_20_2h_n5_3</code></pre>
<p>The annotation can be derived from the file name.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">substr</span>(<span class="fu">colnames</span>(pe[[<span class="dv">1</span>]]),<span class="dv">12</span>,<span class="dv">13</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>                          as.factor</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>biorep  <span class="ot">&lt;-</span> <span class="fu">paste</span>(</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="fu">substr</span>(<span class="fu">colnames</span>(pe[[<span class="dv">1</span>]]),<span class="dv">12</span>,<span class="dv">13</span>),</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="fu">substr</span>(<span class="fu">colnames</span>(pe[[<span class="dv">1</span>]]),<span class="dv">21</span>,<span class="dv">22</span>),</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">sep=</span><span class="st">&quot;_&quot;</span>) <span class="sc">%&gt;%</span> as.factor</span></code></pre></div>
<p>We calculate how many non zero intensities we have per peptide and
this is often useful for filtering.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZero <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code></pre></div>
<p>Because every biorep is assessed in technical triplicate, we will
also calculate the number of biorepeats in which each peptide is
observed.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZeroBiorep <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="fu">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]), </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="dv">1</span>, </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="cf">function</span>(intensity) </span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    <span class="fu">colData</span>(pe)<span class="sc">$</span>biorep[intensity<span class="sc">&gt;</span><span class="dv">0</span>] <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    unique <span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    length)</span></code></pre></div>
<p>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<div id="data-exploration" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Data exploration</h2>
<p>49% of all peptide intensities are missing.</p>
</div>
</div>
<div id="preprocessing" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Preprocessing</h1>
<p>This section preforms preprocessing for the peptide data. This
include</p>
<ul>
<li>log transformation,</li>
<li>filtering and</li>
<li>summarisation of the data.</li>
</ul>
<div id="log-transform-the-data" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Log transform the
data</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
</div>
<div id="filtering" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Filtering</h2>
<ol style="list-style-type: decimal">
<li>Handling overlapping protein groups</li>
</ol>
<p>In our approach a peptide can map to multiple proteins, as long as
there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] <span class="ot">&lt;-</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a> pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a> <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins),]</span></code></pre></div>
<pre><code>## Warning in replaceAssay(x = x, y = value, i = i): Links between assays were
## lost/removed during replacement. See &#39;?addAssayLink&#39; to restore them manually.</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants and peptides that map to decoy
sequences.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] <span class="ot">&lt;-</span> pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>, ]</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] <span class="ot">&lt;-</span> pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>    Contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>, ]</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in a single biorepeat</li>
</ol>
<p>Note, that in experiments without technical repeats we filter on the
number of samples in which a peptide is picked up. Here, we will require
that a peptide is picked up in at least two biorepeats.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideLog&quot;</span>]] <span class="ot">&lt;-</span> pe[[<span class="st">&quot;peptideLog&quot;</span>]][<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>nNonZeroBiorep <span class="sc">&gt;=</span> <span class="dv">2</span>, ]</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 7542</code></pre>
<p>We keep 7542 peptides upon filtering.</p>
</div>
<div id="normalize-the-data-using-median-centering"
class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Normalize the data
using median centering</h2>
<p>We normalize the data by substracting the sample median from every
intensity for peptide <span class="math inline">\(p\)</span> in a sample
<span class="math inline">\(i\)</span>:</p>
<p><span class="math display">\[y_{ip}^\text{norm} = y_{ip} -
\hat\mu_i\]</span></p>
<p>with <span class="math inline">\(\hat\mu_i\)</span> the median
intensity over all observed peptides in sample <span
class="math inline">\(i\)</span>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</div>
<div id="explore-normalized-data" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Explore normalized
data</h2>
<p>Upon the normalisation the density curves are nicely registered</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideNorm&quot;</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  assay <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="fu">gather</span>(sample, intensity) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">biorep =</span> <span class="fu">colData</span>(pe)[sample,<span class="st">&quot;biorep&quot;</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> intensity,<span class="at">group =</span> sample,<span class="at">color =</span> biorep)) <span class="sc">+</span> </span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>    <span class="fu">geom_density</span>()</span></code></pre></div>
<pre><code>## Warning: Removed 40413 rows containing non-finite outside the scale range
## (`stat_density()`).</code></pre>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>We can visualize our data using a Multi Dimensional Scaling plot, eg.
as provided by the <code>limma</code> package.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideNorm&quot;</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  assay <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  limma<span class="sc">::</span><span class="fu">plotMDS</span>(<span class="at">col =</span> <span class="fu">as.numeric</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>genotype),<span class="at">label=</span><span class="fu">colData</span>(pe)<span class="sc">$</span>biorep)</span></code></pre></div>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>The first axis in the plot is showing the leading log fold changes
(differences on the log scale) between the samples. As expected the
samples of the three MS-runs from the same biorepeat cluster together,
indicating that biorepeat is a leading source of variability.</p>
</div>
<div id="summarization-to-protein-level" class="section level2"
number="3.5">
<h2><span class="header-section-number">3.5</span> Summarization to
protein level</h2>
<ul>
<li>By default robust summarization is used:
<code>fun = MsCoreUtils::robustSummary()</code></li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>  <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>  <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<pre><code>## Aggregated: 1/1</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>pe[[<span class="st">&quot;protein&quot;</span>]]<span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>  assay <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>  limma<span class="sc">::</span><span class="fu">plotMDS</span>(<span class="at">col =</span> <span class="fu">as.numeric</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>genotype),<span class="at">label=</span><span class="fu">colData</span>(pe)<span class="sc">$</span>biorep)</span></code></pre></div>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>Note that the samples upon robust summarisation show a clear
separation according to the genotype in the first dimension of the MDS
plot and shows again that the samples from the different MS-runs for the
same biorepeat cluster together.</p>
</div>
</div>
<div id="data-analysis" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Data Analysis</h1>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span> genotype <span class="sc">+</span> biorep)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Unlike the mouse example, we cannot correct for biorepeat using a
randomized complete block analysis. Indeed, once we condition on
biorepeat, we also condition on the genotype. Hence, when infering on
the treatment effect we should correct for both the variability both
within and between biorepeat.</p>
<p>In the original paper, the authors simply ignored the correlation in
the data according to biorepeat and INCORRECTLY act as if each technical
repeat was an independent sample. So they act as if they have a 9 by 9
group comparison instead of correctly accounting that they only have
three independent biorepeats for each genotype.</p>
<p>We will fit this WRONG model to see how this impacts the data
analysis.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span> genotype)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span> genotype,<span class="at">overwrite=</span><span class="cn">TRUE</span>, <span class="at">modelColumnName=</span><span class="st">&quot;wrongModel&quot;</span>)</span></code></pre></div>
<p>Statistical inference: log2FC between WT and D8 is quantified by
<code>genotypeWT</code>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;genotypeWT=0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;genotypeWT&quot;</span>))</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L, <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">modelColumn =</span> <span class="st">&quot;wrongModel&quot;</span>, <span class="at">resultsColumnNamePrefix =</span> <span class="st">&quot;wrongModel-&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span><span class="st">`</span><span class="at">wrongModel-genotypeWT</span><span class="st">`</span>,</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p>Note, that 319 proteins are found to be differentially abundant
according to the WRONG DATA ANALYSIS that ignores that fact that
intensities from technical repeats (MS-runs) from the same biorepeat are
correlated.</p>
<p>As the analysis acts as if there are 9 independent repeats for each
genotype, we can expect the analysis to be overly liberal. So we can
expect many false positive results.</p>
<div id="correct-data-analysis" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Correct Data
analysis</h2>
<p>We model the protein level expression values using
<code>msqrob</code>. By default <code>msqrob2</code> estimates the model
parameters using robust regression.</p>
<p>We will model the data with a different group mean. The group is
incoded in the variable <code>genotype</code> of the colData. We will
also have to include a random effect for bio-repeat to address the
pseudo-replication in the experiment. Indeed, the data from the same
bio-repeat will be correlated!</p>
<p>We can specify this model by using a formula with the factor
<code>genotype</code> as a fixed effect and as the factor
<code>biorep</code> a random effect:
<code>formula = ~genotype + (1|biorep)</code>.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span> genotype <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>biorep),<span class="at">overwrite=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Warning: &#39;experiments&#39; dropped; see &#39;drops()&#39;</code></pre>
<p>For this model we can also use the same contrast. Indeed, the model
now contains a parameter for the intercept, genotype and a random
intercept for each biorepeat.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="fu">getCoef</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>##            (Intercept)             genotypeWT (Intercept)biorepD8_n3 
##            -0.01438052             0.10487683            -0.22937840 
## (Intercept)biorepD8_n4 (Intercept)biorepD8_n5 (Intercept)biorepWT_n3 
##             0.09661496             0.13276344            -0.14908511 
## (Intercept)biorepWT_n4 (Intercept)biorepWT_n5 
##            -0.05118080             0.20026592</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L,<span class="at">overwrite=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span><span class="st">`</span><span class="at">genotypeWT</span><span class="st">`</span>,</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>volcano</span></code></pre></div>
<p><img src="francisellaMixedModel_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>Note, that 164 proteins are found to be differentially abundant.
Hence, when we address the correlation structure in the data, we can
indeed see that the analysis returns way less DA proteins than with the
incorrect analysis of the original authors who acted as if all technical
replicates where independent repeats.</p>
<p>We indeed see that the mixed model is addressing the correlation
structure in the data by . <span class="math display">\[
\left\{
\begin{array}{rcl}
y_i &amp;=&amp; \beta_o + \beta_{WT} x_{WT,i} + u_1 z_{b1,i} + u_2
z_{b2,i} + u_1 z_{b3,i} + u_4 z_{b4,i} + u_5z_{b5,i} + u_6z_{b6,i} +
\epsilon_i\\
u_j &amp;\sim&amp; N(0, \sigma_u^2)\\
\epsilon_i &amp;\sim&amp; N(0,\sigma_\epsilon^2)
\end{array}
\right.,
\]</span> with <span class="math inline">\(u_j\)</span> random effects
that model the correlation in the intensities from technical repeats
(MS-runs) the originate from the same bio-repeat. These random
intercepts are assumed to be i.i.d. normally distributed with mean 0 and
variance <span class="math inline">\(\sigma^2_u\)</span>. Hence, <span
class="math inline">\(\sigma^2_u\)</span> models the between biorepeat
variability and <span class="math inline">\(\sigma^2_\epsilon\)</span>
the within biorepeat variability.</p>
<p>We can also write the data in matrix form</p>
<p><span class="math display">\[
\mathbf{Y} = \mathbf{X}\boldsymbol{\beta} + \mathbf{Zu} +
\boldsymbol{\epsilon}
\]</span> with <span
class="math display">\[\mathbf{Y}=\left[\begin{array}{c} y_1 \\\vdots\\
y_{18}\end{array}\right],
\mathbf{X}=\left[\begin{array}{cc}1&amp;x_{WT,1}\\\vdots&amp;\vdots\\1&amp;x_{WT,18}\end{array}\right],
\boldsymbol{\beta}=\left[\begin{array}{c} \beta_0 \\ \beta_{WT}
\end{array}\right],
\mathbf{Z}=\left[\begin{array}{ccc}z_{b1,1}&amp;\ldots&amp;z_{b6,1}\\\vdots&amp;&amp;\vdots\\z_{b1,18}&amp;\ldots&amp;z_{b6,18}\end{array}\right],
\mathbf{u}=\left[\begin{array}{c} u_1 \\\vdots\\
u_{6}\end{array}\right]\]</span></p>
<p>Hence, with the mixed model, the variance covariance matrix of the
intensities becomes <span class="math display">\[
\begin{array}{rcl}
\boldsymbol{\Sigma}_\mathbf{Y} &amp;=&amp;
\text{var}\left(\mathbf{Y}\right) \\
&amp;=&amp; \text{var}\left(\mathbf{X}\boldsymbol{\beta} + \mathbf{Zu} +
\boldsymbol{\epsilon}\right) \\
&amp;=&amp; \mathbf{Z}\text{var}\left(\mathbf{u}\right)\mathbf{Z}^T +
\mathbf{I}\sigma_\epsilon^2\\
&amp;=&amp; \mathbf{Z}\mathbf{Z}^T\sigma^2_u +
\mathbf{I}\sigma_\epsilon^2
\end{array}
\]</span></p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>genotype,<span class="fu">colData</span>(pe))</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>Z <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~-</span><span class="dv">1</span><span class="sc">+</span>biorep,<span class="fu">colData</span>(pe))</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a><span class="fu">rownames</span>(Z) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;b&quot;</span>,<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>,<span class="at">each=</span><span class="dv">3</span>),<span class="st">&quot;,&quot;</span>,<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">6</span>))</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>Z<span class="sc">%*%</span><span class="fu">t</span>(Z)</span></code></pre></div>
<pre><code>##      b1,1 b1,2 b1,3 b2,1 b2,2 b2,3 b3,1 b3,2 b3,3 b4,1 b4,2 b4,3 b5,1 b5,2 b5,3
## b1,1    1    1    1    0    0    0    0    0    0    0    0    0    0    0    0
## b1,2    1    1    1    0    0    0    0    0    0    0    0    0    0    0    0
## b1,3    1    1    1    0    0    0    0    0    0    0    0    0    0    0    0
## b2,1    0    0    0    1    1    1    0    0    0    0    0    0    0    0    0
## b2,2    0    0    0    1    1    1    0    0    0    0    0    0    0    0    0
## b2,3    0    0    0    1    1    1    0    0    0    0    0    0    0    0    0
## b3,1    0    0    0    0    0    0    1    1    1    0    0    0    0    0    0
## b3,2    0    0    0    0    0    0    1    1    1    0    0    0    0    0    0
## b3,3    0    0    0    0    0    0    1    1    1    0    0    0    0    0    0
## b4,1    0    0    0    0    0    0    0    0    0    1    1    1    0    0    0
## b4,2    0    0    0    0    0    0    0    0    0    1    1    1    0    0    0
## b4,3    0    0    0    0    0    0    0    0    0    1    1    1    0    0    0
## b5,1    0    0    0    0    0    0    0    0    0    0    0    0    1    1    1
## b5,2    0    0    0    0    0    0    0    0    0    0    0    0    1    1    1
## b5,3    0    0    0    0    0    0    0    0    0    0    0    0    1    1    1
## b6,1    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
## b6,2    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
## b6,3    0    0    0    0    0    0    0    0    0    0    0    0    0    0    0
##      b6,1 b6,2 b6,3
## b1,1    0    0    0
## b1,2    0    0    0
## b1,3    0    0    0
## b2,1    0    0    0
## b2,2    0    0    0
## b2,3    0    0    0
## b3,1    0    0    0
## b3,2    0    0    0
## b3,3    0    0    0
## b4,1    0    0    0
## b4,2    0    0    0
## b4,3    0    0    0
## b5,1    0    0    0
## b5,2    0    0    0
## b5,3    0    0    0
## b6,1    1    1    1
## b6,2    1    1    1
## b6,3    1    1    1</code></pre>
<p>So, we see that the correlation of the data from the same biological
repeat are correctly addressed and that the data from distinct
biorepeats are assumed to be independent. Hence, the variance-covariance
matrix of <span class="math inline">\(\mathbf{Y}\)</span> has a block
diagonal structure, with as variance <span
class="math inline">\(\sigma_u^2 + \sigma_\epsilon^2\)</span> and the
covariance between intensities from the same biorepeat equals <span
class="math inline">\(\sigma_u^2\)</span>.</p>
<p><span class="math display">\[
\boldsymbol{\Sigma}_\mathbf{Y} =
\left[\begin{array}{cccccccccc}\sigma^2_u+\sigma^2_\epsilon&amp;\sigma^2_u&amp;\sigma^2_u&amp;0&amp;0&amp;0&amp;\ldots&amp;0&amp;0&amp;0\\
\sigma^2_u&amp;\sigma^2_u+\sigma^2_\epsilon&amp;\sigma^2_u&amp;0&amp;0&amp;0&amp;\ldots&amp;0&amp;0&amp;0\\
\sigma^2_u&amp;\sigma^2_u&amp;\sigma^2_u+\sigma^2_\epsilon&amp;0&amp;0&amp;0&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;\sigma^2_u+\sigma^2_\epsilon&amp;\sigma^2_u&amp;\sigma^2_u&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;\sigma^2_u&amp;\sigma^2_u+\sigma^2_\epsilon&amp;\sigma^2_u&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;\sigma^2_u&amp;\sigma^2_u&amp;\sigma^2_u+\sigma^2_\epsilon&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;0&amp;0&amp;0&amp;\dots&amp;\sigma^2_u+\sigma^2_\epsilon&amp;\sigma^2_u&amp;\sigma^2_u\\
0&amp;0&amp;0&amp;0&amp;0&amp;0&amp;\dots&amp;\sigma^2_u&amp;\sigma^2_u+\sigma^2_\epsilon&amp;\sigma^2_u\\
0&amp;0&amp;0&amp;0&amp;0&amp;0&amp;\dots&amp;\sigma^2_u&amp;\sigma^2_u&amp;\sigma^2_u+\sigma^2_\epsilon\\
\end{array}\right]
\]</span></p>
</div>
</div>
<div id="session-info" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Session Info</h1>
<p>With respect to reproducibility, it is highly recommended to include
a session info in your script so that readers of your output can see
your particular setup of R.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.4.0 RC (2024-04-16 r86468)
## Platform: aarch64-apple-darwin20
## Running under: macOS 15.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Europe/Brussels
## tzcode source: internal
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] ExploreModelMatrix_1.16.0   plotly_4.10.4              
##  [3] msqrob2_1.15.1              QFeatures_1.19.3           
##  [5] MultiAssayExperiment_1.34.0 SummarizedExperiment_1.34.0
##  [7] Biobase_2.64.0              GenomicRanges_1.56.1       
##  [9] GenomeInfoDb_1.40.1         IRanges_2.38.1             
## [11] S4Vectors_0.42.1            BiocGenerics_0.54.0        
## [13] generics_0.1.3              MatrixGenerics_1.16.0      
## [15] matrixStats_1.4.1           limma_3.60.6               
## [17] lubridate_1.9.3             forcats_1.0.0              
## [19] stringr_1.5.1               dplyr_1.1.4                
## [21] purrr_1.0.2                 readr_2.1.5                
## [23] tidyr_1.3.1                 tibble_3.2.1               
## [25] ggplot2_3.5.1               tidyverse_2.0.0            
## 
## loaded via a namespace (and not attached):
##  [1] rlang_1.1.4             magrittr_2.0.3          shinydashboard_0.7.2   
##  [4] clue_0.3-65             compiler_4.4.0          vctrs_0.6.5            
##  [7] reshape2_1.4.4          ProtGenerics_1.36.0     pkgconfig_2.0.3        
## [10] crayon_1.5.3            fastmap_1.2.0           XVector_0.44.0         
## [13] labeling_0.4.3          utf8_1.2.4              promises_1.3.0         
## [16] rmarkdown_2.28          tzdb_0.4.0              UCSC.utils_1.0.0       
## [19] nloptr_2.1.1            xfun_0.47               zlibbioc_1.50.0        
## [22] cachem_1.1.0            jsonlite_1.8.9          later_1.3.2            
## [25] highr_0.11              DelayedArray_0.30.1     BiocParallel_1.38.0    
## [28] parallel_4.4.0          cluster_2.1.6           R6_2.5.1               
## [31] bslib_0.8.0             stringi_1.8.4           boot_1.3-31            
## [34] jquerylib_0.1.4         Rcpp_1.0.13-1           knitr_1.48             
## [37] BiocBaseUtils_1.10.0    httpuv_1.6.15           Matrix_1.7-0           
## [40] splines_4.4.0           igraph_2.0.3            timechange_0.3.0       
## [43] tidyselect_1.2.1        rstudioapi_0.16.0       abind_1.4-8            
## [46] yaml_2.3.10             codetools_0.2-20        lattice_0.22-6         
## [49] plyr_1.8.9              shiny_1.9.1             withr_3.0.1            
## [52] evaluate_1.0.0          pillar_1.9.0            DT_0.33                
## [55] shinyjs_2.1.0           hms_1.1.3               munsell_0.5.1          
## [58] scales_1.3.0            minqa_1.2.8             xtable_1.8-4           
## [61] glue_1.8.0              lazyeval_0.2.2          tools_4.4.0            
## [64] data.table_1.17.6       lme4_1.1-35.5           cowplot_1.1.3          
## [67] grid_4.4.0              MsCoreUtils_1.16.1      colorspace_2.1-1       
## [70] nlme_3.1-166            GenomeInfoDbData_1.2.12 cli_3.6.3              
## [73] fansi_1.0.6             S4Arrays_1.4.1          viridisLite_0.4.2      
## [76] AnnotationFilter_1.28.0 gtable_0.3.5            rintrojs_0.3.4         
## [79] sass_0.4.9              digest_0.6.37           SparseArray_1.4.8      
## [82] htmlwidgets_1.6.4       farver_2.1.2            htmltools_0.5.8.1      
## [85] lifecycle_1.0.4         httr_1.4.7              mime_0.12              
## [88] statmod_1.5.0           MASS_7.3-61</code></pre>
</div>
<div id="references" class="section level1" number="6">
<h1><span class="header-section-number">6</span> References</h1>
</div>

<div id="rmd-source-code">---
title: "Introduction to proteomics data analysis: robust summarization"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    html_document:
      code_download: true
      theme: cosmo
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue

bibliography: msqrob2.bib

---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

This is part of the online course [Statistical Genomics Analysis (SGA)](https://statomics.github.io/SGA/)

# Background
A study on the facultative pathogen Francisella tularensis was conceived by Ramond et al. (2015) [12]. F. tularensis enters the cells of its host by phagocytosis. The authors showed that F. tularensis is arginine deficient and imports arginine from the host cell via an arginine transporter, ArgP, in order to efficiently escape from the phagosome and reach the cytosolic compartment, where it can actively multiply. In their study, they compared the proteome of wild type F. tularensis (WT) to ArgP-gene deleted F. tularensis (knock-out, D8). For this exercise, we use a subset of the F. tularensis dataset where bacterial cultures were grown in biological quadruplicate and each biorep was run in technical triplicate on a nanoRSLC-Q Exactive PLUS instrument. The data were searched with MaxQuant version 1.4.1.2. and are available on the PRIDE repository: [PXD001584](https://www.ebi.ac.uk/pride/archive/projects/PXD001584).


# Data


```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)

peptidesFile <- "https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/master/Francisella/data/maxquant/peptides.txt"

ecols <- grep(
  "Intensity\\.", 
  names(read.delim(peptidesFile))
  )

pe <- readQFeatures(
  assayData = read.delim(peptidesFile),
  quantCols = ecols,
  name = "peptideRaw")

colnames(pe)
```

The annotation can be derived from the file name. 

```{r}
colData(pe)$genotype <- substr(colnames(pe[[1]]),12,13) %>%
                          as.factor
colData(pe)$biorep  <- paste(
  substr(colnames(pe[[1]]),12,13),
  substr(colnames(pe[[1]]),21,22),
  sep="_") %>% as.factor

```

We calculate how many non zero intensities we have per peptide and this is often useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)
```

Because every biorep is assessed in technical triplicate, we will also calculate the number of biorepeats in which each peptide is observed. 

```{r}
rowData(pe[["peptideRaw"]])$nNonZeroBiorep <- apply(
  assay(pe[["peptideRaw"]]), 
  1, 
  function(intensity) 
    colData(pe)$biorep[intensity>0] %>%
    unique %>%
    length)
```

Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.
```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```


## Data exploration

`r format(mean(is.na(assay(pe[["peptideRaw"]])))*100,digits=2)`% of all peptide
intensities are missing.


# Preprocessing

This section preforms preprocessing for the peptide data. 
This include 

- log transformation, 
- filtering and 
- summarisation of the data.

## Log transform the data

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

## Filtering

1. Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe[["peptideLog"]] <-
 pe[["peptideLog"]][rowData(pe[["peptideLog"]])$Proteins
 %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins),]
```

2. Remove reverse sequences (decoys) and contaminants

We now remove the contaminants and peptides that map to decoy sequences.

```{r}
pe[["peptideLog"]] <- pe[["peptideLog"]][rowData(pe[["peptideLog"]])$Reverse != "+", ]
pe[["peptideLog"]] <- pe[["peptideLog"]][rowData(pe[["peptideLog"]])$
    Contaminant != "+", ]
```

3. Drop peptides that were only identified in a single biorepeat

Note, that in experiments without technical repeats we filter on the number of samples in which a peptide is picked up. Here, we will require that a peptide is picked up in at least two biorepeats. 

```{r}
pe[["peptideLog"]] <- pe[["peptideLog"]][rowData(pe[["peptideLog"]])$nNonZeroBiorep >= 2, ]
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.


## Normalize the data using median centering 

We normalize the data by substracting the sample median from every intensity for peptide $p$  in a sample $i$: 

$$y_{ip}^\text{norm} = y_{ip} - \hat\mu_i$$ 

with $\hat\mu_i$ the median intensity over all observed peptides in sample $i$.

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```


## Explore  normalized data

Upon the normalisation the density curves are nicely registered

```{r}
pe[["peptideNorm"]] %>% 
  assay %>%
  as.data.frame() %>%
  gather(sample, intensity) %>% 
  mutate(biorep = colData(pe)[sample,"biorep"]) %>%
  ggplot(aes(x = intensity,group = sample,color = biorep)) + 
    geom_density()
```

We can visualize our data using a Multi Dimensional Scaling plot,
eg. as provided by the `limma` package.

```{r}
pe[["peptideNorm"]] %>% 
  assay %>%
  limma::plotMDS(col = as.numeric(colData(pe)$genotype),label=colData(pe)$biorep)
```

The first axis in the plot is showing the leading log fold changes
(differences on the log scale) between the samples.
As expected the samples of the three MS-runs from the same biorepeat cluster together, indicating that biorepeat is a leading source of variability. 



## Summarization to protein level

- By default robust summarization is used:  `fun = MsCoreUtils::robustSummary()`

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
  i = "peptideNorm",
  fcol = "Proteins",
  na.rm = TRUE,
  name = "protein")
```



```{r}
pe[["protein"]]%>% 
  assay %>%
  limma::plotMDS(col = as.numeric(colData(pe)$genotype),label=colData(pe)$biorep)
```

Note that the samples upon robust summarisation show a clear separation according to the genotype in the first dimension of the MDS plot and shows again that the samples from the different MS-runs for the same biorepeat cluster together.  

# Data Analysis

```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~ genotype + biorep)$plotlist
```

Unlike the mouse example, we cannot correct for biorepeat using a randomized complete block analysis. Indeed, once we condition on biorepeat, we also condition on the genotype. Hence, when infering on the treatment effect we should correct for both the variability both within and between biorepeat. 

In the original paper, the authors simply ignored the correlation in the data according to biorepeat and INCORRECTLY act as if each technical repeat was an independent sample. So they act as if they have a 9 by 9 group comparison instead of correctly accounting that they only have three independent biorepeats for each genotype. 

We will fit this WRONG model to see how this impacts the data analysis. 
```{r}
VisualizeDesign(colData(pe),~ genotype)$plotlist
```

```{r warning=FALSE, message = FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~ genotype,overwrite=TRUE, modelColumnName="wrongModel")
```

Statistical inference: log2FC between WT and D8 is quantified by `genotypeWT`.

```{r}
L <- makeContrast("genotypeWT=0", parameterNames = c("genotypeWT"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L, overwrite = TRUE, modelColumn = "wrongModel", resultsColumnNamePrefix = "wrongModel-")
```

```{r,warning=FALSE}
volcano <- ggplot(rowData(pe[["protein"]])$`wrongModel-genotypeWT`,
                  aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
  geom_point(cex = 2.5) +
  scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcano
```

Note, that `r sum(rowData(pe[["protein"]])[["wrongModel-genotypeWT"]][,"adjPval"] < 0.05, na.rm = TRUE)` proteins are found to be differentially abundant according to the WRONG DATA ANALYSIS that ignores that fact that intensities from technical repeats (MS-runs) from the same biorepeat are correlated.

As the analysis acts as if there are 9 independent repeats for each genotype, we can expect the analysis to be overly liberal. So we can expect many false positive results. 


## Correct Data analysis 

We model the protein level expression values using `msqrob`.
By default `msqrob2` estimates the model parameters using robust regression.

We will model the data with a different group mean. 
The group is incoded in the variable `genotype` of the colData.
We will also have to include a random effect for bio-repeat to address the pseudo-replication in the experiment. 
Indeed, the data from the same bio-repeat will be correlated!

We can specify this model by using a formula with the factor `genotype` as a fixed effect and as the factor `biorep` a random effect: 
`formula = ~genotype + (1|biorep)`.

```{r}
pe <- msqrob(object = pe, i = "protein", formula = ~ genotype + (1|biorep),overwrite=TRUE)
```

For this model we can also use the same contrast. Indeed, the model now contains a parameter for the intercept, genotype and a random intercept for each biorepeat.

```{r}
getCoef(rowData(pe[["protein"]])$msqrobModels[[1]])
pe <- hypothesisTest(object = pe, i = "protein", contrast = L,overwrite=TRUE)
```

```{r,warning=FALSE}
volcano <- ggplot(rowData(pe[["protein"]])$`genotypeWT`,
                  aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
  geom_point(cex = 2.5) +
  scale_color_manual(values = alpha(c("black", "red"), 0.5)) + theme_minimal()
volcano
```

Note, that `r sum(rowData(pe[["protein"]])[["genotypeWT"]][,"adjPval"] < 0.05, na.rm = TRUE)` proteins are found to be differentially abundant.
Hence, when we address the correlation structure in the data, we can indeed see that the analysis returns way less DA proteins than with the incorrect analysis of the original authors who acted as if all technical replicates where independent repeats. 

We indeed see that the mixed model is addressing the correlation structure in the data by . 
$$
\left\{
\begin{array}{rcl}
y_i &=& \beta_o + \beta_{WT} x_{WT,i} + u_1 z_{b1,i} + u_2 z_{b2,i} + u_1 z_{b3,i} + u_4 z_{b4,i} + u_5z_{b5,i} + u_6z_{b6,i} + \epsilon_i\\
u_j &\sim& N(0, \sigma_u^2)\\
\epsilon_i &\sim& N(0,\sigma_\epsilon^2)
\end{array}
\right.,
$$
with $u_j$ random effects that model the correlation in the intensities from technical repeats (MS-runs) the originate from the same bio-repeat. These random intercepts are assumed to be i.i.d. normally distributed with mean 0 and variance $\sigma^2_u$. Hence, $\sigma^2_u$ models the between biorepeat variability and $\sigma^2_\epsilon$ the within biorepeat variability. 

We can also write the data in matrix form 

$$
\mathbf{Y} = \mathbf{X}\boldsymbol{\beta} + \mathbf{Zu} + \boldsymbol{\epsilon}
$$
with 
$$\mathbf{Y}=\left[\begin{array}{c} y_1 \\\vdots\\ y_{18}\end{array}\right], \mathbf{X}=\left[\begin{array}{cc}1&x_{WT,1}\\\vdots&\vdots\\1&x_{WT,18}\end{array}\right],
\boldsymbol{\beta}=\left[\begin{array}{c} \beta_0 \\ \beta_{WT} \end{array}\right],
\mathbf{Z}=\left[\begin{array}{ccc}z_{b1,1}&\ldots&z_{b6,1}\\\vdots&&\vdots\\z_{b1,18}&\ldots&z_{b6,18}\end{array}\right], \mathbf{u}=\left[\begin{array}{c} u_1 \\\vdots\\ u_{6}\end{array}\right]$$ 

Hence, with the mixed model, the variance covariance matrix of the intensities becomes
$$
\begin{array}{rcl}
\boldsymbol{\Sigma}_\mathbf{Y} &=& \text{var}\left(\mathbf{Y}\right) \\
&=& \text{var}\left(\mathbf{X}\boldsymbol{\beta} + \mathbf{Zu} + \boldsymbol{\epsilon}\right) \\
&=& \mathbf{Z}\text{var}\left(\mathbf{u}\right)\mathbf{Z}^T + \mathbf{I}\sigma_\epsilon^2\\
&=& \mathbf{Z}\mathbf{Z}^T\sigma^2_u + \mathbf{I}\sigma_\epsilon^2
\end{array}
$$

```{r}
X <- model.matrix(~genotype,colData(pe))
Z <- model.matrix(~-1+biorep,colData(pe))
rownames(Z) <- paste0("b",rep(1:6,each=3),",",rep(1:3,6))
Z%*%t(Z)
```

So, we see that the correlation of the data from the same biological repeat are correctly addressed and that the data from distinct biorepeats are assumed to be independent. 
Hence, the variance-covariance matrix of $\mathbf{Y}$ has a block diagonal structure, with as variance $\sigma_u^2 + \sigma_\epsilon^2$ and the covariance between intensities from the same biorepeat equals $\sigma_u^2$. 

$$
\boldsymbol{\Sigma}_\mathbf{Y} = \left[\begin{array}{cccccccccc}\sigma^2_u+\sigma^2_\epsilon&\sigma^2_u&\sigma^2_u&0&0&0&\ldots&0&0&0\\
\sigma^2_u&\sigma^2_u+\sigma^2_\epsilon&\sigma^2_u&0&0&0&\ldots&0&0&0\\
\sigma^2_u&\sigma^2_u&\sigma^2_u+\sigma^2_\epsilon&0&0&0&\ldots&0&0&0\\
0&0&0&\sigma^2_u+\sigma^2_\epsilon&\sigma^2_u&\sigma^2_u&\ldots&0&0&0\\
0&0&0&\sigma^2_u&\sigma^2_u+\sigma^2_\epsilon&\sigma^2_u&\ldots&0&0&0\\
0&0&0&\sigma^2_u&\sigma^2_u&\sigma^2_u+\sigma^2_\epsilon&\ldots&0&0&0\\
0&0&0&0&0&0&\dots&\sigma^2_u+\sigma^2_\epsilon&\sigma^2_u&\sigma^2_u\\
0&0&0&0&0&0&\dots&\sigma^2_u&\sigma^2_u+\sigma^2_\epsilon&\sigma^2_u\\
0&0&0&0&0&0&\dots&\sigma^2_u&\sigma^2_u&\sigma^2_u+\sigma^2_\epsilon\\
\end{array}\right]
$$


# Session Info

With respect to reproducibility, it is highly recommended to include a session info in your script so that readers of your output can see your particular setup of R. 

```{r}
sessionInfo()
```


# References
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("francisellaMixedModel.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
