<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Statistical Methods for Quantitative MS-based Proteomics: Blocking - Wrap-up</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.1 MS Basics</a>
    </li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.2 Identification</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1 Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2 Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1 Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Statistical Methods for Quantitative
MS-based Proteomics: Blocking - Wrap-up</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date"><a href="https://statomics.github.io">statOmics</a>,
Ghent University</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a
href="https://statomics.github.io/PDA22GTPB/">Proteomics Data Analysis
(PDA)</a></p>
<div id="import-data-and-preprocessing" class="section level1"
number="1">
<h1><span class="header-section-number">1</span> Import Data and
Preprocessing</h1>
<div id="data" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Data</h2>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>peptidesFile <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/mouseTcell/peptidesRCB.txt&quot;</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>peptidesFile2 <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/mouseTcell/peptidesCRD.txt&quot;</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>peptidesFile3 <span class="ot">&lt;-</span> <span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/mouseTcell/peptides.txt&quot;</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>ecols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">names</span>(<span class="fu">read.delim</span>(peptidesFile)))</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>  <span class="at">assayData =</span> <span class="fu">read.delim</span>(peptidesFile),</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>  <span class="at">quantCols =</span> ecols,</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a>ecols2 <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">names</span>(<span class="fu">read.delim</span>(peptidesFile2)))</span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a>  <span class="at">assayData =</span> <span class="fu">read.delim</span>(peptidesFile2),</span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a>  <span class="at">quantCols =</span> ecols2,</span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a>ecols3 <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">names</span>(<span class="fu">read.delim</span>(peptidesFile3)))</span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a>  <span class="at">assayData =</span> <span class="fu">read.delim</span>(peptidesFile3),</span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a>  <span class="at">quantCols =</span> ecols3,</span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a><span class="do">### Design</span></span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a>  <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a>  <span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-38"><a href="#cb1-38" tabindex="-1"></a>  unlist <span class="sc">%&gt;%</span>  </span>
<span id="cb1-39"><a href="#cb1-39" tabindex="-1"></a>  as.factor</span>
<span id="cb1-40"><a href="#cb1-40" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>mouse <span class="ot">&lt;-</span> pe[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb1-42"><a href="#cb1-42" tabindex="-1"></a>  colnames <span class="sc">%&gt;%</span></span>
<span id="cb1-43"><a href="#cb1-43" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="at">split=</span><span class="st">&quot;[.]&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb1-44"><a href="#cb1-44" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb1-45"><a href="#cb1-45" tabindex="-1"></a>  as.factor</span>
<span id="cb1-46"><a href="#cb1-46" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" tabindex="-1"></a><span class="fu">colData</span>(pe2)<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb1-48"><a href="#cb1-48" tabindex="-1"></a>  <span class="fu">colnames</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb1-49"><a href="#cb1-49" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb1-50"><a href="#cb1-50" tabindex="-1"></a>  <span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-51"><a href="#cb1-51" tabindex="-1"></a>  unlist <span class="sc">%&gt;%</span>  </span>
<span id="cb1-52"><a href="#cb1-52" tabindex="-1"></a>  as.factor</span>
<span id="cb1-53"><a href="#cb1-53" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" tabindex="-1"></a><span class="fu">colData</span>(pe2)<span class="sc">$</span>mouse <span class="ot">&lt;-</span> pe2[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb1-55"><a href="#cb1-55" tabindex="-1"></a>  colnames <span class="sc">%&gt;%</span></span>
<span id="cb1-56"><a href="#cb1-56" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="at">split=</span><span class="st">&quot;[.]&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb1-57"><a href="#cb1-57" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb1-58"><a href="#cb1-58" tabindex="-1"></a>  as.factor</span>
<span id="cb1-59"><a href="#cb1-59" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" tabindex="-1"></a><span class="fu">colData</span>(pe3)<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb1-61"><a href="#cb1-61" tabindex="-1"></a>  <span class="fu">colnames</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb1-62"><a href="#cb1-62" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb1-63"><a href="#cb1-63" tabindex="-1"></a>  <span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-64"><a href="#cb1-64" tabindex="-1"></a>  unlist <span class="sc">%&gt;%</span>  </span>
<span id="cb1-65"><a href="#cb1-65" tabindex="-1"></a>  as.factor</span>
<span id="cb1-66"><a href="#cb1-66" tabindex="-1"></a></span>
<span id="cb1-67"><a href="#cb1-67" tabindex="-1"></a><span class="fu">colData</span>(pe3)<span class="sc">$</span>mouse <span class="ot">&lt;-</span> pe3[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span></span>
<span id="cb1-68"><a href="#cb1-68" tabindex="-1"></a>  colnames <span class="sc">%&gt;%</span></span>
<span id="cb1-69"><a href="#cb1-69" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="at">split=</span><span class="st">&quot;[.]&quot;</span>)  <span class="sc">%&gt;%</span></span>
<span id="cb1-70"><a href="#cb1-70" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb1-71"><a href="#cb1-71" tabindex="-1"></a>  as.factor</span></code></pre></div>
</p>
<p></design></p>
</div>
<div id="preprocessing" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Preprocessing</h2>
<div id="log-transform" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Log-transform</h3>
<details>
<summary>
Click to see code to log-transfrom the data
</summary>
<p>
<ul>
<li>We calculate how many non zero intensities we have for each peptide
and this can be useful for filtering.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZero <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="fu">rowData</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZero <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="fu">rowData</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]])<span class="sc">$</span>nNonZero <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]]) <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code></pre></div>
<ul>
<li>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe2, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe3, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe2, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe3, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="filtering" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Filtering</h3>
<details>
<summary>
Click to see details on filtering
</summary>
<p>
<ol style="list-style-type: decimal">
<li>Handling overlapping protein groups</li>
</ol>
<p>In our approach a peptide can map to multiple proteins, as long as
there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> Proteins <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins))</span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe2, <span class="sc">~</span> Proteins <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(<span class="fu">rowData</span>(pe2[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins))</span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe3, <span class="sc">~</span> Proteins <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(<span class="fu">rowData</span>(pe3[[<span class="st">&quot;peptideLog&quot;</span>]])<span class="sc">$</span>Proteins))</span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants, peptides that map to decoy sequences,
and proteins which were only identified by peptides with
modifications.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe2,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe2,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe3,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe3,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in one sample</li>
</ol>
<p>We keep peptides that were observed at last twice.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> nNonZero <span class="sc">&gt;=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## &#39;nNonZero&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe2,<span class="sc">~</span> nNonZero <span class="sc">&gt;=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## &#39;nNonZero&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe3,<span class="sc">~</span> nNonZero <span class="sc">&gt;=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## &#39;nNonZero&#39; found in 2 out of 2 assay(s).</code></pre>
</p>
</details>
</div>
</div>
<div id="normalization" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Normalization</h2>
<details>
<summary>
Click to see code to normalize the data
</summary>
<p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe2, </span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe3, </span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="summarization" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Summarization</h2>
<details>
<summary>
Click to see code to summarize the data
</summary>
<p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<pre><code>## Aggregated: 1/1</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe2,</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.
## Aggregated: 1/1</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe3,</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.
## Aggregated: 1/1</code></pre>
</p>
</details>
</div>
<div id="data-exploration-what-is-impact-of-blocking"
class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Data Exploration:
what is impact of blocking?</h2>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(pe3)<span class="sc">$</span>mouse) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>mdsObj3 <span class="ot">&lt;-</span> <span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe3[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>mdsOrig <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe3) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mds1 =</span> mdsObj3<span class="sc">$</span>x,</span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>         <span class="at">mds2 =</span> mdsObj3<span class="sc">$</span>y,</span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>         <span class="at">lab =</span> <span class="fu">paste</span>(mouse,celltype,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mds1, <span class="at">y =</span> mds2, <span class="at">label =</span> lab, <span class="at">color =</span> celltype, <span class="at">group =</span> mouse)) <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb37-11"><a href="#cb37-11" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-12"><a href="#cb37-12" tabindex="-1"></a>  <span class="fu">xlab</span>(</span>
<span id="cb37-13"><a href="#cb37-13" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb37-14"><a href="#cb37-14" tabindex="-1"></a>      mdsObj3<span class="sc">$</span>axislabel,</span>
<span id="cb37-15"><a href="#cb37-15" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb37-16"><a href="#cb37-16" tabindex="-1"></a>      <span class="dv">1</span>, </span>
<span id="cb37-17"><a href="#cb37-17" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb37-18"><a href="#cb37-18" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb37-19"><a href="#cb37-19" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj3<span class="sc">$</span>var.explained[<span class="dv">1</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb37-20"><a href="#cb37-20" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb37-21"><a href="#cb37-21" tabindex="-1"></a>        ),</span>
<span id="cb37-22"><a href="#cb37-22" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb37-23"><a href="#cb37-23" tabindex="-1"></a>      )</span>
<span id="cb37-24"><a href="#cb37-24" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb37-25"><a href="#cb37-25" tabindex="-1"></a>  <span class="fu">ylab</span>(</span>
<span id="cb37-26"><a href="#cb37-26" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb37-27"><a href="#cb37-27" tabindex="-1"></a>      mdsObj3<span class="sc">$</span>axislabel,</span>
<span id="cb37-28"><a href="#cb37-28" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb37-29"><a href="#cb37-29" tabindex="-1"></a>      <span class="dv">2</span>, </span>
<span id="cb37-30"><a href="#cb37-30" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb37-31"><a href="#cb37-31" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb37-32"><a href="#cb37-32" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj3<span class="sc">$</span>var.explained[<span class="dv">2</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb37-33"><a href="#cb37-33" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb37-34"><a href="#cb37-34" tabindex="-1"></a>        ),</span>
<span id="cb37-35"><a href="#cb37-35" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb37-36"><a href="#cb37-36" tabindex="-1"></a>      )</span>
<span id="cb37-37"><a href="#cb37-37" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb37-38"><a href="#cb37-38" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Original (RCB)&quot;</span>)</span>
<span id="cb37-39"><a href="#cb37-39" tabindex="-1"></a></span>
<span id="cb37-40"><a href="#cb37-40" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>mouse) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb37-41"><a href="#cb37-41" tabindex="-1"></a>mdsObj <span class="ot">&lt;-</span> <span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb37-42"><a href="#cb37-42" tabindex="-1"></a>mdsRCB <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe) <span class="sc">%&gt;%</span></span>
<span id="cb37-43"><a href="#cb37-43" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb37-44"><a href="#cb37-44" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mds1 =</span> mdsObj<span class="sc">$</span>x,</span>
<span id="cb37-45"><a href="#cb37-45" tabindex="-1"></a>         <span class="at">mds2 =</span> mdsObj<span class="sc">$</span>y,</span>
<span id="cb37-46"><a href="#cb37-46" tabindex="-1"></a>         <span class="at">lab =</span> <span class="fu">paste</span>(mouse,celltype,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb37-47"><a href="#cb37-47" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mds1, <span class="at">y =</span> mds2, <span class="at">label =</span> lab, <span class="at">color =</span> celltype, <span class="at">group =</span> mouse)) <span class="sc">+</span></span>
<span id="cb37-48"><a href="#cb37-48" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb37-49"><a href="#cb37-49" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb37-50"><a href="#cb37-50" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-51"><a href="#cb37-51" tabindex="-1"></a>  <span class="fu">xlab</span>(</span>
<span id="cb37-52"><a href="#cb37-52" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb37-53"><a href="#cb37-53" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb37-54"><a href="#cb37-54" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb37-55"><a href="#cb37-55" tabindex="-1"></a>      <span class="dv">1</span>, </span>
<span id="cb37-56"><a href="#cb37-56" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb37-57"><a href="#cb37-57" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb37-58"><a href="#cb37-58" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj<span class="sc">$</span>var.explained[<span class="dv">1</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb37-59"><a href="#cb37-59" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb37-60"><a href="#cb37-60" tabindex="-1"></a>        ),</span>
<span id="cb37-61"><a href="#cb37-61" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb37-62"><a href="#cb37-62" tabindex="-1"></a>      )</span>
<span id="cb37-63"><a href="#cb37-63" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb37-64"><a href="#cb37-64" tabindex="-1"></a>  <span class="fu">ylab</span>(</span>
<span id="cb37-65"><a href="#cb37-65" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb37-66"><a href="#cb37-66" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb37-67"><a href="#cb37-67" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb37-68"><a href="#cb37-68" tabindex="-1"></a>      <span class="dv">2</span>, </span>
<span id="cb37-69"><a href="#cb37-69" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb37-70"><a href="#cb37-70" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb37-71"><a href="#cb37-71" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj<span class="sc">$</span>var.explained[<span class="dv">2</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb37-72"><a href="#cb37-72" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb37-73"><a href="#cb37-73" tabindex="-1"></a>        ),</span>
<span id="cb37-74"><a href="#cb37-74" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb37-75"><a href="#cb37-75" tabindex="-1"></a>      )</span>
<span id="cb37-76"><a href="#cb37-76" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb37-77"><a href="#cb37-77" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Randomized Complete Block (RCB)&quot;</span>)</span>
<span id="cb37-78"><a href="#cb37-78" tabindex="-1"></a></span>
<span id="cb37-79"><a href="#cb37-79" tabindex="-1"></a></span>
<span id="cb37-80"><a href="#cb37-80" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(pe2)<span class="sc">$</span>mouse) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)</span>
<span id="cb37-81"><a href="#cb37-81" tabindex="-1"></a>mdsObj2 <span class="ot">&lt;-</span> <span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe2[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb37-82"><a href="#cb37-82" tabindex="-1"></a>mdsCRD <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe2) <span class="sc">%&gt;%</span></span>
<span id="cb37-83"><a href="#cb37-83" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb37-84"><a href="#cb37-84" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mds1 =</span> mdsObj2<span class="sc">$</span>x,</span>
<span id="cb37-85"><a href="#cb37-85" tabindex="-1"></a>         <span class="at">mds2 =</span> mdsObj2<span class="sc">$</span>y,</span>
<span id="cb37-86"><a href="#cb37-86" tabindex="-1"></a>         <span class="at">lab =</span> <span class="fu">paste</span>(mouse,celltype,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb37-87"><a href="#cb37-87" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mds1, <span class="at">y =</span> mds2, <span class="at">label =</span> lab, <span class="at">color =</span> celltype, <span class="at">group =</span> mouse)) <span class="sc">+</span></span>
<span id="cb37-88"><a href="#cb37-88" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb37-89"><a href="#cb37-89" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb37-90"><a href="#cb37-90" tabindex="-1"></a>  <span class="fu">xlab</span>(</span>
<span id="cb37-91"><a href="#cb37-91" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb37-92"><a href="#cb37-92" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb37-93"><a href="#cb37-93" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb37-94"><a href="#cb37-94" tabindex="-1"></a>      <span class="dv">1</span>, </span>
<span id="cb37-95"><a href="#cb37-95" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb37-96"><a href="#cb37-96" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb37-97"><a href="#cb37-97" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj2<span class="sc">$</span>var.explained[<span class="dv">1</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb37-98"><a href="#cb37-98" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb37-99"><a href="#cb37-99" tabindex="-1"></a>        ),</span>
<span id="cb37-100"><a href="#cb37-100" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb37-101"><a href="#cb37-101" tabindex="-1"></a>      )</span>
<span id="cb37-102"><a href="#cb37-102" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb37-103"><a href="#cb37-103" tabindex="-1"></a>  <span class="fu">ylab</span>(</span>
<span id="cb37-104"><a href="#cb37-104" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb37-105"><a href="#cb37-105" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb37-106"><a href="#cb37-106" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb37-107"><a href="#cb37-107" tabindex="-1"></a>      <span class="dv">2</span>, </span>
<span id="cb37-108"><a href="#cb37-108" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb37-109"><a href="#cb37-109" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb37-110"><a href="#cb37-110" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj2<span class="sc">$</span>var.explained[<span class="dv">2</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb37-111"><a href="#cb37-111" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb37-112"><a href="#cb37-112" tabindex="-1"></a>        ),</span>
<span id="cb37-113"><a href="#cb37-113" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb37-114"><a href="#cb37-114" tabindex="-1"></a>      )</span>
<span id="cb37-115"><a href="#cb37-115" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb37-116"><a href="#cb37-116" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Completely Randomized Design (CRD)&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a>mdsOrig</span></code></pre></div>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a>mdsRCB</span></code></pre></div>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-11-2.png" width="672" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>mdsCRD</span></code></pre></div>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-11-3.png" width="672" /></p>
<ul>
<li>We observe that the leading fold change is according to mouse</li>
<li>In the second dimension we see a separation according to
cell-type</li>
<li>With the Randomized Complete Block design (RCB) we can remove the
mouse effect from the analysis!</li>
</ul>
</div>
<div id="modeling-and-inference" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Modeling and
inference</h2>
<div id="rcb-analysis" class="section level3" number="1.6.1">
<h3><span class="header-section-number">1.6.1</span> RCB analysis</h3>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>  <span class="at">object =</span> pe,</span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> celltype <span class="sc">+</span> mouse)</span></code></pre></div>
</div>
<div id="rcb-wrong-analysis" class="section level3" number="1.6.2">
<h3><span class="header-section-number">1.6.2</span> RCB wrong
analysis</h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>  <span class="at">object =</span> pe,</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> celltype, <span class="at">modelColumnName =</span> <span class="st">&quot;wrongModel&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="crd-analysis" class="section level2" number="1.7">
<h2><span class="header-section-number">1.7</span> CRD analysis</h2>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>  <span class="at">object =</span> pe2,</span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> celltype)</span></code></pre></div>
<div id="inference" class="section level3" number="1.7.1">
<h3><span class="header-section-number">1.7.1</span> Inference</h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix)</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span> celltype <span class="sc">+</span> mouse)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe2),<span class="sc">~</span> celltype)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-15-2.png" width="672" /></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;celltypeTreg = 0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;celltypeTreg&quot;</span>))</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L)</span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L, <span class="at">modelColumn =</span> <span class="st">&quot;wrongModel&quot;</span>, <span class="at">resultsColumnNamePrefix=</span><span class="st">&quot;wrong&quot;</span>)</span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe2, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
</div>
</div>
</div>
<div id="advantage-of-blocking-comparison-between-designs"
class="section level1" number="2">
<h1><span class="header-section-number">2</span> Advantage of Blocking:
comparison between designs</h1>
<div id="volcano-plots" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Volcano plots</h2>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a>volcanoRCB <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a>    <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>celltypeTreg,</span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;RCB: </span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>                <span class="fu">sum</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>celltypeTreg<span class="sc">$</span>adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a>            <span class="st">&quot; significant&quot;</span>))</span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a></span>
<span id="cb49-12"><a href="#cb49-12" tabindex="-1"></a>volcanoRCBwrong <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb49-13"><a href="#cb49-13" tabindex="-1"></a>    <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>wrongcelltypeTreg,</span>
<span id="cb49-14"><a href="#cb49-14" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb49-15"><a href="#cb49-15" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb49-16"><a href="#cb49-16" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb49-17"><a href="#cb49-17" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb49-18"><a href="#cb49-18" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb49-19"><a href="#cb49-19" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;RCB wrong: </span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb49-20"><a href="#cb49-20" tabindex="-1"></a>                <span class="fu">sum</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>wrongcelltypeTreg<span class="sc">$</span>adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb49-21"><a href="#cb49-21" tabindex="-1"></a>            <span class="st">&quot; significant&quot;</span>))</span>
<span id="cb49-22"><a href="#cb49-22" tabindex="-1"></a></span>
<span id="cb49-23"><a href="#cb49-23" tabindex="-1"></a>volcanoCRD <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb49-24"><a href="#cb49-24" tabindex="-1"></a>    <span class="fu">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>celltypeTreg,</span>
<span id="cb49-25"><a href="#cb49-25" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb49-26"><a href="#cb49-26" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb49-27"><a href="#cb49-27" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb49-28"><a href="#cb49-28" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb49-29"><a href="#cb49-29" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb49-30"><a href="#cb49-30" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;CRD: </span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb49-31"><a href="#cb49-31" tabindex="-1"></a>                <span class="fu">sum</span>(<span class="fu">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>celltypeTreg<span class="sc">$</span>adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb49-32"><a href="#cb49-32" tabindex="-1"></a>            <span class="st">&quot; significant&quot;</span>))</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="fu">grid.arrange</span>(volcanoRCB,volcanoCRD, volcanoRCBwrong,<span class="at">ncol=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 666 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<pre><code>## Warning: Removed 382 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<pre><code>## Warning: Removed 262 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div id="anova-table-q7tpr4-alpha-actinin-1" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> Anova table: Q7TPR4,
Alpha-actinin-1</h2>
<p>Disclaimer: the Anova analysis is only for didactical purposes. In
practice we assess the hypotheses using msqrob2.</p>
<ul>
<li><p>We illustrate the power gain of blocking using an Anova analysis
on 1 protein.</p></li>
<li><p>Note, that msqrob2 will perform a similar analysis, but, it uses
robust regression and it uses an empirical Bayes estimator for the
variance.</p></li>
</ul>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a>prot <span class="ot">&lt;-</span> <span class="st">&quot;Q7TPR4&quot;</span></span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a>dataHlp <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe) <span class="sc">%&gt;%</span> </span>
<span id="cb54-3"><a href="#cb54-3" tabindex="-1"></a>  as.data.frame <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="#cb54-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">intensity=</span><span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]])[prot,],</span>
<span id="cb54-5"><a href="#cb54-5" tabindex="-1"></a>         <span class="at">intensityCRD=</span><span class="fu">assay</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])[prot,])</span>
<span id="cb54-6"><a href="#cb54-6" tabindex="-1"></a></span>
<span id="cb54-7"><a href="#cb54-7" tabindex="-1"></a>  <span class="fu">anova</span>(<span class="fu">lm</span>(intensity<span class="sc">~</span> celltype <span class="sc">+</span> mouse, dataHlp)) </span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["int"],"align":["right"]},{"label":["Sum Sq"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mean Sq"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"5.33887147","3":"5.338871472","4":"785.65428","5":"9.968667e-05","_rn_":"celltype"},{"1":"3","2":"0.43342769","3":"0.144475896","4":"21.26069","5":"1.594409e-02","_rn_":"mouse"},{"1":"3","2":"0.02038634","3":"0.006795446","4":"NA","5":"NA","_rn_":"Residuals"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a>  <span class="fu">anova</span>(<span class="fu">lm</span>(intensity<span class="sc">~</span> celltype,dataHlp))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["int"],"align":["right"]},{"label":["Sum Sq"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mean Sq"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"5.338871","3":"5.33887147","4":"70.58669","5":"0.0001548941","_rn_":"celltype"},{"1":"6","2":"0.453814","3":"0.07563567","4":"NA","5":"NA","_rn_":"Residuals"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a>  <span class="fu">anova</span>(<span class="fu">lm</span>(intensityCRD<span class="sc">~</span> celltype,dataHlp))</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Df"],"name":[1],"type":["int"],"align":["right"]},{"label":["Sum Sq"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Mean Sq"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["F value"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["Pr(>F)"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"1","2":"5.5183267","3":"5.5183267","4":"59.28184","5":"0.0002514749","_rn_":"celltype"},{"1":"6","2":"0.5585178","3":"0.0930863","4":"NA","5":"NA","_rn_":"Residuals"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="comparison-empirical-bayes-standard-deviation-in-msqrob2"
class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Comparison Empirical
Bayes standard deviation in msqrob2</h2>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a>accessions <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pe[[<span class="st">&quot;protein&quot;</span>]])[<span class="fu">rownames</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">%in%</span><span class="fu">rownames</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])]</span>
<span id="cb57-2"><a href="#cb57-2" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb57-3"><a href="#cb57-3" tabindex="-1"></a><span class="at">sigmaRBC =</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels[accessions], getSigmaPosterior),</span>
<span id="cb57-4"><a href="#cb57-4" tabindex="-1"></a><span class="at">sigmaRBCwrong =</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>wrongModel[accessions], getSigmaPosterior),</span>
<span id="cb57-5"><a href="#cb57-5" tabindex="-1"></a>sigmaCRD <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels[accessions], getSigmaPosterior)</span>
<span id="cb57-6"><a href="#cb57-6" tabindex="-1"></a>)</span>
<span id="cb57-7"><a href="#cb57-7" tabindex="-1"></a></span>
<span id="cb57-8"><a href="#cb57-8" tabindex="-1"></a> plotRBCvsWrong <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat, <span class="fu">aes</span>(sigmaRBC, sigmaRBCwrong)) <span class="sc">+</span></span>
<span id="cb57-9"><a href="#cb57-9" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">shape =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb57-10"><a href="#cb57-10" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb57-11"><a href="#cb57-11" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb57-12"><a href="#cb57-12" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>,<span class="at">slope=</span><span class="dv">1</span>)</span>
<span id="cb57-13"><a href="#cb57-13" tabindex="-1"></a>plotCRDvsWrong <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat, <span class="fu">aes</span>(sigmaCRD, sigmaRBCwrong)) <span class="sc">+</span></span>
<span id="cb57-14"><a href="#cb57-14" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">shape =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb57-15"><a href="#cb57-15" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb57-16"><a href="#cb57-16" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb57-17"><a href="#cb57-17" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>,<span class="at">slope=</span><span class="dv">1</span>)</span>
<span id="cb57-18"><a href="#cb57-18" tabindex="-1"></a>plotRBCvsCRD <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat, <span class="fu">aes</span>(sigmaRBC, sigmaCRD)) <span class="sc">+</span></span>
<span id="cb57-19"><a href="#cb57-19" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">shape =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb57-20"><a href="#cb57-20" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb57-21"><a href="#cb57-21" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb57-22"><a href="#cb57-22" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>,<span class="at">slope=</span><span class="dv">1</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" tabindex="-1"></a><span class="fu">grid.arrange</span>(</span>
<span id="cb58-2"><a href="#cb58-2" tabindex="-1"></a>  plotRBCvsWrong,</span>
<span id="cb58-3"><a href="#cb58-3" tabindex="-1"></a>  plotCRDvsWrong,</span>
<span id="cb58-4"><a href="#cb58-4" tabindex="-1"></a>  plotRBCvsCRD,</span>
<span id="cb58-5"><a href="#cb58-5" tabindex="-1"></a>  <span class="at">nrow=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 619 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<pre><code>## Warning: Removed 387 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<pre><code>## Warning: Removed 650 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_blocking_wrapup_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<ul>
<li><p>We clearly observe that the standard deviation of the protein
expression in the RCB is smaller for the majority of the proteins than
that obtained with the CRD</p></li>
<li><p>The standard deviation of the protein expression RCB where we
perform a wrong analysis without considering the blocking factor
according to mouse is much larger for the marjority of the proteins than
that obtained with the correct analysis.</p></li>
<li><p>Indeed, when we ignore the blocking factor in the RCB design we
do not remove the variability according to mouse from the analysis and
the mouse effect is absorbed in the error term. The standard deviation
than becomes very comparable to that observed in the completely
randomised design where we could not remove the mouse effect from the
analysis.</p></li>
<li><p>Why are some of the standard deviations for the RCB with the
correct analysis larger than than of the RCB with the incorrect analysis
that ignored the mouse blocking factor?</p></li>
<li><p>Can you think of a reason why it would not be useful to block on
a particular factor?</p></li>
</ul>
</div>
</div>

<div id="rmd-source-code">---
title: "Statistical Methods for Quantitative MS-based Proteomics: Blocking - Wrap-up"
author: "Lieven Clement"
date: "[statOmics](https://statomics.github.io), Ghent University"
output:
    html_document:
      code_download: true
      theme: flatly
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue

bibliography: msqrob2.bib

---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

This is part of the online course [Proteomics Data Analysis (PDA)](https://statomics.github.io/PDA22GTPB/)

# Import Data and Preprocessing 

## Data 
<details><summary> Click to see code  </summary><p>
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)
library(gridExtra)

peptidesFile <- "https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/mouseTcell/peptidesRCB.txt"
peptidesFile2 <- "https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/mouseTcell/peptidesCRD.txt"
peptidesFile3 <- "https://raw.githubusercontent.com/statOmics/PDA22GTPB/data/quantification/mouseTcell/peptides.txt"

ecols <- grep("Intensity\\.", names(read.delim(peptidesFile)))
pe <- readQFeatures(
  assayData = read.delim(peptidesFile),
  fnames = 1,
  quantCols = ecols,
  name = "peptideRaw")

ecols2 <- grep("Intensity\\.", names(read.delim(peptidesFile2)))
pe2 <- readQFeatures(
  assayData = read.delim(peptidesFile2),
  fnames = 1,
  quantCols = ecols2,
  name = "peptideRaw")

ecols3 <- grep("Intensity\\.", names(read.delim(peptidesFile3)))
pe3 <- readQFeatures(
  assayData = read.delim(peptidesFile3),
  fnames = 1,
  quantCols = ecols3,
  name = "peptideRaw")

### Design
colData(pe)$celltype <- substr(
  colnames(pe[["peptideRaw"]]),
  11,
  14) %>%
  unlist %>%  
  as.factor

colData(pe)$mouse <- pe[[1]] %>%
  colnames %>%
  strsplit(split="[.]")  %>%
  sapply(function(x) x[3]) %>%
  as.factor

colData(pe2)$celltype <- substr(
  colnames(pe2[["peptideRaw"]]),
  11,
  14) %>%
  unlist %>%  
  as.factor

colData(pe2)$mouse <- pe2[[1]] %>%
  colnames %>%
  strsplit(split="[.]")  %>%
  sapply(function(x) x[3]) %>%
  as.factor

colData(pe3)$celltype <- substr(
  colnames(pe3[["peptideRaw"]]),
  11,
  14) %>%
  unlist %>%  
  as.factor

colData(pe3)$mouse <- pe3[[1]] %>%
  colnames %>%
  strsplit(split="[.]")  %>%
  sapply(function(x) x[3]) %>%
  as.factor
```
</p></design>

## Preprocessing 


### Log-transform

<details><summary> Click to see code to log-transfrom the data </summary><p>
- We calculate how many non zero intensities we have for each peptide and this can be useful for filtering.

```{r}
rowData(pe[["peptideRaw"]])$nNonZero <- rowSums(assay(pe[["peptideRaw"]]) > 0)

rowData(pe2[["peptideRaw"]])$nNonZero <- rowSums(assay(pe2[["peptideRaw"]]) > 0)

rowData(pe3[["peptideRaw"]])$nNonZero <- rowSums(assay(pe3[["peptideRaw"]]) > 0)
```

- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA

pe2 <- zeroIsNA(pe2, "peptideRaw") # convert 0 to NA

pe3 <- zeroIsNA(pe3, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")

pe2 <- logTransform(pe2, base = 2, i = "peptideRaw", name = "peptideLog")

pe3 <- logTransform(pe3, base = 2, i = "peptideRaw", name = "peptideLog")
```
</p></details>


### Filtering
<details><summary> Click to see details on filtering </summary><p>

1. Handling overlapping protein groups

In our approach a peptide can map to multiple proteins, as long as there is
none of these proteins present in a smaller subgroup.

```{r}
pe <- filterFeatures(pe, ~ Proteins %in% smallestUniqueGroups(rowData(pe[["peptideLog"]])$Proteins))

pe2 <- filterFeatures(pe2, ~ Proteins %in% smallestUniqueGroups(rowData(pe2[["peptideLog"]])$Proteins))

pe3 <- filterFeatures(pe3, ~ Proteins %in% smallestUniqueGroups(rowData(pe3[["peptideLog"]])$Proteins))
```
2. Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Potential.contaminant != "+")

pe2 <- filterFeatures(pe2,~Reverse != "+")
pe2 <- filterFeatures(pe2,~ Potential.contaminant != "+")

pe3 <- filterFeatures(pe3,~Reverse != "+")
pe3 <- filterFeatures(pe3,~ Potential.contaminant != "+")
```
3. Drop peptides that were only identified in one sample

We keep peptides that were observed at last twice.

```{r}
pe <- filterFeatures(pe,~ nNonZero >=2)
pe2 <- filterFeatures(pe2,~ nNonZero >=2)
pe3 <- filterFeatures(pe3,~ nNonZero >=2)
```

</p></details>

## Normalization 

<details><summary> Click to see code to normalize the data </summary><p>
```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")

pe2 <- normalize(pe2, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")


pe3 <- normalize(pe3, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

</p></details>

## Summarization

<details><summary> Click to see code to summarize the data </summary><p>

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")


pe2 <- aggregateFeatures(pe2,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")

pe3 <- aggregateFeatures(pe3,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")
```

</p></details>

## Data Exploration: what is impact of blocking? 

<details><summary> Click to see code </summary><p>
```{r}
levels(colData(pe3)$mouse) <- paste0("m",1:7)
mdsObj3 <- plotMDS(assay(pe3[["protein"]]), plot = FALSE)
mdsOrig <- colData(pe3) %>%
  as.data.frame %>%
  mutate(mds1 = mdsObj3$x,
         mds2 = mdsObj3$y,
         lab = paste(mouse,celltype,sep="_")) %>%
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  geom_line(color = "black", linetype = "dashed") +
  xlab(
    paste0(
      mdsObj3$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj3$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj3$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj3$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Original (RCB)")

levels(colData(pe)$mouse) <- paste0("m",1:4)
mdsObj <- plotMDS(assay(pe[["protein"]]), plot = FALSE)
mdsRCB <- colData(pe) %>%
  as.data.frame %>%
  mutate(mds1 = mdsObj$x,
         mds2 = mdsObj$y,
         lab = paste(mouse,celltype,sep="_")) %>%
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  geom_line(color = "black", linetype = "dashed") +
  xlab(
    paste0(
      mdsObj$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Randomized Complete Block (RCB)")


levels(colData(pe2)$mouse) <- paste0("m",1:8)
mdsObj2 <- plotMDS(assay(pe2[["protein"]]), plot = FALSE)
mdsCRD <- colData(pe2) %>%
  as.data.frame %>%
  mutate(mds1 = mdsObj2$x,
         mds2 = mdsObj2$y,
         lab = paste(mouse,celltype,sep="_")) %>%
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  xlab(
    paste0(
      mdsObj$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj2$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj2$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Completely Randomized Design (CRD)")
```
</p></details>
```{r}
mdsOrig
mdsRCB
mdsCRD
```

- We observe that the leading fold change is according to mouse
- In the second dimension we see a separation according to cell-type 
- With the Randomized Complete Block design (RCB) we can remove the mouse effect from the analysis!

## Modeling and inference

### RCB analysis
```{r warning=FALSE}
pe <- msqrob(
  object = pe,
  i = "protein",
  formula = ~ celltype + mouse)
```

### RCB wrong analysis
```{r warning=FALSE}
pe <- msqrob(
  object = pe,
  i = "protein",
  formula = ~ celltype, modelColumnName = "wrongModel")
```

## CRD analysis 
```{r warning = FALSE}
pe2 <- msqrob(
  object = pe2,
  i = "protein",
  formula = ~ celltype)
```

### Inference 

```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~ celltype + mouse)$plotlist
VisualizeDesign(colData(pe2),~ celltype)$plotlist
```


```{r}
L <- makeContrast("celltypeTreg = 0", parameterNames = c("celltypeTreg"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
pe <- hypothesisTest(object = pe, i = "protein", contrast = L, modelColumn = "wrongModel", resultsColumnNamePrefix="wrong")
pe2 <- hypothesisTest(object = pe2, i = "protein", contrast = L)
```

# Advantage of Blocking: comparison between designs

## Volcano plots

<details><summary> Click to see code </summary><p>
```{r}
volcanoRCB <- ggplot(
    rowData(pe[["protein"]])$celltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("RCB: \n", 
                sum(rowData(pe[["protein"]])$celltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))

volcanoRCBwrong <- ggplot(
    rowData(pe[["protein"]])$wrongcelltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("RCB wrong: \n", 
                sum(rowData(pe[["protein"]])$wrongcelltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))

volcanoCRD <- ggplot(
    rowData(pe2[["protein"]])$celltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("CRD: \n", 
                sum(rowData(pe2[["protein"]])$celltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))
```
</p></details>
  
```{r}
grid.arrange(volcanoRCB,volcanoCRD, volcanoRCBwrong,ncol=2)
```

## Anova table: Q7TPR4, Alpha-actinin-1

Disclaimer: the Anova analysis is only for didactical purposes. In practice we assess the hypotheses using msqrob2. 

- We illustrate the power gain of blocking using an Anova analysis on 1 protein. 

- Note, that msqrob2 will perform a similar analysis, but, it uses robust regression and it uses an empirical Bayes estimator for the variance.

```{r}
prot <- "Q7TPR4"
dataHlp <- colData(pe) %>% 
  as.data.frame %>%
  mutate(intensity=assay(pe[["protein"]])[prot,],
         intensityCRD=assay(pe2[["protein"]])[prot,])

  anova(lm(intensity~ celltype + mouse, dataHlp)) 
  anova(lm(intensity~ celltype,dataHlp))
  anova(lm(intensityCRD~ celltype,dataHlp))
```

## Comparison Empirical Bayes standard deviation in msqrob2 

<details><summary> Click to see code </summary><p>
```{r}
accessions <- rownames(pe[["protein"]])[rownames(pe[["protein"]])%in%rownames(pe2[["protein"]])]
dat <- data.frame(
sigmaRBC = sapply(rowData(pe[["protein"]])$msqrobModels[accessions], getSigmaPosterior),
sigmaRBCwrong = sapply(rowData(pe[["protein"]])$wrongModel[accessions], getSigmaPosterior),
sigmaCRD <- sapply(rowData(pe2[["protein"]])$msqrobModels[accessions], getSigmaPosterior)
)

 plotRBCvsWrong <- ggplot(data = dat, aes(sigmaRBC, sigmaRBCwrong)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
plotCRDvsWrong <- ggplot(data = dat, aes(sigmaCRD, sigmaRBCwrong)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
plotRBCvsCRD <- ggplot(data = dat, aes(sigmaRBC, sigmaCRD)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
```
</p></details>

```{r}
grid.arrange(
  plotRBCvsWrong,
  plotCRDvsWrong,
  plotRBCvsCRD,
  nrow=2)
```

- We clearly observe that the standard deviation of the protein expression in the RCB is smaller for the majority of the proteins than that obtained with the CRD

- The standard deviation of the protein expression RCB where we perform a wrong analysis without considering the blocking factor according to mouse is much larger for the marjority of the proteins than that obtained with the correct analysis. 

- Indeed, when we ignore the blocking factor in the RCB design we do not remove the variability according to mouse from the analysis and the mouse effect is absorbed in the error term. The standard deviation than becomes very comparable to that observed in the completely randomised design where we could not remove the mouse effect from the analysis. 

- Why are some of the standard deviations for the RCB with the correct analysis larger than than of the RCB with the incorrect analysis that ignored the mouse blocking factor?

- Can you think of a reason why it would not be useful to block on a particular factor? 
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("pda_blocking_wrapup.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
