<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement &amp; Koen Van den Berge" />


<title>Sequencing: Selected technical topics</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Lectures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="intro.html">1.1 Position</a>
    </li>
    <li>
      <a href="recapGeneralLinearModel.html">1.2 Recap linear model</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.1 MS Basics</a>
    </li>
    <li>
      <a href="./figs/part1_ms_basics.pdf">1.2 Identification</a>
    </li>
    <li>
      <a href="TargetDecoy.html">2.1 Identification - FDR - QC</a>
    </li>
    <li>
      <a href="pda_quantification_preprocessing.html">2.2.1 Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">2.2.2 Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">2.3.1 Quantification - Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="pda_blocking_wrapup.html">2.3.2. Wrap-up DA - Blocking</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">---- INTRODUCTION ----</li>
    <li>
      <a href="multipleRegression_KPNA2.html">1 Intro KPNA2</a>
    </li>
    <li class="dropdown-header">---- MODULE I PROTEOMICS ----</li>
    <li>
      <a href="https://www.compomics.com/bioinformatics-for-proteomics/">1. Bioinformatics for Proteomics</a>
    </li>
    <li>
      <a href="TargetDecoyTutorial.html">2.1. Target Decoy QC</a>
    </li>
    <li>
      <a href="pda_tutorialPreprocessing.html">2.2. Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">2.3. Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Sequencing: Selected technical topics</h1>
<h4 class="author">Lieven Clement &amp; Koen Van den Berge</h4>
<h4 class="date">Last edited on 05 November, 2025</h4>

</div>


<div
id="parameter-estimation-and-inference-in-generalized-linear-models"
class="section level1" number="1">
<h1><span class="header-section-number">1</span> Parameter Estimation
and Inference in Generalized linear models</h1>
<div id="simulate-poisson-data" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Simulate Poisson
Data</h2>
<ul>
<li>We simulate data for 100 observations.</li>
<li>Covariates x are simulated from normal distribution</li>
<li>The <span class="math inline">\(\beta\)</span> are chosen at <span
class="math inline">\(\beta_0=2\)</span>, <span
class="math inline">\(\beta_1=0.8\)</span>, <span
class="math inline">\(\beta_2=1.2\)</span></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">300</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>xhlp<span class="ot">&lt;-</span><span class="fu">cbind</span>(<span class="dv">1</span>,<span class="fu">rnorm</span>(<span class="dv">100</span>),<span class="fu">rnorm</span>(<span class="dv">100</span>))</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>betasTrue<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="fl">0.8</span>,<span class="fl">1.2</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>etaTrue<span class="ot">&lt;-</span>xhlp<span class="sc">%*%</span>betasTrue</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>y<span class="ot">&lt;-</span><span class="fu">rpois</span>(<span class="dv">100</span>,<span class="fu">exp</span>(etaTrue))</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">coef =</span><span class="dv">0</span><span class="sc">:</span><span class="dv">2</span>,<span class="at">betasTrue=</span>betasTrue) <span class="sc">%&gt;%</span> </span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>coef,<span class="at">y=</span>betasTrue)) <span class="sc">+</span> </span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;parameter value&quot;</span>) <span class="sc">+</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;beta&quot;</span>) <span class="sc">+</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="plotly html-widget html-fill-item" id="htmlwidget-480fa3e06a73f5c42de4" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-480fa3e06a73f5c42de4">{"x":{"visdat":{"a489362044bb":["function () ","plotlyVisDat"]},"cur_data":"a489362044bb","attrs":{"a489362044bb":{"x":{},"y":{},"z":{},"mode":"markers","size":0.5,"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d","inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"aspectmode":"cube","xaxis":{"range":[-2.3424957105876252,2.5745294317880107],"title":"x1"},"yaxis":{"title":"x2"},"zaxis":{"title":"y"}},"yaxis":{"range":[-2.0896111015455219,2.1056439009136327]},"zaxis":{"range":[0,496]},"hovermode":"closest","showlegend":false},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[1.3737908809195094,0.86210687172823852,0.47348910439561598,0.70126281414690683,-0.085055274125850941,1.568702123415022,0.8173919735882893,0.39476860270947117,1.2126985527906875,0.35508065933832028,2.2162742059926384,-0.09054038613482876,-1.3165281108188345,0.066534789021790094,0.51217261794843105,0.050029740995909143,1.4673971806795283,-1.2895205172910114,-0.25769151249919464,0.29305131488208713,-0.33742027524420537,0.63325794588540507,-0.31317859524643482,0.13489312485259145,-1.1447550750922821,-0.31139820998045953,0.23490838742006326,1.3386738166137495,-0.099362220925867828,1.0368993492963112,-0.3455232470647539,0.36289479803801339,-0.80546817274725757,-0.16105157281120572,0.78568505725921145,0.33678422685966014,-0.6797621241984606,0.018678048219562173,-0.20502278988425368,-1.1730775274302627,0.89228159664815843,-0.8211799317287205,-0.21010543321852929,0.39126296753632678,1.3301609843194473,1.3787259492289523,0.97554036432451519,0.66610969595663294,0.45522248121936859,0.25168538999288714,1.0180587408136983,1.6678041857980805,-1.5259593576454331,0.070677460146053653,0.13349706423647448,-0.054982746171481592,-0.71287693284389864,0.085301940023051817,-0.70592842098519293,-0.86302551607147249,-0.061228914782895209,2.5005080753494009,0.65033499678505247,-0.40100817244992781,1.3031543367610237,-0.80739868670466419,1.4529592632023594,-1.2218494802781135,-0.061687461288743882,-0.34623046018732651,-0.031577700937192074,-2.3424957105876252,-0.79534969520878607,-0.72793432146930404,-0.046372881788747038,1.3360597508078036,2.0843301862183536,-0.14673934634900765,-0.50541290556062546,-0.82226985945525843,0.60168269235803784,-0.38096474106048805,-0.1203016323046881,-0.022104041719181559,-1.3633553057537686,0.70334347582582246,0.60318568895455771,-1.0008425814390611,-0.12856218566967104,0.18070804625697853,-0.17666494408157277,1.0653730188555501,0.83114953863099961,-1.2824447706737669,2.5745294317880107,1.4786610032302421,-0.47370913720839569,0.50881297385662894,0.39315021759197727,1.5882712569547759],"y":[1.1831719813830279,0.33215594869814852,-1.4970230980757466,-0.10589397510465613,-0.44454213473616694,0.38237522458763845,0.56396794685742702,-0.045993285138678258,0.018071300813195041,0.73127704118846804,-0.40444925603536569,1.4266006477725905,1.1009521403547411,-0.57506095267543067,1.2515410378803822,0.35922537035482593,-1.7912071199992543,-0.93734439057992791,0.86509203001415791,-0.79285555869134872,0.70306587833317713,-0.010114321310096821,-0.29576001091049736,-0.51354097385034425,-2.0896111015455219,0.21556496049811413,-0.3178444470962955,0.42299048275507811,-1.8115890469776681,2.0608120807864703,-1.6615335396099036,-0.028901886183595953,0.66020084321902883,-0.9756289507831889,-0.23118486517753045,-0.1379223480617644,0.42719909191411204,1.9201828707431676,-1.0861396958995537,0.90641342589605445,0.73113373673963955,1.4915199343245162,0.53701877901546269,0.48483583871872638,0.19441289844731213,0.30564826606121837,0.069130809723756445,-0.066386302015005852,1.2276370130443011,-0.34689383155733711,0.90200909738107871,0.95460500621415934,1.0911873399152492,1.540251054624816,0.79100193444724876,1.2550421354873287,2.1056439009136327,0.47631653070163898,0.86085664807532336,0.99547031906294903,0.58438371265851541,-0.091809183250119816,-0.29349159703352684,-0.96399694498954958,-1.270934818540624,0.41950750781845486,0.57934433973211608,-1.8743365539325871,-0.7316044641806162,-0.48263864877395668,1.4728030439569511,-0.9178944454338297,-1.6001671518501557,-0.67994468074839554,1.745864161657348,1.8733654520252763,0.13503293730943111,-0.68656924450691403,1.2352931283474327,-1.7184028971546526,-0.39469999766938596,-1.5068059534610918,0.29309737109356127,0.1625442984776331,0.61671865101524248,-0.81669214852254701,-0.83873881088451907,0.33924118900133499,0.54325872409221432,0.47390222120824976,1.2363931386521021,-0.81406598906384453,0.038110282211458506,1.003685285757181,1.8495535448814142,0.39837868180483982,-0.77653859209713605,-0.97248382607769046,-0.14595488058437386,1.5966865379161956],"z":[101,23,2,8,3,29,27,9,16,23,21,32,13,7,47,12,1,1,18,2,11,12,1,3,0,6,5,25,1,179,0,8,3,0,5,13,4,65,1,4,40,33,8,23,29,20,22,15,51,6,48,94,11,43,28,32,52,16,16,10,16,47,8,2,4,9,47,0,1,4,47,1,0,1,56,202,35,1,20,0,10,0,18,8,4,3,2,7,9,12,49,6,19,13,496,44,0,1,7,174],"mode":"markers","type":"scatter3d","marker":{"color":"rgba(31,119,180,1)","size":[55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55],"sizemode":"area","line":{"color":"rgba(31,119,180,1)"}},"textfont":{"size":55},"error_y":{"color":"rgba(31,119,180,1)","width":55},"error_x":{"color":"rgba(31,119,180,1)","width":55},"line":{"color":"rgba(31,119,180,1)","width":55},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div class="plotly html-widget html-fill-item" id="htmlwidget-a60541631e7863eb7ae0" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-a60541631e7863eb7ae0">{"x":{"visdat":{"a4897a1ac548":["function () ","plotlyVisDat"]},"cur_data":"a4897a1ac548","attrs":{"a4897a1ac548":{"x":{},"y":{},"z":{},"mode":"markers","size":0.5,"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d","inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"scene":{"aspectmode":"cube","xaxis":{"range":[-2.3424957105876252,2.5745294317880107],"title":"x1"},"yaxis":{"title":"x2"},"zaxis":{"title":"log2(y + 0.5)"}},"yaxis":{"range":[-2.0896111015455219,2.1056439009136327]},"zaxis":{"range":[0,496]},"hovermode":"closest","showlegend":false},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[1.3737908809195094,0.86210687172823852,0.47348910439561598,0.70126281414690683,-0.085055274125850941,1.568702123415022,0.8173919735882893,0.39476860270947117,1.2126985527906875,0.35508065933832028,2.2162742059926384,-0.09054038613482876,-1.3165281108188345,0.066534789021790094,0.51217261794843105,0.050029740995909143,1.4673971806795283,-1.2895205172910114,-0.25769151249919464,0.29305131488208713,-0.33742027524420537,0.63325794588540507,-0.31317859524643482,0.13489312485259145,-1.1447550750922821,-0.31139820998045953,0.23490838742006326,1.3386738166137495,-0.099362220925867828,1.0368993492963112,-0.3455232470647539,0.36289479803801339,-0.80546817274725757,-0.16105157281120572,0.78568505725921145,0.33678422685966014,-0.6797621241984606,0.018678048219562173,-0.20502278988425368,-1.1730775274302627,0.89228159664815843,-0.8211799317287205,-0.21010543321852929,0.39126296753632678,1.3301609843194473,1.3787259492289523,0.97554036432451519,0.66610969595663294,0.45522248121936859,0.25168538999288714,1.0180587408136983,1.6678041857980805,-1.5259593576454331,0.070677460146053653,0.13349706423647448,-0.054982746171481592,-0.71287693284389864,0.085301940023051817,-0.70592842098519293,-0.86302551607147249,-0.061228914782895209,2.5005080753494009,0.65033499678505247,-0.40100817244992781,1.3031543367610237,-0.80739868670466419,1.4529592632023594,-1.2218494802781135,-0.061687461288743882,-0.34623046018732651,-0.031577700937192074,-2.3424957105876252,-0.79534969520878607,-0.72793432146930404,-0.046372881788747038,1.3360597508078036,2.0843301862183536,-0.14673934634900765,-0.50541290556062546,-0.82226985945525843,0.60168269235803784,-0.38096474106048805,-0.1203016323046881,-0.022104041719181559,-1.3633553057537686,0.70334347582582246,0.60318568895455771,-1.0008425814390611,-0.12856218566967104,0.18070804625697853,-0.17666494408157277,1.0653730188555501,0.83114953863099961,-1.2824447706737669,2.5745294317880107,1.4786610032302421,-0.47370913720839569,0.50881297385662894,0.39315021759197727,1.5882712569547759],"y":[1.1831719813830279,0.33215594869814852,-1.4970230980757466,-0.10589397510465613,-0.44454213473616694,0.38237522458763845,0.56396794685742702,-0.045993285138678258,0.018071300813195041,0.73127704118846804,-0.40444925603536569,1.4266006477725905,1.1009521403547411,-0.57506095267543067,1.2515410378803822,0.35922537035482593,-1.7912071199992543,-0.93734439057992791,0.86509203001415791,-0.79285555869134872,0.70306587833317713,-0.010114321310096821,-0.29576001091049736,-0.51354097385034425,-2.0896111015455219,0.21556496049811413,-0.3178444470962955,0.42299048275507811,-1.8115890469776681,2.0608120807864703,-1.6615335396099036,-0.028901886183595953,0.66020084321902883,-0.9756289507831889,-0.23118486517753045,-0.1379223480617644,0.42719909191411204,1.9201828707431676,-1.0861396958995537,0.90641342589605445,0.73113373673963955,1.4915199343245162,0.53701877901546269,0.48483583871872638,0.19441289844731213,0.30564826606121837,0.069130809723756445,-0.066386302015005852,1.2276370130443011,-0.34689383155733711,0.90200909738107871,0.95460500621415934,1.0911873399152492,1.540251054624816,0.79100193444724876,1.2550421354873287,2.1056439009136327,0.47631653070163898,0.86085664807532336,0.99547031906294903,0.58438371265851541,-0.091809183250119816,-0.29349159703352684,-0.96399694498954958,-1.270934818540624,0.41950750781845486,0.57934433973211608,-1.8743365539325871,-0.7316044641806162,-0.48263864877395668,1.4728030439569511,-0.9178944454338297,-1.6001671518501557,-0.67994468074839554,1.745864161657348,1.8733654520252763,0.13503293730943111,-0.68656924450691403,1.2352931283474327,-1.7184028971546526,-0.39469999766938596,-1.5068059534610918,0.29309737109356127,0.1625442984776331,0.61671865101524248,-0.81669214852254701,-0.83873881088451907,0.33924118900133499,0.54325872409221432,0.47390222120824976,1.2363931386521021,-0.81406598906384453,0.038110282211458506,1.003685285757181,1.8495535448814142,0.39837868180483982,-0.77653859209713605,-0.97248382607769046,-0.14595488058437386,1.5966865379161956],"z":[6.6653359171851765,4.5545888516776376,1.3219280948873624,3.0874628412503395,1.8073549220576042,4.8826430493618416,4.7813597135246599,3.2479275134435857,4.0443941193584534,4.5545888516776376,4.4262647547020979,5.0223678130284544,3.7548875021634687,2.9068905956085187,5.5698556083309478,3.6438561897747248,0.58496250072115619,0.58496250072115619,4.2094533656289501,1.3219280948873624,3.5235619560570131,3.6438561897747248,0.58496250072115619,1.8073549220576042,-1,2.7004397181410922,2.4594316186372973,4.6724253419714952,0.58496250072115619,7.4878400338230513,-1,3.0874628412503395,1.8073549220576042,-1,2.4594316186372973,3.7548875021634687,2.1699250014423122,6.0334230015374501,0.58496250072115619,2.1699250014423122,5.3398500028846243,5.0660891904577721,3.0874628412503395,4.5545888516776376,4.8826430493618416,4.3575520046180838,4.4918530963296748,3.9541963103868754,5.6865005271832185,2.7004397181410922,5.5999128421871278,6.5622424242210728,3.5235619560570131,5.4429434958487279,4.8328900141647413,5.0223678130284544,5.7142455176661224,4.0443941193584534,4.0443941193584534,3.3923174227787602,4.0443941193584534,5.5698556083309478,3.0874628412503395,1.3219280948873624,2.1699250014423122,3.2479275134435857,5.5698556083309478,-1,0.58496250072115619,2.1699250014423122,5.5698556083309478,0.58496250072115619,-1,0.58496250072115619,5.8201789624151878,7.6617780977719869,5.1497471195046822,0.58496250072115619,4.3575520046180838,-1,3.3923174227787602,-1,4.2094533656289501,3.0874628412503395,2.1699250014423122,1.8073549220576042,1.3219280948873624,2.9068905956085187,3.2479275134435857,3.6438561897747248,5.6293566200796095,2.7004397181410922,4.2854022188622487,3.7548875021634687,8.9556499075283735,5.4757334309663976,-1,0.58496250072115619,2.9068905956085187,7.447083226209652],"mode":"markers","type":"scatter3d","marker":{"color":"rgba(31,119,180,1)","size":[55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55],"sizemode":"area","line":{"color":"rgba(31,119,180,1)"}},"textfont":{"size":55},"error_y":{"color":"rgba(31,119,180,1)","width":55},"error_x":{"color":"rgba(31,119,180,1)","width":55},"line":{"color":"rgba(31,119,180,1)","width":55},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="exponential-family" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Exponential
Family</h2>
<p><span class="math display">\[
f(y_i\vert \theta_i,\phi)=\exp\left\{ \frac{y_i\theta_i-
b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}
\]</span></p>
<p>with</p>
<ul>
<li><span class="math inline">\(\theta_i\)</span>: canonical
parameters</li>
<li><span class="math inline">\(\phi\)</span>: dispersion parameter</li>
<li><span class="math inline">\(a(.)\)</span>, <span
class="math inline">\(b(.)\)</span>, <span
class="math inline">\(c(.)\)</span>: specific functions that depend on
the distribution,</li>
</ul>
<p>e.g. for normal distribution</p>
<ul>
<li><span class="math inline">\(\phi=\sigma^2\)</span>,</li>
<li><span class="math inline">\(\theta=\mu\)</span>,</li>
<li><span class="math inline">\(a(\phi)=\phi=\sigma^2\)</span>,</li>
<li><span class="math inline">\(b(\theta_i)=\theta_i^2/2\)</span>,</li>
<li><span
class="math inline">\(c(y_i,\phi)=-\frac{1}{2}[y^2/\phi+\log(2\pi\phi)]\)</span></li>
</ul>
<div id="poisson-distribution" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Poisson
Distribution</h3>
<p><span class="math display">\[
y_i \sim \frac{\mu_i^{y_i}e^{-\mu_i}}{y_i!}
\]</span></p>
<p>In format of exponential family:</p>
<p><span class="math display">\[
y_i \sim \exp\left\{y_i\log(\mu_i) - \mu_i - \log(y_i!)\right\}
\]</span></p>
</div>
</div>
<div id="components-of-generalized-linear-model" class="section level2"
number="1.3">
<h2><span class="header-section-number">1.3</span> Components of
Generalized Linear Model</h2>
<p><span class="math display">\[
\left\{\begin{array}{ccc}
y_i\vert x_i&amp;\sim&amp;f(y_i\vert{\theta}_i,\phi)\\\\
\text{E}\left[ y_i\vert \mathbf{x}_i\right]&amp;=&amp;\mu_i\\\\
g(\mu_i)&amp;=&amp;\eta(\mathbf{x}_i)\\\\
\eta(\mathbf{x}_i)&amp;=&amp;\mathbf{x}_i^T\boldsymbol{\beta}
\end{array}\right.,
\]</span> with <span class="math inline">\(g(.)\)</span> the link
function, e.g. </p>
<ul>
<li><span class="math inline">\(g(.)=.\)</span> : identity link for
Normal distribution</li>
<li><span class="math inline">\(g(.)=\log(.)\)</span> : canonical link
for Poisson distribution</li>
<li><span
class="math inline">\(g(.)=\text{logit}(.)=\log\left[\frac{(.)}{(1-.)}\right]\)</span>
: canonical link for Bernouilli distribution.</li>
</ul>
<div id="poisson-glm" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Poisson GLM</h3>
<p><span class="math display">\[\left\{\begin{array}{lcr}
y_i &amp;\sim&amp; Poisson(\mu_i)\\
E[y_i]&amp;=&amp;\mu_i\\
\log(\mu_i)&amp;=&amp;\eta_i\\
\eta_i&amp;=&amp;\mathbf{x}_i\boldsymbol{\beta}
\end{array}\right.\]</span></p>
</div>
</div>
<div id="likelihood" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Likelihood</h2>
<p>We start from a sample, and consider it as fixed and known.</p>
<ul>
<li><p>In particular we do NOT consider the sample observations as
random variables.</p></li>
<li><p>Therefore we write the observed sample as <span
class="math inline">\(y_i, . . . , y_n\)</span></p></li>
<li><p>The theory is based on the likelihood function, which can be
interpreted as a measure for the probability that the given sample is
observed as a realisation of a sequence of random variables <span
class="math inline">\(Y_1, \ldots Y_n\)</span></p></li>
<li><p>The random variables <span class="math inline">\(Y_i\)</span> are
characterized by a distribution or density function which has typically
unknown parameters, e.g. a Poisson distribution <span
class="math inline">\(f(Y_i)\sim
\text{Poisson}(\theta_i)\)</span>.</p></li>
<li><p>When the subjects are mutually independent the joint likelihood
to observe <span class="math inline">\(y_1, \ldots, y_n\)</span> equals
<span class="math display">\[\prod\limits_{i=1}^n
f(y_i,\theta_i,\phi)\]</span></p></li>
<li><p>The densities are actually also a function of the parameters
<span class="math inline">\(\theta_i, \phi\)</span>. To stress this, we
indicated that in the density formulation.</p></li>
<li><p>The likelihood function is a function of all parameters</p></li>
</ul>
<p><span class="math display">\[
L(\boldsymbol{\theta},\phi\vert \boldsymbol{y})=\prod\limits_{i=1}^n
f(y_i,\theta_i,\phi)
\]</span></p>
<ul>
<li>The log-likelihood function is often used, which is defined as <span
class="math display">\[
l(\boldsymbol{\theta},\phi\vert \boldsymbol{y}) = \log
L(\boldsymbol{\theta},\phi\vert \boldsymbol{y}) =  \sum\limits_{i=1}^n
\log  f(y_i,\theta_i,\phi)
\]</span></li>
</ul>
<div id="poisson-example" class="section level3" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Poisson
Example</h3>
<p>The log-likelihood for our simulated dataset given the real model
parameters is:</p>
<p>For one observation: <span class="math display">\[l(\mu_i \vert y_i)
= y_i \log \mu_i - \mu_i - \log y_i!\]</span></p>
<ul>
<li>Verify in R.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>muTrue <span class="ot">&lt;-</span> <span class="fu">exp</span>(etaTrue)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>loglikPois <span class="ot">&lt;-</span> <span class="fu">dpois</span>(y,muTrue,<span class="at">log =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>logLikSelf <span class="ot">&lt;-</span> y<span class="sc">*</span><span class="fu">log</span>(muTrue) <span class="sc">-</span> muTrue <span class="sc">-</span> <span class="fu">lfactorial</span>(y)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="fu">qplot</span>(loglikPois,logLikSelf) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope=</span><span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Warning: `qplot()` was deprecated in ggplot2 3.4.0.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.</code></pre>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>The log-likelihood can also be written in terms of the canonical
model parameters <span class="math inline">\(\theta\)</span></p>
<p><span class="math display">\[l(\mu_i \vert y_i) = y_i \theta_i -
e^{\theta_i} - \log y_i!\]</span></p>
<ul>
<li>Note that <span class="math inline">\(\theta_i = \eta_i\)</span>.
The canonical parameter for the poisson equals the linear predictor!
<span
class="math display">\[\theta_i=\eta_i=\mathbf{x}_i^t\boldsymbol{\beta}\]</span></li>
</ul>
<p>Log-likelihood for all observations, given that they are independent:
<span class="math display">\[l(\boldsymbol{\mu} \vert \mathbf{y}) =
\sum\limits_{i=1}^n \left\{ y_i \theta_i - e^{\theta_i} - \log
y_i!\right\}\]</span></p>
<ul>
<li>Calculate in R:</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">dpois</span>(y,<span class="at">lambda =</span> muTrue, <span class="at">log =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="fu">sum</span>()</span></code></pre></div>
<pre><code>## [1] -260.485</code></pre>
</div>
<div id="properties-of-the-log-likelihood" class="section level3"
number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Properties of the
Log-Likelihood</h3>
<p><span class="math display">\[
l(\theta_i,\phi\vert y_i)=\left\{ \frac{y_i\theta_i-
b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}
\]</span></p>
<ul>
<li><span
class="math inline">\(E[y_i]=\mu_i=b^\prime(\theta_i)\)</span></li>
<li><span
class="math inline">\(\text{var}[y_i]=b^{\prime\prime}(\theta_i)
a(\phi)\)</span></li>
</ul>
<p>Note that,</p>
<ul>
<li>Variance <span class="math inline">\(\text{var}[y_i]\)</span>
depends on mean!</li>
<li>Often there is no dispersion parameter e.g. Bernouilli: <span
class="math inline">\(\text{var}[y_i]=\mu_i(1-\mu_i)\)</span>,</li>
</ul>
</div>
<div id="poisson-example-1" class="section level3" number="1.4.3">
<h3><span class="header-section-number">1.4.3</span> Poisson
Example</h3>
<ul>
<li>Canonical model parameter <span
class="math inline">\(\theta_i=\log{\mu_i}\)</span>.</li>
<li><span class="math inline">\(b(\theta_i) =
\exp(\theta_i)\)</span></li>
<li><span class="math inline">\(c(y_i,\phi) = - \log(y_i!)\)</span></li>
<li><span class="math inline">\(\phi = 1\)</span></li>
<li><span class="math inline">\(a(\phi)= 1\)</span></li>
<li><span class="math inline">\(\mu_i =\frac{\partial
b(\theta_i)}{\partial \theta_i}=  \frac{\partial
\exp(\theta_i)}{\partial \theta_i}=\exp(\theta_i)\)</span></li>
<li><span class="math inline">\(\text{Var}\left[y_i \right]=
a(\phi)\frac{\partial^2 b(\theta_i)}{(\partial
\theta_i)^2}=  \frac{\partial^2 \exp(\theta_i)}{\partial
\theta_i^2}=\exp(\theta_i)\)</span>.</li>
<li>Mean is equal to variance for Poisson!</li>
</ul>
</div>
</div>
<div id="parameter-estimation-maximum-likelihood" class="section level2"
number="1.5">
<h2><span class="header-section-number">1.5</span> Parameter Estimation:
Maximum Likelihood</h2>
<p>Choose the parameters <span
class="math inline">\(\boldsymbol{\beta}\)</span> so that the likelihood
to observe the sample under the statistical model is maximized.</p>
<p>It is easier and equivalent to maximize the log-likihood</p>
<p><span class="math display">\[\text{argmax}_{\boldsymbol{\beta}}
l(\boldsymbol{\mu} \vert \mathbf{y})\]</span></p>
<p><span class="math display">\[
\frac{\partial  l(\boldsymbol{\mu} \vert \mathbf{y})}{
\partial{\boldsymbol{\beta}}}=0
\]</span></p>
<p><span class="math inline">\(\frac{\partial  l(\boldsymbol{\mu} \vert
\mathbf{y})}{
\partial{\boldsymbol{\beta}}}\)</span> is also referred to as the score
function.</p>
<div id="score-function" class="section level3" number="1.5.1">
<h3><span class="header-section-number">1.5.1</span> Score function</h3>
<p><span class="math display">\[
S_i(\theta_i)= \frac{\partial l(\theta_i,\phi\vert y_i)}{\partial
\theta_i}
\]</span></p>
<p><span class="math display">\[
S_i(\theta_i)= \frac{\partial l(\theta_i,\phi\vert y_i)}{\partial
\theta_i}=\frac{y_i - \mu_i}{a(\phi)}
\]</span></p>
<p>when canonical link function is used:</p>
<ul>
<li><span class="math inline">\(\mu_i=b^\prime(\theta_i)\)</span></li>
</ul>
<p>Regression (chain rule and <span
class="math inline">\(i=1,\ldots,n\)</span> i.i.d observations)</p>
<p><span class="math display">\[\begin{eqnarray*}
S(\boldsymbol{\beta})&amp;=&amp;\frac{\partial  \sum\limits_{i=1}^n
\left\{ \frac{y_i \theta_i -
b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}}{\partial \beta}\\
&amp;=&amp;\sum\limits_{i=1}^n\frac{\partial  \left\{ \frac{y_i \theta_i
- b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}}{\partial
\theta}\frac{\partial
\theta}{\partial\mu}\frac{\partial\mu}{\partial\eta}\frac{\partial\eta}{\partial\beta}\\
&amp;=&amp;\sum\limits_{i=1}^n\frac{y_i-\mu_i}{a(\phi)}\frac{1}{b^{\prime\prime}(\theta)}\frac{\partial\mu}{\partial\eta}\mathbf{x}^t\\
&amp;=&amp;\mathbf{X}^T\mathbf{A}\left(\mathbf{y}-\boldsymbol{\mu}\right)
\end{eqnarray*}\]</span></p>
<ul>
<li><span class="math inline">\(\boldsymbol{A}\)</span> is a diagonal
matrix: <span
class="math inline">\(a_{ii}=\left(\text{var}[y_i]\frac{\partial
\eta_i}{\partial \mu_i}\right)^{-1}\)</span>, <span
class="math inline">\(\boldsymbol{y}=[y_1,\ldots, y_n]^T\)</span>, <span
class="math inline">\(\boldsymbol{\mu}=[\mu_1,\ldots,\mu_n]^T\)</span></li>
</ul>
<div id="poisson-example-2" class="section level4" number="1.5.1.1">
<h4><span class="header-section-number">1.5.1.1</span> Poisson
Example</h4>
<p>For poisson data: <span
class="math inline">\(a(\phi)b^{\prime\prime}(\theta)=\mu\)</span> and
<span class="math inline">\(\frac{\partial \mu}{\partial
\eta}=\mu\)</span>. So <span
class="math inline">\(\mathbf{A}=\mathbf{I}\)</span> and</p>
<p><span class="math display">\[ S(\beta)=\mathbf{X}^T
\left\{\mathbf{Y}-\boldsymbol{\mu}\right\}\]</span></p>
</div>
</div>
<div id="solve-score-equations" class="section level3" number="1.5.2">
<h3><span class="header-section-number">1.5.2</span> Solve Score
Equations</h3>
<p>Find parameter estimator <span
class="math inline">\(\hat{\boldsymbol{\beta}}\)</span> so that <span
class="math display">\[ S(\boldsymbol{\beta}) = \mathbf{0}\]</span>
<span
class="math display">\[\mathbf{X}^T\mathbf{A}\left(\mathbf{y}-\boldsymbol{\mu}\right)
=\mathbf{0}\]</span> Problem! Non linear in <span
class="math inline">\(\beta\)</span> due to</p>
<ul>
<li>link function: <span
class="math inline">\(\boldsymbol{\mu}=h^{-1}(\eta)\)</span></li>
<li><span
class="math inline">\(a_{ii}=\left(\text{var}[y_i]\frac{\partial
\eta_i}{\partial \mu_i}\right)^{-1}\)</span></li>
</ul>
<p><span class="math inline">\(\rightarrow\)</span> Find roots of score
equation by using Newton-Raphson method.</p>
</div>
<div id="newton-raphson" class="section level3" number="1.5.3">
<h3><span class="header-section-number">1.5.3</span> Newton-Raphson</h3>
<div class="figure">
<img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-7-1.png" alt="Newton-Raphson algorithm: The algorithm starts at an initial guess bk for the root of the score function S. It walks along the tangent line of the score function in S(bk)  to move into the direction where the score function becomes zero (and where the log-likelihood function is maximal). The new estimate of the root is taken at bk+1 where the tangent line becomes zero. It then calculates the score for the updated parameter estimator and the whole process is repeated until the root is found." width="672" />
<p class="caption">
Newton-Raphson algorithm: The algorithm starts at an initial guess bk
for the root of the score function S. It walks along the tangent line of
the score function in S(bk) to move into the direction where the score
function becomes zero (and where the log-likelihood function is
maximal). The new estimate of the root is taken at bk+1 where the
tangent line becomes zero. It then calculates the score for the updated
parameter estimator and the whole process is repeated until the root is
found.
</p>
</div>
<p>Newton Raphson algorithm to find the root of the score function.</p>
<ol style="list-style-type: decimal">
<li>Choose initial parameter estimate <span
class="math inline">\(\boldsymbol{\beta}^k=\boldsymbol{\beta}^0\)</span></li>
<li>Calculate score <span
class="math inline">\(S(\boldsymbol{\beta})\vert_{\boldsymbol{\beta}=\boldsymbol{\beta}^k}\)</span></li>
<li>Calculate derivative of the function for which you want to calculate
the roots</li>
<li>Walk along first derivative until line (plane) of the derivative
crosses zero</li>
<li>Update the betas <span
class="math inline">\(\boldsymbol{\beta}^{k+1}\)</span></li>
<li>Iterate from step 2 - 5 until convergence.</li>
</ol>
<div id="derivative-of-score" class="section level4" number="1.5.3.1">
<h4><span class="header-section-number">1.5.3.1</span> Derivative of
Score</h4>
<p>We have to implement an interative algorithm for optimisation. To
make things tractable we will act as if <span
class="math inline">\(\mathbf{A}\)</span> is known and fix it using the
current values of <span
class="math inline">\(\boldsymbol{\beta}^k\)</span>. Note, that for
Poisson regression <span
class="math inline">\(\mathbf{A}=\mathbf{I}\)</span>.</p>
<p><span class="math display">\[\begin{eqnarray*}
\frac{\partial S(\boldsymbol{\beta})}{\partial \boldsymbol{\beta}}
&amp;=&amp; \frac{\partial \mathbf{X}^T\mathbf{A}
\left\{\mathbf{Y}-\boldsymbol{\mu}\right\}}{\partial
\boldsymbol{\beta}}\\
&amp;\approx&amp; - \mathbf{X}^T \mathbf{A}\left[
\begin{array}{cccc} \frac{\partial \mu_1}{\partial \eta_1}
&amp;0&amp;\ldots&amp;0\\
0&amp;\frac{\partial \mu_2}{\partial \eta_2} &amp;\ldots&amp;0\\
\vdots&amp;\vdots&amp;\vdots&amp;\vdots\\
0&amp;0&amp;\ldots&amp; \frac{\partial \mu_n}{\partial \eta_n}\\
\end{array}\right] \frac{\partial \boldsymbol{\eta}}{\partial
\boldsymbol{\beta}}\\
&amp;\approx&amp;-\mathbf{X}^T\mathbf{WX}
\end{eqnarray*}\]</span></p>
</div>
<div id="define-equation-of-tangent-line-plane" class="section level4"
number="1.5.3.2">
<h4><span class="header-section-number">1.5.3.2</span> Define equation
of Tangent Line (Plane)</h4>
<ul>
<li><p>We know two points of the tangent plane <span
class="math inline">\((\boldsymbol{\beta}^k,S(\boldsymbol{\beta}^k))\)</span>
and <span
class="math inline">\((\boldsymbol{\beta}^{k+1},0)\)</span></p></li>
<li><p>We know the direction of the plane <span
class="math inline">\(S^\prime(\boldsymbol{\beta})=\frac{\partial
S(\boldsymbol{\beta})}{\partial \boldsymbol{\beta}}\)</span></p></li>
<li><p>Equation of Plane: <span
class="math display">\[S(\boldsymbol{\beta})={\alpha}_0+S^\prime\vert_{\boldsymbol{\beta}^k}
\boldsymbol{\beta}\]</span></p></li>
<li><p>Get <span class="math inline">\(\boldsymbol{\beta}_{k+1}\)</span>
<span class="math display">\[\begin{eqnarray*}
\mathbf{0}&amp;=&amp;{\alpha}_0+S^\prime\vert_{\boldsymbol{\beta}^{k}}
\boldsymbol{\beta}^{k+1}\\
\boldsymbol{\beta}^{k+1}&amp;=&amp;-\left(S^{\prime}\vert_{\boldsymbol{\beta}^{k}}\right)^{-1}{\alpha}_0\\
\end{eqnarray*}\]</span></p></li>
<li><p>Get <span class="math inline">\({\alpha}_0\)</span> <span
class="math display">\[\begin{eqnarray*}
S(\boldsymbol{\beta}^k)&amp;=&amp;\boldsymbol{\alpha}_0+S^\prime\vert_{\boldsymbol{\beta}^k}
\boldsymbol{\beta}^k\\
{\alpha}_0&amp;=&amp;-S^\prime\vert_{\boldsymbol{\beta}^k}
\boldsymbol{\beta}^k + S(\boldsymbol{\beta}^k)\\
\end{eqnarray*}\]</span></p></li>
<li><p>Get <span
class="math inline">\(\boldsymbol{\beta}_{k+1}\)</span></p></li>
</ul>
<p><span class="math display">\[\begin{eqnarray*}
\boldsymbol{\beta}^{k+1}&amp;=&amp;\boldsymbol{\beta}^k-\left(S^{\prime}\vert_{\boldsymbol{\beta}^{k}}\right)^{-1}S(\boldsymbol{\beta}^k)\\
\boldsymbol{\beta}^{k+1}&amp;=&amp;\boldsymbol{\beta}^k+
\left(\mathbf{X}^T\mathbf{WX}\right)^{-1} S(\boldsymbol{\beta}^k)
\end{eqnarray*}\]</span></p>
<p>With <span
class="math inline">\(J(\boldsymbol{\beta})=I(\boldsymbol{\beta})=\mathbf{X}^T\mathbf{WX}\)</span>
the Fisher information matrix.</p>
</div>
</div>
<div id="fisher-scoring" class="section level3" number="1.5.4">
<h3><span class="header-section-number">1.5.4</span> Fisher Scoring</h3>
<p>Because we use the canonical model parameters the observed Fisher
information matrix equals the expected Fisher information matrix <span
class="math inline">\(J(\boldsymbol{\beta})=I(\boldsymbol{\beta})\)</span>.
Indeed, the observed Fisher information matrix is not depending on the
observations, but only on the design and the variance of the data (via
the weights).</p>
<p>Hence, Newton-Raphson is equivalent to Fisher scoring when the
canonical link function is used.</p>
<p>Note, that the Fisher matrix, minus second derivative (or hessian) of
the likelihood to the model parameters, is also the inverse of the
variance covariance matrix of the model parameters. It is thus related
to the precision.</p>
</div>
<div id="iteratively-reweighted-least-squares-irls."
class="section level3" number="1.5.5">
<h3><span class="header-section-number">1.5.5</span> Iteratively
Reweighted Least Squares (IRLS).</h3>
<p>We can rewrite Newton Raphson or Fisher scoring as IRLS.</p>
<p><span class="math display">\[\begin{eqnarray*}
\boldsymbol{\beta}^{k+1}&amp;=&amp;\boldsymbol{\beta}^k+
\left(\mathbf{X}^T\mathbf{WX}\right)^{-1} S(\boldsymbol{\beta}^k)\\
\boldsymbol{\beta}^{k+1}&amp;=&amp;\boldsymbol{\beta}^k+
\left(\mathbf{X}^T\mathbf{WX}\right)^{-1} \mathbf{X}^T\mathbf{A}
\left(\mathbf{Y}-\boldsymbol{\mu}\right)\\
\boldsymbol{\beta}^{k+1}&amp;=&amp;
\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\mathbf{X}^T\mathbf{WX}\boldsymbol{\beta}^k+
\left(\boldsymbol{X}^T\mathbf{WX}\right)^{-1} \mathbf{X}^T
\mathbf{W}\frac{\partial \eta}{\partial
\mu}  \left(\mathbf{Y}-\boldsymbol{\mu}\right)\\
\boldsymbol{\beta}^{k+1}&amp;=&amp;
\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\mathbf{X}^T\mathbf{W}
\left[\mathbf{X}\boldsymbol{\beta}^k + \frac{\partial \eta}{\partial
\mu}  \left(\mathbf{Y}-\boldsymbol{\mu}\right)
\right]\\
\boldsymbol{\beta}^{k+1}&amp;=&amp;
\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\mathbf{X}^T\mathbf{Wz}
\end{eqnarray*}\]</span></p>
<p>with <span
class="math inline">\(\mathbf{z}=\left[\mathbf{X}\boldsymbol{\beta}^k +
\frac{\partial \eta}{\partial
\mu}  \left(\mathbf{Y}-\boldsymbol{\mu}\right)\right]\)</span></p>
<p>So we can fit the model by performing iterative regressions of the
pseudo data <span class="math inline">\(\mathbf{z}\)</span> on <span
class="math inline">\(\mathbf{X}\)</span>. In each iteration we will
update <span class="math inline">\(\mathbf{z}\)</span>, the weights
<span class="math inline">\(\mathbf{W}\)</span> and the model
parameters.</p>
<p>For Poisson data</p>
<ul>
<li><span class="math inline">\(\frac{\partial \eta}{\partial
\mu}=\frac{\partial\log
\mu}{\partial\mu}=\frac{1}{\mu}=\exp(-\eta)\)</span><br />
</li>
<li><span
class="math inline">\(\mathbf{W}=\mathbf{A}\frac{\partial{\mu}}{{\partial
\eta}}\)</span> is a diagonal matrix with <span
class="math inline">\([\frac{\partial{\mu_i}}{{\partial
\eta_i}}]_{ii}=[\mu_i]_{ii}=[\exp(\eta_i)]_{ii}\)</span> on its diagonal
elements.</li>
</ul>
</div>
<div id="variance-covariance-matrix-of-mean-model-parameters"
class="section level3" number="1.5.6">
<h3><span class="header-section-number">1.5.6</span> Variance-Covariance
Matrix of Mean Model Parameters?</h3>
<p>In the IRWLS algorithm, the data is weighted according to the
variance of <span class="math inline">\(\mathbf{Y}\)</span>. We correct
for the fact that the data are heteroscedastic.</p>
<p>Count data have a mean variance relation (e.g. in Poisson case <span
class="math inline">\(\text{E}\left[Y \right]=\text{var}\left[Y
\right]=\mu\)</span>). The IRWLS also corrects for the scale parameter
<span class="math inline">\(\phi\)</span> in <span
class="math inline">\(\mathbf{W}\)</span>. (Note that the scale
parameter for Poisson is <span
class="math inline">\(\phi=1\)</span>).</p>
<p>So IRWLS the variance-covariance matrix for the model parameter
equals <span
class="math display">\[\mathbf{\Sigma}_{\hat\beta}=\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}.\]</span></p>
<p>Note, that the Fisher Information Matrix equals the inverse of the
variance-covariance matrix of the experiment. The larger the Fisher
Information Matrix the more information we have on the experiment to
estimate the model parameters. FIM <span
class="math inline">\(\uparrow\)</span>, precision <span
class="math inline">\(\uparrow\)</span>, <span
class="math inline">\(\text{SE}\downarrow\)</span></p>
</div>
</div>
<div id="poisson-example-3" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Poisson Example</h2>
<div id="initial-estimate" class="section level3" number="1.6.1">
<h3><span class="header-section-number">1.6.1</span> Initial
Estimate</h3>
<p>This is a very poor initial estimate used to illustrate the
algorithm. Otherwise convergence for this simple example is way too
quick</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>iteration<span class="ot">=</span><span class="dv">0</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>betas<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="fu">log</span>(<span class="fu">mean</span>(y)),<span class="dv">0</span>,<span class="dv">0</span>)</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="fu">plot</span>(betasTrue,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>     <span class="at">ylab=</span><span class="fu">expression</span>(beta),</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>     <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>),</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>     <span class="at">pch=</span><span class="dv">19</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>     <span class="at">type=</span><span class="st">&quot;b&quot;</span>, </span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>     <span class="at">main=</span><span class="fu">paste0</span>(<span class="st">&quot;likelihood real beta=&quot;</span>,</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>                 <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(etaTrue),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>),<span class="st">&quot;</span><span class="sc">\n</span><span class="st">likelihood fit=&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(xhlp<span class="sc">%*%</span>betas),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>))</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>)</span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="fu">lines</span>(betas,<span class="at">type=</span><span class="st">&quot;b&quot;</span>,<span class="at">lty=</span><span class="dv">2</span> )</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
<div id="iteratively-reweighted-least-squares" class="section level3"
number="1.6.2">
<h3><span class="header-section-number">1.6.2</span> Iteratively
Reweighted Least Squares</h3>
<div id="pseudo-data" class="section level4" number="1.6.2.1">
<h4><span class="header-section-number">1.6.2.1</span> Pseudo Data</h4>
<p><span class="math display">\[z_i= \eta_i + \frac{\partial
\eta_i}{\partial \mu_i}(y_i -\mu_i)\]</span> <span
class="math display">\[z_i= \eta_i + e^{-\eta_i} y_i -1\]</span></p>
</div>
<div id="weight-matrix" class="section level4" number="1.6.2.2">
<h4><span class="header-section-number">1.6.2.2</span> Weight
Matrix?</h4>
<p><span class="math display">\[[w_{ii}]= var_{y_i}^{-1}
\left(\frac{\partial \mu}{\partial \eta}\right)^2\]</span> <span
class="math display">\[[w_{ii}]= e^{\eta_i}\]</span></p>
</div>
<div id="run-update-step-multiple-times" class="section level4"
number="1.6.2.3">
<h4><span class="header-section-number">1.6.2.3</span> Run Update Step
Multiple Times</h4>
<p>First 3 times (colors are black 0, red iteration 1, green iteration
2, blue iteration 3)</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">plot</span>(betasTrue,<span class="at">ylab=</span><span class="fu">expression</span>(beta),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>),<span class="at">pch=</span><span class="dv">19</span>,<span class="at">type=</span><span class="st">&quot;b&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">lines</span>(betas,<span class="at">type=</span><span class="st">&quot;b&quot;</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">likelihood TRUE=&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(xhlp<span class="sc">%*%</span>betasTrue),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>))</span></code></pre></div>
<pre><code>## 
## likelihood TRUE= -260.5</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">likelihood initial fit=&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(xhlp<span class="sc">%*%</span>betas),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>))</span></code></pre></div>
<pre><code>## 
## likelihood initial fit= -2891.5</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co">#Calculate current eta</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>eta<span class="ot">&lt;-</span>xhlp<span class="sc">%*%</span>betas</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>iteration<span class="ot">=</span><span class="dv">0</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>{</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#start IRLS UPDATE STEP</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>iteration<span class="ot">=</span>iteration<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#calculate pseudo data based on current betas</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>z<span class="ot">=</span>eta<span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span>eta)<span class="sc">*</span>(y<span class="sc">-</span><span class="fu">exp</span>(eta))</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#calculate new weights: diagonal elements</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>w<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="fu">exp</span>(eta))</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#update betas</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a>lmUpdate<span class="ot">&lt;-</span><span class="fu">lm</span>(z<span class="sc">~-</span><span class="dv">1</span><span class="sc">+</span>xhlp,<span class="at">weight=</span>w)</span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#eta&lt;-xhlp%*%betas</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a>eta<span class="ot">&lt;-</span>lmUpdate<span class="sc">$</span>fitted</span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a>betas<span class="ot">&lt;-</span>lmUpdate<span class="sc">$</span>coef</span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="fu">lines</span>(betas,<span class="at">type=</span><span class="st">&quot;b&quot;</span>,<span class="at">col=</span>iteration<span class="sc">+</span><span class="dv">1</span>,<span class="at">pch=</span>iteration,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">likelihood current fit=&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(xhlp<span class="sc">%*%</span>betas),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>))</span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre><code>## 
## likelihood current fit= -1883.9
## likelihood current fit= -431.8
## likelihood current fit= -261.7</code></pre>
</div>
</div>
</div>
<div id="comparison-with-glm-function" class="section level2"
number="1.7">
<h2><span class="header-section-number">1.7</span> Comparison with GLM
Function</h2>
<div id="smarter-initialisation" class="section level3" number="1.7.1">
<h3><span class="header-section-number">1.7.1</span> Smarter
Initialisation</h3>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>z<span class="ot">&lt;-</span><span class="fu">log</span>(y<span class="fl">+.5</span>)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>betas<span class="ot">&lt;-</span><span class="fu">lm</span>(z<span class="sc">~-</span><span class="dv">1</span><span class="sc">+</span>xhlp)<span class="sc">$</span>coef</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="fu">plot</span>(betasTrue,<span class="at">ylab=</span><span class="fu">expression</span>(beta),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>),<span class="at">pch=</span><span class="dv">19</span>,<span class="at">type=</span><span class="st">&quot;b&quot;</span>)</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="fu">lines</span>(betas,<span class="at">col=</span><span class="dv">2</span>,<span class="at">type=</span><span class="st">&quot;b&quot;</span>,<span class="at">lty=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co">#calculate current eta</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>eta<span class="ot">&lt;-</span>xhlp<span class="sc">%*%</span>betas</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">likelihood TRUE=&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(xhlp<span class="sc">%*%</span>betasTrue),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>))</span></code></pre></div>
<pre><code>## 
## likelihood TRUE= -260.5</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">likelihood initial fit=&quot;</span>, <span class="fu">round</span>(<span class="fu">sum</span>(<span class="fu">dpois</span>(y,<span class="fu">exp</span>(xhlp<span class="sc">%*%</span>betas),<span class="at">log=</span><span class="cn">TRUE</span>)),<span class="dv">1</span>))</span></code></pre></div>
<pre><code>## 
## likelihood initial fit= -263.8</code></pre>
</div>
<div id="evaluation-stopping-criterion" class="section level3"
number="1.7.2">
<h3><span class="header-section-number">1.7.2</span> Evaluation Stopping
Criterion</h3>
<ul>
<li>Residual deviance: Is 2 log of LR between best possible fit and
current fit <span
class="math display">\[LR=\frac{L_\text{best}}{L_\text{current}}\]</span>
<span class="math display">\[D=2 (\log L_\text{best}- \log
L_\text{current})\]</span> <span class="math display">\[D=2
(l_\text{best}-l_\text{current})\]</span></li>
<li>Best fit: <span class="math inline">\(\mu=y\)</span></li>
<li>Optimal poisson: <span class="math display">\[
l_\text{best}=\sum\left[y_i \log(y_i) - y_i -
\log\left(y_i!\right)\right]\]</span></li>
<li>Current fit <span class="math display">\[ l_\text{current}=\sum
\left[y_i \eta_i -e^{\eta_i} - log\left(y_i!\right)\right]\]</span></li>
<li>Deviance D: <span class="math display">\[D = 2 \sum \left[ y_i
log(y_i) - y_i \eta_i - (y_i -e^{\eta_i})\right]\]</span></li>
<li>Problem to calculate it if y=0 but by apply l’Hopital’s rule we know
<span class="math display">\[\lim_{y_i \to 0} y_i \log(y_i)
=0\]</span></li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>ylogy<span class="ot">&lt;-</span><span class="cf">function</span>(y)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>{</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="fu">return</span>(<span class="fu">ifelse</span>(y<span class="sc">==</span><span class="dv">0</span>,<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(y)),y<span class="sc">*</span><span class="fu">log</span>(y)))</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>}</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>deviance<span class="ot">&lt;-</span><span class="dv">2</span><span class="sc">*</span><span class="fu">sum</span>(<span class="fu">ylogy</span>(y)<span class="sc">-</span>y<span class="sc">*</span>eta<span class="sc">-</span>(y<span class="sc">-</span><span class="fu">exp</span>(eta)))</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a>devianceOld<span class="ot">&lt;-</span><span class="fl">1e30</span></span></code></pre></div>
</div>
<div id="run-update-step-until-convergence" class="section level3"
number="1.7.3">
<h3><span class="header-section-number">1.7.3</span> Run Update Step
until Convergence</h3>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">plot</span>(betasTrue,<span class="at">ylab=</span><span class="fu">expression</span>(beta),<span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>),<span class="at">pch=</span><span class="dv">19</span>,<span class="at">type=</span><span class="st">&quot;b&quot;</span>)</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">lines</span>(betas,<span class="at">type=</span><span class="st">&quot;b&quot;</span>,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>tol<span class="ot">&lt;-</span><span class="fl">1e-6</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>iteration<span class="ot">=</span><span class="dv">0</span></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="cf">while</span>(((devianceOld<span class="sc">-</span>deviance)<span class="sc">/</span>devianceOld)<span class="sc">&gt;</span>tol)</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>{</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="co">#start IRLS UPDATE STEP</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>iteration<span class="ot">=</span>iteration<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="co">#calculate pseudo data based on current betas</span></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>z<span class="ot">=</span>eta<span class="sc">+</span><span class="fu">exp</span>(<span class="sc">-</span>eta)<span class="sc">*</span>(y<span class="sc">-</span><span class="fu">exp</span>(eta))</span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="co">#calculate new weights: diagonal elements</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>w<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="fu">exp</span>(eta))</span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a><span class="co">#update betas</span></span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>lmUpdate<span class="ot">&lt;-</span><span class="fu">lm</span>(z<span class="sc">~-</span><span class="dv">1</span><span class="sc">+</span>xhlp,<span class="at">weight=</span>w)</span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a><span class="co">#eta&lt;-xhlp%*%betas</span></span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a>eta<span class="ot">&lt;-</span>lmUpdate<span class="sc">$</span>fitted</span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a>betas<span class="ot">&lt;-</span>lmUpdate<span class="sc">$</span>coef</span>
<span id="cb19-20"><a href="#cb19-20" tabindex="-1"></a><span class="fu">lines</span>(betas,<span class="at">type=</span><span class="st">&quot;b&quot;</span>,<span class="at">col=</span>iteration<span class="sc">+</span><span class="dv">1</span>,<span class="at">pch=</span>iteration,<span class="at">lty=</span><span class="dv">2</span>)</span>
<span id="cb19-21"><a href="#cb19-21" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" tabindex="-1"></a><span class="co">#criterion for convergence</span></span>
<span id="cb19-23"><a href="#cb19-23" tabindex="-1"></a>devianceOld<span class="ot">&lt;-</span>deviance</span>
<span id="cb19-24"><a href="#cb19-24" tabindex="-1"></a>deviance<span class="ot">&lt;-</span><span class="dv">2</span><span class="sc">*</span><span class="fu">sum</span>(<span class="fu">ylogy</span>(y)<span class="sc">-</span>y<span class="sc">*</span>eta<span class="sc">-</span>(y<span class="sc">-</span><span class="fu">exp</span>(eta)))</span>
<span id="cb19-25"><a href="#cb19-25" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;iteration&quot;</span>,iteration,<span class="st">&quot;Deviance Old&quot;</span>,devianceOld,<span class="st">&quot;Deviance&quot;</span>, deviance,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb19-26"><a href="#cb19-26" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre><code>## iteration 1 Deviance Old 129.1127 Deviance 114.3748 
## iteration 2 Deviance Old 114.3748 Deviance 114.3374 
## iteration 3 Deviance Old 114.3374 Deviance 114.3374</code></pre>
</div>
<div id="variance-beta" class="section level3" number="1.7.4">
<h3><span class="header-section-number">1.7.4</span> Variance <span
class="math inline">\(\beta\)</span>?</h3>
<p><span
class="math display">\[\Sigma_{\beta}=\left(\mathbf{X}^T\mathbf{W}
\mathbf{X}\right)^{-1}\]</span></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>varBeta<span class="ot">=</span><span class="fu">solve</span>(<span class="fu">t</span>(xhlp)<span class="sc">%*%</span><span class="fu">diag</span>(w)<span class="sc">%*%</span>xhlp)</span></code></pre></div>
</div>
<div id="comparison-with-glm-fit" class="section level3" number="1.7.5">
<h3><span class="header-section-number">1.7.5</span> Comparison with GLM
fit</h3>
<p>Use -1 because intercept is already in xhlp</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>glmfit<span class="ot">=</span><span class="fu">glm</span>(y<span class="sc">~-</span><span class="dv">1</span><span class="sc">+</span>xhlp,<span class="at">family=</span>poisson)</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>comp<span class="ot">=</span><span class="fu">data.frame</span>(<span class="at">glmfit=</span><span class="fu">c</span>(glmfit<span class="sc">$</span>deviance,glmfit<span class="sc">$</span>coef,<span class="fu">summary</span>(glmfit)<span class="sc">$</span>coef[,<span class="dv">2</span>]),<span class="at">ourFit=</span><span class="fu">c</span>(deviance,betas,<span class="fu">sqrt</span>(<span class="fu">diag</span>(varBeta))))</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="fu">row.names</span>(comp)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;deviance&quot;</span>,<span class="fu">paste</span>(<span class="st">&quot;beta&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>),<span class="fu">paste</span>(<span class="st">&quot;se&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>comp</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">glmfit</th>
<th align="right">ourFit</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">deviance</td>
<td align="right">114.3373950</td>
<td align="right">114.3373950</td>
</tr>
<tr class="even">
<td align="left">beta1</td>
<td align="right">1.9680569</td>
<td align="right">1.9680569</td>
</tr>
<tr class="odd">
<td align="left">beta2</td>
<td align="right">0.7613664</td>
<td align="right">0.7613664</td>
</tr>
<tr class="even">
<td align="left">beta3</td>
<td align="right">1.2333003</td>
<td align="right">1.2333003</td>
</tr>
<tr class="odd">
<td align="left">se1</td>
<td align="right">0.0381438</td>
<td align="right">0.0381438</td>
</tr>
<tr class="even">
<td align="left">se2</td>
<td align="right">0.0189121</td>
<td align="right">0.0189120</td>
</tr>
<tr class="odd">
<td align="left">se3</td>
<td align="right">0.0255667</td>
<td align="right">0.0255666</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="hypothesis-testing-large-sample-theory" class="section level2"
number="1.8">
<h2><span class="header-section-number">1.8</span> Hypothesis testing:
Large sample theory</h2>
<div id="wald-test" class="section level3" number="1.8.1">
<h3><span class="header-section-number">1.8.1</span> Wald test</h3>
<ul>
<li>Follows immediately from the information matrix for generalized
linear models <span class="math display">\[I(\boldsymbol{\beta}) =
\mathbf{X}^T\mathbf{WX}\]</span></li>
</ul>
<p>so large sample distribution of the maximum likelihood estimator
<span class="math inline">\(\hat{\boldsymbol{\beta}}\)</span> is
multivariate normal <span class="math display">\[
\hat{\boldsymbol{\beta}} \sim
MVN\left[\boldsymbol{\beta},\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\right]
\]</span></p>
<p>We can perform a Wald Test for a single model parameter</p>
<p><span class="math display">\[
W = \frac{\hat\beta_m}{\hat{\text{se}}_{\hat\beta_m}} \approx
N(0,1)\vert H_0
\]</span> to test for</p>
<p><span class="math display">\[
H_0: \beta_p=0 \leftrightarrow H_1:\beta_p\neq0
\]</span></p>
<p>Again, we can also assess contrasts! Indeed, linear combinations of
model parameter estimators also follow a normal distribution.</p>
<p><span class="math display">\[
\mathbf{L}^T\hat{\boldsymbol{\beta}} \sim
N\left[\mathbf{L}^T\boldsymbol{\beta},\mathbf{L}^T\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}\right]
\]</span></p>
<p>With <span class="math inline">\(\mathbf{L}\)</span> a vector for a
single contrast.</p>
<p><span class="math display">\[
W =
\frac{\mathbf{L}^T\hat{\boldsymbol{\beta}}}{\hat{\text{se}}_{\mathbf{L}^T\hat{\boldsymbol{\beta}}}}
\approx N(0,1)\vert H_0
\]</span> testing for <span class="math display">\[
H_0: \mathbf{L}^T\boldsymbol{\beta}=0 \leftrightarrow
H_1:\mathbf{L}^T\boldsymbol{\beta}\neq0
\]</span></p>
<p>We can also test for multiple contrasts simultaneously, e.g. by
assuming that multiple model parameters are zero. Suppose that <span
class="math inline">\(\mathbf{L}\)</span> is the contrast matrix that
corresponds testing for <span class="math inline">\(c\)</span> model
parameters, simultaneously. Then</p>
<p><span class="math display">\[
\mathbf{L}^T\hat{\boldsymbol{\beta}} \sim
MVN\left[\mathbf{L}^T\boldsymbol{\beta},\mathbf{L}^T\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}\right]
\]</span></p>
<p>and</p>
<p><span class="math display">\[
W =
\mathbf{L}^T\hat{\boldsymbol{\beta}}\left(\mathbf{L}^T\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}\right)^{-1}\hat{\boldsymbol{\beta}}\mathbf{L}
\sim \chi^2_c\vert H_0
\]</span> to test for</p>
<p><span class="math display">\[
H_0: \mathbf{L}^T\boldsymbol{\beta}=\mathbf{0} \leftrightarrow
H_1:\mathbf{L}^T\boldsymbol{\beta}=\mathbf{0}\neq0
\]</span></p>
<p>In general, when we test for <span
class="math inline">\(c\geq1\)</span> contrasts, then the test statistic
<span class="math inline">\(W∼\chi^2_r|H_0\)</span>, with <span
class="math inline">\(r\)</span> the rank of the contrast matrix.</p>
</div>
<div id="likelihood-ratio-test" class="section level3" number="1.8.2">
<h3><span class="header-section-number">1.8.2</span> Likelihood ratio
test</h3>
<p>The likelihood ratio test (LRT) measures the discrepancy in
log-likelihood between our current model (sometimes also referred to as
full model) and a reduced model (sometimes also referred to as null or
alternative model).</p>
<p>The reduced model must be nested in (and therefore of lower dimension
as compared to) the full model.</p>
<p>While adding more covariates will always explain more variability in
our response variable, the LRT tests whether this is actually
significant.</p>
<p>For example, in the example of gene differential expression between
healthy versus tumoral tissue, the full model could be a GLM where the
mean is modeled according to an intercept and a tissue indicator
variable (healthy / tumor), while the alternative model could be a GLM
with just an intercept. Indeed, if the gene is similarly expressed
between healthy and tumor tissue, the log-likelihood of the alternative
model will decrease only a little as compared to the full model.</p>
<p>As the name suggests, the likelihood ratio test assesses whether the
ratio of the log-likelihoods provides sufficient evidence for a worse
fit of the alternative versus full model</p>
<p><span class="math display">\[
\lambda=2\left[l(\hat{\boldsymbol{\beta}}_\text{full})-2l(\hat{\boldsymbol{\beta}}_\text{0})\right]
\]</span></p>
<pre><code>Asymptotically, under the null hypothesis it can be shown that</code></pre>
<p><span class="math display">\[ L \sim \chi_c^2 | H_0, \]</span> with
<span class="math inline">\(c\)</span> the number of parameters dropped
in the alternative model versus the full model.</p>
<p>Let <span class="math inline">\(\mathbf{C}\)</span> denote the <span
class="math inline">\(c \times p\)</span> contrast matrix denoting the
contrast for the parameters being dropped, the null and alternative
hypothesis are as in the Wald test setting: <span
class="math display">\[ H_0: \mathbf{C} \beta = 0\]</span> <span
class="math display">\[ H_1: \mathbf{C} \beta \ne 0\]</span></p>
<ul>
<li>It is important to keep in mind that standard statistical inference
theory in GLMs works <strong>asymptotically in terms of the sample
size</strong>.</li>
</ul>
<p>Thus we need many data points in order for the theory to hold in
practice. In order for the <span class="math inline">\(p\)</span>-values
to be correct, our parametric (distributional) assumptions as well as
the independence assumption, must also hold.</p>
<ul>
<li>In bulk RNA-seq, we are often working with a limited number of
samples and so we typically do not expect asymptotic theory to hold yet.
In single-cell RNA-seq, we often perform several preprocessing steps
before calculating <span class="math inline">\(p\)</span>-values for
each gene and so we may be ‘using the data multiple times’. Rather than
attaching strong probabilistic interpretations to the <span
class="math inline">\(p\)</span>-values, we therefore advice to view the
<span class="math inline">\(p\)</span>-values simply as useful numerical
summaries for ranking the genes for further inspection in genomics
applications.</li>
</ul>
<p><img src="images_sequencing/likTests.png" width="425" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="edger" class="section level1" number="2">
<h1><span class="header-section-number">2</span> EdgeR</h1>
<p><a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3378882/">McCarthy
and Smyth, 2012</a></p>
<p><img src="figs/McCarthySmyth2012.png" width="2551" /></p>
<p><img src="figs/McCarthySmyth2012_testing.png" width="2551" /></p>
</div>
<div id="edger---quasi-likelihood" class="section level1" number="3">
<h1><span class="header-section-number">3</span> EdgeR -
Quasi-Likelihood</h1>
<p><a href="https://publications.wehi.edu.au/documentaspdf/256.pdf">Lund
et al. 2012</a></p>
<p>For quasi-likelihood we do not specify the full distribution, only
the first two moments: the mean and variance.</p>
<p><span class="math display">\[
\left\{
\begin{array}{lcl}
E[y_{ig}\vert \mathbf{x}_{ig}]&amp;=&amp;\mu_{ig}\\
log(\mu_{ig})&amp;=&amp;\eta_{ig}\\
\eta_{ig}&amp;=&amp; \mathbf{x}_{i}^T\boldsymbol{\beta}+ \log N_i\\
\text{Var}[y_{ig}\vert
\mathbf{x}_{ig}]&amp;=&amp;\sigma^2_g\left(\mu_{ig}+\phi\mu_{ig}^2\right)
\end{array}\right.
\]</span></p>
<p>We will look-up the details in the paper.</p>
</div>
<div id="limma---voom" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Limma - Voom</h1>
<p><a
href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29">Law
et al. (2013). Genome Biology</a></p>
<ul>
<li>Count models vs transformation: Poisson counts, <span
class="math inline">\(\sqrt(y)\)</span> stabilises the variance,
insufficient for negative binomial. Log transformation: the transformed
data are still heteroscedastic.<span
class="math inline">\(\rightarrow\)</span> limma-voom</li>
<li>Use normalized log-cpm Limma pipeline for sequencing</li>
</ul>
<p>Problem: counts have a mean variance relationship:
heteroscedastic</p>
<p>How do we deal with heteroscedasticity in traditional linear
models?</p>
<p>Two stage approach:</p>
<ol style="list-style-type: decimal">
<li>Stage I</li>
</ol>
<ul>
<li>OLS</li>
<li>Estimate variances at each data point</li>
<li>Use variances as weights: <span
class="math inline">\(W=\text{diag}[1/\hat\sigma_i^2]\)</span></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Stage II WLS <span
class="math inline">\(\text{argmin}_{\boldsymbol{\beta}} \{
(\mathbf{y}-\mathbf{X}\boldsymbol{\beta})^T\mathbf{W}
(\mathbf{y}-\mathbf{X}\boldsymbol{\beta})\}\)</span></li>
</ol>
<p>Port this idea to RNA-seq pipeline</p>
<p><img src="figs/limmaVoomPaperChunk1.png" width="2030" /></p>
<p><img src="figs/limmaVoomPaperChunk2.png" width="1983" /></p>
<p><img src="figs/limmaVoomPaperFig2.png" width="2040" /></p>
</div>
<div id="independent-filtering" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Independent
Filtering</h1>
<p>Independent filtering is a strategy to remove features (in this case,
genes) prior to the analysis. Removal of these features may lower the
multiple testing correction for other genes that pass the filter. We try
to remove genes that have a low power to be found statistically
significant, and/or that are biologically less relevant. A common
filtering strategy is to remove genes with a generally low expression,
as low counts have lower relative uncertainty (hence lower statistical
power), and may be considered biologically less relevant.</p>
<p>Implementation in edgeR.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>?edgeR<span class="sc">::</span>filterByExpr</span></code></pre></div>
<div class="r-help-page"><div class="container"><main>

<table style="width: 100%;"><tr><td>filterByExpr</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Filter Genes By Expression Level</h2>

<h3>Description</h3>

<p>Determine which genes have sufficiently large counts to be retained in a statistical analysis.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'DGEList'
filterByExpr(y, design = NULL, group = NULL, lib.size = NULL, ...)
## S3 method for class 'SummarizedExperiment'
filterByExpr(y, design = NULL, group = NULL, lib.size = NULL, ...)
## Default S3 method:
filterByExpr(y, design = NULL, group = NULL, lib.size = NULL,
             min.count = 10, min.total.count = 15, large.n = 10, min.prop = 0.7, ...)
</code></pre>


<h3>Arguments</h3>

 
<table>
<tr><td><code id="y">y</code></td>
<td>
<p>matrix of counts, or a <code>DGEList</code> object, or a <code>SummarizedExperiment</code> object.</p>
</td></tr>
<tr><td><code id="design">design</code></td>
<td>
<p>design matrix. Ignored if <code>group</code> is not <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="group">group</code></td>
<td>
<p>vector or factor giving group membership for a oneway layout, if appropriate.</p>
</td></tr>
<tr><td><code id="lib.size">lib.size</code></td>
<td>
<p>library size, defaults to <code>colSums(y)</code>.</p>
</td></tr>
<tr><td><code id="min.count">min.count</code></td>
<td>
<p>numeric. Minimum count required for at least some samples.</p>
</td></tr>
<tr><td><code id="min.total.count">min.total.count</code></td>
<td>
<p>numeric. Minimum total count required.</p>
</td></tr>
<tr><td><code id="large.n">large.n</code></td>
<td>
<p>integer. Number of samples per group that is considered to be &ldquo;large&rdquo;.</p>
</td></tr>
<tr><td><code id="min.prop">min.prop</code></td>
<td>
<p>numeric. In large sample situations, the minimum proportion of samples in a group that a gene needs to be expressed in. See Details below for the exact formula.</p>
</td></tr>
<tr><td><code id="...">...</code></td>
<td>
<p>any other arguments.
For the <code>DGEList</code> and <code>SummarizedExperiment</code> methods, other arguments will be passed to the default method.
For the default method, other arguments are not currently used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the filtering strategy that was described informally by Chen et al (2016).
Roughly speaking, the strategy keeps genes that have at least <code>min.count</code> reads in a worthwhile number samples.
</p>
<p>More precisely, the filtering keeps genes that have <code>CPM &gt;= CPM.cutoff</code> in <code>MinSampleSize</code> samples,
where <code>CPM.cutoff = min.count/median(lib.size)*1e6</code> and <code>MinSampleSize</code> is the smallest group sample size or, more generally, the minimum inverse leverage computed from the design matrix.
</p>
<p>If all the group samples sizes are large, then the above filtering rule is relaxed slightly.
If <code>MinSampleSize &gt; large.n</code>, then genes are kept if <code>CPM &gt;= CPM.cutoff</code> in <code>k</code> samples where
<code>k = large.n + (MinSampleSize - large.n) * min.prop</code>.
This rule requires that genes are expressed in at least <code>min.prop * MinSampleSize</code> samples, even when <code>MinSampleSize</code> is large.
</p>
<p>In addition, each kept gene is required to have at least <code>min.total.count</code> reads across all the samples.
</p>


<h3>Value</h3>

<p>Logical vector of length <code>nrow(y)</code> indicating which rows of <code>y</code> to keep in the analysis.
</p>


<h3>Author(s)</h3>

<p>Gordon Smyth</p>


<h3>References</h3>

<p>Chen Y, Lun ATL, and Smyth, GK (2016).
From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline.
<em>F1000Research</em> 5, 1438.
<a href="https://f1000research.com/articles/5-1438">https://f1000research.com/articles/5-1438</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
keep &lt;- filterByExpr(y, design)
y &lt;- y[keep,]

## End(Not run)</code></pre>

</main>

</div>
</div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>  <span class="fu">library</span>(limma)</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>  <span class="fu">library</span>(edgeR)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>  <span class="fu">library</span>(DESeq2)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>})</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">makeExampleDESeqDataSet</span>()</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a>simCounts <span class="ot">&lt;-</span><span class="fu">counts</span>(dds)</span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>group <span class="ot">&lt;-</span> dds<span class="sc">$</span>condition</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>dge <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(simCounts)</span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>group)</span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(dge, design)</span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">FALSE</th>
<th align="right">TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">373</td>
<td align="right">627</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>lib.size <span class="ot">&lt;-</span> dge<span class="sc">$</span>samples<span class="sc">$</span>lib.size <span class="sc">*</span> dge<span class="sc">$</span>samples<span class="sc">$</span>norm.factors</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>cpmMinCount <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">/</span><span class="fu">median</span>(lib.size)<span class="sc">*</span><span class="fl">1e6</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="fu">summary</span>(group)</span></code></pre></div>
<pre><code>## A B 
## 6 6</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>minSampSize <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">summary</span>(group))</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>minSampSize</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">cpm</span>(dge) <span class="sc">&gt;</span> cpmMinCount) <span class="sc">&gt;=</span> minSampSize</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a><span class="fu">table</span>(keep)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">FALSE</th>
<th align="right">TRUE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">373</td>
<td align="right">627</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>leverage <span class="ot">&lt;-</span> design<span class="sc">%*%</span> <span class="fu">solve</span>(<span class="fu">t</span>(design)<span class="sc">%*%</span>design)<span class="sc">%*%</span><span class="fu">t</span>(design) <span class="sc">%&gt;%</span><span class="fu">diag</span>()</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a><span class="dv">1</span><span class="sc">/</span>leverage</span></code></pre></div>
<pre><code>##  1  2  3  4  5  6  7  8  9 10 11 12 
##  6  6  6  6  6  6  6  6  6  6  6  6</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="fu">min</span>(<span class="dv">1</span><span class="sc">/</span>leverage)</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<p>Independent filtering has been formalized by <a
href="https://www.pnas.org/content/107/21/9546">Bourgon <em>et al.</em>
(2010)</a>.</p>
<div class="figure">
<img src="images_sequencing/independentFiltering.png" alt="Figure 1 from Bourgon *et al.* (2010)." width="1440" />
<p class="caption">
Figure 1 from Bourgon <em>et al.</em> (2010).
</p>
</div>
<hr />
<p>The concept of independent filtering can be summarized as
follows:</p>
<ul>
<li>For each feature we calculate two statistics, <span
class="math inline">\(S_F\)</span> and <span
class="math inline">\(S_T\)</span>, respectively used for two stages:
filtering and testing (e.g., differential expression).</li>
<li>In order for a feature to be deemed significant, both of its
statistics must be greater than some cut-off.</li>
<li>We want to control the type I error rate of the second stage
(testing). But note that <strong>the second stage is conditional on the
first stage</strong>, as we only test features passing the filter, and
basically ignore the fact that filtering was performed. Indeed, one
criticism is that computing and correcting the <span
class="math inline">\(p\)</span>-values as if filtering had not been
performed may lead to overoptimistic adjusted <span
class="math inline">\(p\)</span>-values.</li>
<li><a href="https://www.pnas.org/content/107/21/9546">Bourgon <em>et
al.</em> (2010)</a> show that filtering is only appropriate (i.e., does
not inflate type I error rate) if the conditional null distribution of
test statistics for features passing the filter is the same as the
unconditional null distribution. Therefore, <strong>filtering is
appropriate if the statistic used for filtering is independent of the
statistic used for testing under the null hypothesis</strong>.</li>
</ul>
<div class="figure" style="text-align: center">
<img src="images_sequencing/independentFiltering2.png" alt="Figure 2 from Bourgon *et al.* (2010)." width="1438" />
<p class="caption">
Figure 2 from Bourgon <em>et al.</em> (2010).
</p>
</div>
<hr />
<p>Let’s try a couple of examples to get some intuition using simulated
data.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(DESeq2))</span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">24</span>)</span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">makeExampleDESeqDataSet</span>()</span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>simCounts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds)</span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a>group <span class="ot">&lt;-</span> dds<span class="sc">$</span>condition</span></code></pre></div>
<div id="a-dependent-test-statistic" class="section level2"
number="5.1">
<h2><span class="header-section-number">5.1</span> A Dependent Test
Statistic</h2>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>filterStatEffectSize <span class="ot">&lt;-</span> <span class="fu">abs</span>(<span class="fu">rowMeans</span>(simCounts[,group <span class="sc">==</span> <span class="st">&quot;A&quot;</span>]) <span class="sc">-</span> <span class="fu">rowMeans</span>(simCounts[,group <span class="sc">==</span> <span class="st">&quot;B&quot;</span>]))</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>testStat <span class="ot">&lt;-</span> genefilter<span class="sc">::</span><span class="fu">rowttests</span>(simCounts, group)</span></code></pre></div>
<pre><code>## Warning: replacing previous import &#39;BiocGenerics::setequal&#39; by
## &#39;S4Vectors::setequal&#39; when loading &#39;AnnotationDbi&#39;</code></pre>
<pre><code>## Warning: replacing previous import &#39;BiocGenerics::setequal&#39; by
## &#39;S4Vectors::setequal&#39; when loading &#39;Biostrings&#39;</code></pre>
<pre><code>## Warning: multiple methods tables found for &#39;union&#39;</code></pre>
<pre><code>## Warning: multiple methods tables found for &#39;intersect&#39;</code></pre>
<pre><code>## Warning: multiple methods tables found for &#39;setdiff&#39;</code></pre>
<pre><code>## Warning: multiple methods tables found for &#39;setequal&#39;</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="do">## unconditional distribution</span></span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">density</span>(testStat<span class="sc">$</span>statistic, <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Test statistic&quot;</span>,</span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Unconditional distribution&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="do">## conditional distribution: very different!</span></span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="fu">mean</span>(filterStatEffectSize <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## [1] 0.792</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="fu">hist</span>(filterStatEffectSize, <span class="at">breaks=</span><span class="dv">40</span>)</span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v=</span><span class="dv">1</span>, <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-28-2.png" width="672" /></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" tabindex="-1"></a>keepEffectSize <span class="ot">&lt;-</span> filterStatEffectSize <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb47-2"><a href="#cb47-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">density</span>(testStat<span class="sc">$</span>statistic[keepEffectSize], <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb47-3"><a href="#cb47-3" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Test statistic&quot;</span>,</span>
<span id="cb47-4"><a href="#cb47-4" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Conditional distribution&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-28-3.png" width="672" /></p>
</div>
<div id="an-independent-test-statistic" class="section level2"
number="5.2">
<h2><span class="header-section-number">5.2</span> An Independent Test
Statistic</h2>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a>filterStatGlobalMean <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(simCounts)</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a><span class="fu">mean</span>(filterStatGlobalMean <span class="sc">&gt;</span> <span class="dv">5</span>) <span class="co"># we remove a similar fraction</span></span></code></pre></div>
<pre><code>## [1] 0.771</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a>keepGlobalMean <span class="ot">&lt;-</span> filterStatGlobalMean <span class="sc">&gt;</span> <span class="dv">5</span></span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a></span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a><span class="do">## unconditional distribution</span></span>
<span id="cb50-4"><a href="#cb50-4" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">density</span>(testStat<span class="sc">$</span>statistic, <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb50-5"><a href="#cb50-5" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Test statistic&quot;</span>,</span>
<span id="cb50-6"><a href="#cb50-6" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Unconditional distribution&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a><span class="do">## conditional distribution: the same.</span></span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">density</span>(testStat<span class="sc">$</span>statistic[keepGlobalMean], <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Test statistic&quot;</span>,</span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Conditional distribution&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-29-2.png" width="672" /></p>
</div>
</div>
<div id="normalization" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Normalization</h1>
<p>Normalization is necessary to correct for several sources of
technical variation:</p>
<ul>
<li><strong>Differences in sequencing depth</strong> between samples.
Some samples get sequenced deeper in the sense that they consist of more
(mapped) reads and therefore can be considered to contain a higher
amount of information, which we should be taking into account. In
addition, if a sample is sequenced deeper, it is natural that the counts
for each gene will be higher, jeopardizing a direct comparison of the
expression counts.</li>
<li><strong>Differences in RNA population composition</strong> between
samples. As an extreme example, suppose that two samples have been
sequenced to the exact same depth. One sample is contaminated and has a
very high concentration of the contaminant cDNA being sequenced, but
otherwise the two samples are identical. Since the contaminant will be
taking up a significant proportion of the reads being sequenced, the
counts will not be directly comparable between the samples. Hence, we
may also want to correct for differences in the composition of the RNA
population of the samples.</li>
<li><strong>Other technical variation</strong> such as sample-specific
GC-content or transcript length effects may also be accounted for.</li>
</ul>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;parathyroidGenesSE&quot;</span>, <span class="at">package=</span><span class="st">&quot;parathyroidSE&quot;</span>)</span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>se1 <span class="ot">&lt;-</span> parathyroidGenesSE</span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a><span class="fu">rm</span>(parathyroidGenesSE)</span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" tabindex="-1"></a><span class="fu">colData</span>(se1) <span class="sc">%&gt;%</span> </span>
<span id="cb52-6"><a href="#cb52-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb52-7"><a href="#cb52-7" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">duplicated</span>(experiment)) </span></code></pre></div>
<table>
<colgroup>
<col width="13%" />
<col width="14%" />
<col width="10%" />
<col width="13%" />
<col width="6%" />
<col width="14%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">run</th>
<th align="left">experiment</th>
<th align="left">patient</th>
<th align="left">treatment</th>
<th align="left">time</th>
<th align="left">submission</th>
<th align="left">study</th>
<th align="left">sample</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SRR479061</td>
<td align="left">SRX140511</td>
<td align="left">2</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308873</td>
</tr>
<tr class="even">
<td align="left">SRR479064</td>
<td align="left">SRX140513</td>
<td align="left">2</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308875</td>
</tr>
<tr class="odd">
<td align="left">SRR479075</td>
<td align="left">SRX140523</td>
<td align="left">4</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308885</td>
</tr>
<tr class="even">
<td align="left">SRR479078</td>
<td align="left">SRX140525</td>
<td align="left">4</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308887</td>
</tr>
</tbody>
</table>
<p>There are technical repeats in the data.</p>
<p>We mentioned previous lectures that we can sum over technical
repeats, because techical repeats are Poisson and the sum of two Poisson
variables is again Poisson.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a>dupExps <span class="ot">&lt;-</span> <span class="fu">colData</span>(se1) <span class="sc">%&gt;%</span> </span>
<span id="cb53-2"><a href="#cb53-2" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb53-3"><a href="#cb53-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">duplicated</span>(experiment))  <span class="sc">%&gt;%</span> </span>
<span id="cb53-4"><a href="#cb53-4" tabindex="-1"></a>  <span class="fu">pull</span>(experiment)</span>
<span id="cb53-5"><a href="#cb53-5" tabindex="-1"></a></span>
<span id="cb53-6"><a href="#cb53-6" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">assays</span>(se1)<span class="sc">$</span>counts</span>
<span id="cb53-7"><a href="#cb53-7" tabindex="-1"></a>newCounts <span class="ot">&lt;-</span> counts</span>
<span id="cb53-8"><a href="#cb53-8" tabindex="-1"></a>cd <span class="ot">&lt;-</span> <span class="fu">colData</span>(se1)</span>
<span id="cb53-9"><a href="#cb53-9" tabindex="-1"></a><span class="cf">for</span>(ss <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dupExps)){</span>
<span id="cb53-10"><a href="#cb53-10" tabindex="-1"></a>  <span class="co"># check which samples are duplicates</span></span>
<span id="cb53-11"><a href="#cb53-11" tabindex="-1"></a>  relevantId <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colData</span>(se1)<span class="sc">$</span>experiment <span class="sc">==</span> dupExps[ss])</span>
<span id="cb53-12"><a href="#cb53-12" tabindex="-1"></a>  <span class="co"># sum counts</span></span>
<span id="cb53-13"><a href="#cb53-13" tabindex="-1"></a>  newCounts[,relevantId[<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(counts[,relevantId])</span>
<span id="cb53-14"><a href="#cb53-14" tabindex="-1"></a>  <span class="co"># keep which columns / rows to remove.</span></span>
<span id="cb53-15"><a href="#cb53-15" tabindex="-1"></a>  <span class="cf">if</span>(ss <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb53-16"><a href="#cb53-16" tabindex="-1"></a>    toRemove <span class="ot">&lt;-</span> relevantId[<span class="dv">2</span>]</span>
<span id="cb53-17"><a href="#cb53-17" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb53-18"><a href="#cb53-18" tabindex="-1"></a>    toRemove <span class="ot">&lt;-</span> <span class="fu">c</span>(toRemove, relevantId[<span class="dv">2</span>])</span>
<span id="cb53-19"><a href="#cb53-19" tabindex="-1"></a>  }</span>
<span id="cb53-20"><a href="#cb53-20" tabindex="-1"></a>}</span>
<span id="cb53-21"><a href="#cb53-21" tabindex="-1"></a></span>
<span id="cb53-22"><a href="#cb53-22" tabindex="-1"></a><span class="co"># remove after summing counts (otherwise IDs get mixed up)</span></span>
<span id="cb53-23"><a href="#cb53-23" tabindex="-1"></a>newCounts <span class="ot">&lt;-</span> newCounts[,<span class="sc">-</span>toRemove]</span>
<span id="cb53-24"><a href="#cb53-24" tabindex="-1"></a>newCD <span class="ot">&lt;-</span> cd[<span class="sc">-</span>toRemove,]</span>
<span id="cb53-25"><a href="#cb53-25" tabindex="-1"></a></span>
<span id="cb53-26"><a href="#cb53-26" tabindex="-1"></a><span class="co"># Create new SummarizedExperiment</span></span>
<span id="cb53-27"><a href="#cb53-27" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="st">&quot;counts&quot;</span> <span class="ot">=</span> newCounts),</span>
<span id="cb53-28"><a href="#cb53-28" tabindex="-1"></a>                            <span class="at">colData =</span> newCD,</span>
<span id="cb53-29"><a href="#cb53-29" tabindex="-1"></a>                            <span class="at">metadata =</span> <span class="fu">metadata</span>(se1))</span>
<span id="cb53-30"><a href="#cb53-30" tabindex="-1"></a></span>
<span id="cb53-31"><a href="#cb53-31" tabindex="-1"></a>treatment <span class="ot">&lt;-</span> <span class="fu">colData</span>(se)<span class="sc">$</span>treatment</span>
<span id="cb53-32"><a href="#cb53-32" tabindex="-1"></a><span class="fu">table</span>(treatment)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">Control</th>
<th align="right">DPN</th>
<th align="right">OHT</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">7</td>
<td align="right">8</td>
<td align="right">8</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a><span class="fu">qplot</span>(<span class="fu">colSums</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts)<span class="sc">/</span><span class="fl">1e6</span>, <span class="at">geom=</span><span class="st">&quot;histogram&quot;</span>, <span class="at">bins=</span><span class="dv">10</span>,<span class="at">col=</span><span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb54-3"><a href="#cb54-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;libsize (million reads)&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a><span class="fu">qplot</span>(</span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a>  <span class="fu">colData</span>(se)<span class="sc">$</span>treatment<span class="sc">:</span><span class="fu">colData</span>(se)<span class="sc">$</span>time,</span>
<span id="cb55-3"><a href="#cb55-3" tabindex="-1"></a>  <span class="fu">colSums</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts)<span class="sc">/</span><span class="fl">1e6</span>,<span class="at">geom=</span><span class="st">&quot;boxplot&quot;</span></span>
<span id="cb55-4"><a href="#cb55-4" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb55-5"><a href="#cb55-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;treatment&quot;</span>)<span class="sc">+</span></span>
<span id="cb55-6"><a href="#cb55-6" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;libsize (million reads)&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-32-2.png" width="672" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a><span class="fu">qplot</span>(</span>
<span id="cb56-2"><a href="#cb56-2" tabindex="-1"></a>  <span class="fu">colData</span>(se)<span class="sc">$</span>patient,</span>
<span id="cb56-3"><a href="#cb56-3" tabindex="-1"></a>  <span class="fu">colSums</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts)<span class="sc">/</span><span class="fl">1e6</span>,<span class="at">geom=</span><span class="st">&quot;boxplot&quot;</span></span>
<span id="cb56-4"><a href="#cb56-4" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb56-5"><a href="#cb56-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Patient&quot;</span>)<span class="sc">+</span></span>
<span id="cb56-6"><a href="#cb56-6" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;libsize (million reads)&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-32-3.png" width="672" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" tabindex="-1"></a>ma2Samp <span class="ot">&lt;-</span> <span class="cf">function</span>(countMx,<span class="at">libSize=</span><span class="cn">NULL</span>) {</span>
<span id="cb57-2"><a href="#cb57-2" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="st">&quot;`countMx` is not a matrix with two columns&quot;</span> <span class="ot">=</span> <span class="fu">ncol</span>(countMx) <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb57-3"><a href="#cb57-3" tabindex="-1"></a>A <span class="ot">&lt;-</span> countMx <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> rowMeans</span>
<span id="cb57-4"><a href="#cb57-4" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(libSize)) </span>
<span id="cb57-5"><a href="#cb57-5" tabindex="-1"></a>  M <span class="ot">&lt;-</span> countMx <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> <span class="fu">apply</span>(.,<span class="dv">1</span>,diff) </span>
<span id="cb57-6"><a href="#cb57-6" tabindex="-1"></a><span class="cf">else</span> </span>
<span id="cb57-7"><a href="#cb57-7" tabindex="-1"></a>  M <span class="ot">&lt;-</span> countMx <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> <span class="fu">apply</span>(.,<span class="dv">1</span>,diff) <span class="sc">-</span> libSize <span class="sc">%&gt;%</span> log2 <span class="sc">%&gt;%</span> diff</span>
<span id="cb57-8"><a href="#cb57-8" tabindex="-1"></a>w <span class="ot">&lt;-</span> countMx[,<span class="dv">1</span>]<span class="sc">==</span><span class="fu">min</span>(countMx[,<span class="dv">1</span>]) <span class="sc">|</span> countMx[,<span class="dv">2</span>]<span class="sc">==</span><span class="fu">min</span>(countMx[,<span class="dv">2</span>])</span>
<span id="cb57-9"><a href="#cb57-9" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">any</span>(w)) {</span>
<span id="cb57-10"><a href="#cb57-10" tabindex="-1"></a>            A[w] <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="fu">sum</span>(w), <span class="at">min =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">max =</span> .<span class="dv">1</span>)</span>
<span id="cb57-11"><a href="#cb57-11" tabindex="-1"></a>            M[w] <span class="ot">&lt;-</span> <span class="fu">log2</span>(countMx[w,<span class="dv">2</span>] <span class="sc">+</span> <span class="dv">1</span>) <span class="sc">-</span> <span class="fu">log2</span>(countMx[w,<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb57-12"><a href="#cb57-12" tabindex="-1"></a>}</span>
<span id="cb57-13"><a href="#cb57-13" tabindex="-1"></a>MAplot <span class="ot">&lt;-</span> <span class="fu">qplot</span>(A, M, <span class="at">col=</span>w) <span class="sc">+</span></span>
<span id="cb57-14"><a href="#cb57-14" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span> </span>
<span id="cb57-15"><a href="#cb57-15" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;orange&quot;</span>)) <span class="sc">+</span></span>
<span id="cb57-16"><a href="#cb57-16" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;A: log2 Average&quot;</span>) <span class="sc">+</span></span>
<span id="cb57-17"><a href="#cb57-17" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;M: log2 Fold Change&quot;</span>)</span>
<span id="cb57-18"><a href="#cb57-18" tabindex="-1"></a></span>
<span id="cb57-19"><a href="#cb57-19" tabindex="-1"></a>MAplot <span class="sc">+</span></span>
<span id="cb57-20"><a href="#cb57-20" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>,<span class="at">slope=</span><span class="dv">0</span>,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>) <span class="sc">+</span></span>
<span id="cb57-21"><a href="#cb57-21" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="fu">median</span>(M[<span class="sc">!</span>w],<span class="at">na.rm=</span><span class="cn">TRUE</span>),<span class="at">slope=</span><span class="dv">0</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb57-22"><a href="#cb57-22" tabindex="-1"></a>}</span></code></pre></div>
<p>Let’s take a look at how comparable different replicates are in the
Control condition at 48h in our dataset. We will investigate this using
MD-plots (mean-difference plots as introduced by Dudoit et al. (2002)),
also sometimes referred to as MA-plots.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" tabindex="-1"></a>ids <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colData</span>(se)<span class="sc">$</span>treatment <span class="sc">==</span><span class="st">&quot;Control&quot;</span> <span class="sc">&amp;</span> <span class="fu">colData</span>(se)<span class="sc">$</span>time <span class="sc">==</span> <span class="st">&quot;48h&quot;</span>)</span>
<span id="cb58-2"><a href="#cb58-2" tabindex="-1"></a>ids</span></code></pre></div>
<pre><code>## [1]  2  8 14 19</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" tabindex="-1"></a><span class="fu">colSums</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts[,ids]) <span class="sc">/</span> <span class="fl">1e6</span></span></code></pre></div>
<pre><code>## [1] 10.827109  6.844144  8.064268  7.701432</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" tabindex="-1"></a>pairComb <span class="ot">&lt;-</span> <span class="fu">combn</span>(</span>
<span id="cb62-2"><a href="#cb62-2" tabindex="-1"></a>  ids, </span>
<span id="cb62-3"><a href="#cb62-3" tabindex="-1"></a>  <span class="at">m=</span><span class="dv">2</span>)</span>
<span id="cb62-4"><a href="#cb62-4" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">apply</span>(pairComb,<span class="dv">2</span>,<span class="cf">function</span>(x) <span class="fu">ma2Samp</span>(<span class="fu">assay</span>(se)[,x]) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">&quot;samples&quot;</span>,x[<span class="dv">2</span>],<span class="st">&quot;vs&quot;</span>, x[<span class="dv">1</span>])))</span>
<span id="cb62-5"><a href="#cb62-5" tabindex="-1"></a><span class="fu">do.call</span>(<span class="st">&quot;grid.arrange&quot;</span>,<span class="fu">c</span>(plots,<span class="at">ncol=</span><span class="dv">3</span>))</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-35-1.png" width="672" /></p>
<p>We see clear bias for some pairwise comparisons. For example, in the
first plot comparing sample 8 versus sample 2, the log fold-changes are
biased downwards. This means that, on average, a gene is lower expressed
in sample 8 versus sample 2. Looking at the library sizes, we can indeed
see that the library size for sample 2 is about <span
class="math inline">\(11×10^6\)</span> while it is only about <span
class="math inline">\(7×10^6\)</span> for sample 8! This is a clear
library size effect that we should take into account.</p>
<p>We can solve these issues by introducing offsets in our model.</p>
<p><span class="math display">\[
  \left\{
  \begin{array}{ccc}
  Y_{gi} &amp; \sim &amp; Poi(\mu_{gi}) \\
  \log \mu_{gi} &amp; = &amp; \eta_{gi} \\
  \eta_{gi} &amp; = &amp; \mathbf{X}^T_i \beta_g + log(O_{gi}) \\
  \end{array}
  \right.
  \]</span></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" tabindex="-1"></a>libSize <span class="ot">&lt;-</span> <span class="fu">colSums</span>(<span class="fu">assay</span>(se))</span>
<span id="cb63-2"><a href="#cb63-2" tabindex="-1"></a>plots2 <span class="ot">&lt;-</span> <span class="fu">apply</span>(pairComb,<span class="dv">2</span>,<span class="cf">function</span>(x) <span class="fu">ma2Samp</span>(<span class="fu">assay</span>(se)[,x],<span class="at">libSize =</span> libSize[x]) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">&quot;samples&quot;</span>,x[<span class="dv">2</span>],<span class="st">&quot;vs&quot;</span>, x[<span class="dv">1</span>])))</span>
<span id="cb63-3"><a href="#cb63-3" tabindex="-1"></a><span class="fu">do.call</span>(<span class="st">&quot;grid.arrange&quot;</span>,<span class="fu">c</span>(plots2,<span class="at">ncol=</span><span class="dv">3</span>))</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<div id="tmm-method-default-of-edger" class="section level2"
number="6.1">
<h2><span class="header-section-number">6.1</span> TMM method (default
of <code>edgeR</code>)</h2>
<p><a
href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25">Robinson
and Oshlack (2010). Genome Biology</a></p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;./figs/edgeRNormIntro.png&quot;</span>)</span></code></pre></div>
<p><img src="figs/edgeRNormIntro.png" width="511" style="display: block; margin: auto;" /></p>
<ul>
<li>On the plot we see a clear effect on all genes</li>
<li>Correcting for library size tends to over correct.</li>
<li>Some DE genes are highly abundant and determine the library size to
a large extend</li>
</ul>
<p>The trimmed mean of M-values (TMM) method introduced by <a
href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25">Robinson
&amp; Oshlack (2010)</a> is a normalization procedure that calculates a
single normalization factor for each sample. As the name suggests, it is
based on a trimmed mean of fold-changes (<span
class="math inline">\(M\)</span>-values) as the scaling factor. A
trimmed mean is an average after removing a set of ``extreme’’ values.
Specifically, TMM calculates a normalization factor <span
class="math inline">\(F_i^{(r)}\)</span> across genes <span
class="math inline">\(g\)</span> for each sample <span
class="math inline">\(i\)</span> as compared to a reference sample <span
class="math inline">\(r\)</span>, <span class="math display">\[
\log_2(F_i^{(r)}) = \frac{\sum_{g \in {\cal G}^*} w_{gi}^r
M_{gi}^r}{\sum_{g \in {\cal G}^*} w_{gi}^r},
\]</span> where <span class="math inline">\(M_{gi}^r\)</span> represents
the <span class="math inline">\(\log_2\)</span>-fold-change of the gene
expression fraction as compared to a reference sample <span
class="math inline">\(r\)</span>, i.e., <span class="math display">\[
M_{gi}^r = \log_2\left( \frac{Y_{gi} / N_i}{ Y_{gr} / N_r} \right),
\]</span> and <span class="math inline">\(w_{gi}^r\)</span> represents a
precision weight calculated as <span class="math display">\[
w_{gi}^r = \frac{N_i - Y_{gi}}{N_i Y_{gi}} + \frac{N_r - Y_{gr}}{N_r
Y_{gr}},
\]</span> and <span class="math inline">\({\cal G}^*\)</span> represents
the set of genes after trimming those with the most extreme average
expression. The weights serve to account for the fact that fold-changes
for genes with lower read counts are more variable.</p>
<p>The procedure only takes genes into account where both <span
class="math inline">\(Y_{gi}&gt;0\)</span> and <span
class="math inline">\(Y_{gr}&gt;0\)</span>. By default, TMM trims genes
with the <span class="math inline">\(30\%\)</span> most extreme <span
class="math inline">\(M\)</span>-values and <span
class="math inline">\(5\%\)</span> most extreme average gene expression,
and chooses as reference <span class="math inline">\(r\)</span> the
sample whose upper-quartile is closest to the across-sample average
upper-quartile.</p>
<p>TMM normalization may be performed from the
<code>calcNormFactors</code> function implemented in
<code>edgeR</code>:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" tabindex="-1"></a>dge <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">calcNormFactors</span>(se)</span>
<span id="cb65-2"><a href="#cb65-2" tabindex="-1"></a>dge<span class="sc">$</span>samples <span class="co">#normalization factors added to colData</span></span></code></pre></div>
<table>
<colgroup>
<col width="8%" />
<col width="5%" />
<col width="8%" />
<col width="11%" />
<col width="8%" />
<col width="9%" />
<col width="7%" />
<col width="8%" />
<col width="4%" />
<col width="9%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">group</th>
<th align="right">lib.size</th>
<th align="right">norm.factors</th>
<th align="left">run</th>
<th align="left">experiment</th>
<th align="left">patient</th>
<th align="left">treatment</th>
<th align="left">time</th>
<th align="left">submission</th>
<th align="left">study</th>
<th align="left">sample</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Sample1</td>
<td align="left">1</td>
<td align="right">9102683</td>
<td align="right">0.9782830</td>
<td align="left">SRR479052</td>
<td align="left">SRX140503</td>
<td align="left">1</td>
<td align="left">Control</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308865</td>
</tr>
<tr class="even">
<td align="left">Sample2</td>
<td align="left">1</td>
<td align="right">10827109</td>
<td align="right">0.9728700</td>
<td align="left">SRR479053</td>
<td align="left">SRX140504</td>
<td align="left">1</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308866</td>
</tr>
<tr class="odd">
<td align="left">Sample3</td>
<td align="left">1</td>
<td align="right">5217761</td>
<td align="right">0.9898593</td>
<td align="left">SRR479054</td>
<td align="left">SRX140505</td>
<td align="left">1</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308867</td>
</tr>
<tr class="even">
<td align="left">Sample4</td>
<td align="left">1</td>
<td align="right">9706035</td>
<td align="right">0.9930169</td>
<td align="left">SRR479055</td>
<td align="left">SRX140506</td>
<td align="left">1</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308868</td>
</tr>
<tr class="odd">
<td align="left">Sample5</td>
<td align="left">1</td>
<td align="right">5700022</td>
<td align="right">0.9850867</td>
<td align="left">SRR479056</td>
<td align="left">SRX140507</td>
<td align="left">1</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308869</td>
</tr>
<tr class="even">
<td align="left">Sample6</td>
<td align="left">1</td>
<td align="right">7854568</td>
<td align="right">0.9897270</td>
<td align="left">SRR479057</td>
<td align="left">SRX140508</td>
<td align="left">1</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308870</td>
</tr>
<tr class="odd">
<td align="left">Sample7</td>
<td align="left">1</td>
<td align="right">8610014</td>
<td align="right">0.9266581</td>
<td align="left">SRR479058</td>
<td align="left">SRX140509</td>
<td align="left">2</td>
<td align="left">Control</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308871</td>
</tr>
<tr class="even">
<td align="left">Sample8</td>
<td align="left">1</td>
<td align="right">6844144</td>
<td align="right">0.9544240</td>
<td align="left">SRR479059</td>
<td align="left">SRX140510</td>
<td align="left">2</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308872</td>
</tr>
<tr class="odd">
<td align="left">Sample9</td>
<td align="left">1</td>
<td align="right">24584280</td>
<td align="right">0.9188545</td>
<td align="left">SRR479060</td>
<td align="left">SRX140511</td>
<td align="left">2</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308873</td>
</tr>
<tr class="even">
<td align="left">Sample10</td>
<td align="left">1</td>
<td align="right">8267977</td>
<td align="right">0.9398000</td>
<td align="left">SRR479062</td>
<td align="left">SRX140512</td>
<td align="left">2</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308874</td>
</tr>
<tr class="odd">
<td align="left">Sample11</td>
<td align="left">1</td>
<td align="right">23590411</td>
<td align="right">0.9096695</td>
<td align="left">SRR479063</td>
<td align="left">SRX140513</td>
<td align="left">2</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308875</td>
</tr>
<tr class="even">
<td align="left">Sample12</td>
<td align="left">1</td>
<td align="right">8247122</td>
<td align="right">0.9369050</td>
<td align="left">SRR479065</td>
<td align="left">SRX140514</td>
<td align="left">2</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308876</td>
</tr>
<tr class="odd">
<td align="left">Sample13</td>
<td align="left">1</td>
<td align="right">7341000</td>
<td align="right">1.0668032</td>
<td align="left">SRR479066</td>
<td align="left">SRX140515</td>
<td align="left">3</td>
<td align="left">Control</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308877</td>
</tr>
<tr class="even">
<td align="left">Sample14</td>
<td align="left">1</td>
<td align="right">8064268</td>
<td align="right">1.0552688</td>
<td align="left">SRR479067</td>
<td align="left">SRX140516</td>
<td align="left">3</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308878</td>
</tr>
<tr class="odd">
<td align="left">Sample15</td>
<td align="left">1</td>
<td align="right">12481958</td>
<td align="right">1.0461698</td>
<td align="left">SRR479068</td>
<td align="left">SRX140517</td>
<td align="left">3</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308879</td>
</tr>
<tr class="even">
<td align="left">Sample16</td>
<td align="left">1</td>
<td align="right">16310090</td>
<td align="right">1.0260056</td>
<td align="left">SRR479069</td>
<td align="left">SRX140518</td>
<td align="left">3</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308880</td>
</tr>
<tr class="odd">
<td align="left">Sample17</td>
<td align="left">1</td>
<td align="right">23697329</td>
<td align="right">1.0268459</td>
<td align="left">SRR479070</td>
<td align="left">SRX140519</td>
<td align="left">3</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308881</td>
</tr>
<tr class="even">
<td align="left">Sample18</td>
<td align="left">1</td>
<td align="right">7642648</td>
<td align="right">1.0409451</td>
<td align="left">SRR479071</td>
<td align="left">SRX140520</td>
<td align="left">3</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308882</td>
</tr>
<tr class="odd">
<td align="left">Sample19</td>
<td align="left">1</td>
<td align="right">7701432</td>
<td align="right">1.0559132</td>
<td align="left">SRR479072</td>
<td align="left">SRX140521</td>
<td align="left">4</td>
<td align="left">Control</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308883</td>
</tr>
<tr class="even">
<td align="left">Sample20</td>
<td align="left">1</td>
<td align="right">7135899</td>
<td align="right">1.0675040</td>
<td align="left">SRR479073</td>
<td align="left">SRX140522</td>
<td align="left">4</td>
<td align="left">DPN</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308884</td>
</tr>
<tr class="odd">
<td align="left">Sample21</td>
<td align="left">1</td>
<td align="right">13818393</td>
<td align="right">1.0327004</td>
<td align="left">SRR479074</td>
<td align="left">SRX140523</td>
<td align="left">4</td>
<td align="left">DPN</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308885</td>
</tr>
<tr class="even">
<td align="left">Sample22</td>
<td align="left">1</td>
<td align="right">6099942</td>
<td align="right">1.0890994</td>
<td align="left">SRR479076</td>
<td align="left">SRX140524</td>
<td align="left">4</td>
<td align="left">OHT</td>
<td align="left">24h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308886</td>
</tr>
<tr class="odd">
<td align="left">Sample23</td>
<td align="left">1</td>
<td align="right">15825211</td>
<td align="right">1.0286470</td>
<td align="left">SRR479077</td>
<td align="left">SRX140525</td>
<td align="left">4</td>
<td align="left">OHT</td>
<td align="left">48h</td>
<td align="left">SRA051611</td>
<td align="left">SRP012167</td>
<td align="left">SRS308887</td>
</tr>
</tbody>
</table>
<p>Let’s check how our MD-plots look like after normalization. Note
that, we can rewrite the GLM as <span class="math display">\[ \log\left(
\frac{\mu_{gi}}{N_i^s} \right) = \mathbf{X}_i^T \beta_g \]</span> and so
<span class="math inline">\(\frac{\mu_{gi}}{N_i^s}\)</span> can be
considered as an ‘offset-corrected count’.</p>
<p>We see that all MD-plots are now nicely centered around a
log-fold-change of zero!</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" tabindex="-1"></a><span class="do">## normalize</span></span>
<span id="cb66-2"><a href="#cb66-2" tabindex="-1"></a>effLibSize <span class="ot">&lt;-</span> dge<span class="sc">$</span>samples<span class="sc">$</span>lib.size <span class="sc">*</span> dge<span class="sc">$</span>samples<span class="sc">$</span>norm.factors</span>
<span id="cb66-3"><a href="#cb66-3" tabindex="-1"></a><span class="co">#normCountTMM &lt;- sweep(assays(se)$counts, 2, FUN=&quot;/&quot;, effLibSize)</span></span>
<span id="cb66-4"><a href="#cb66-4" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" tabindex="-1"></a>plotsNorm <span class="ot">&lt;-</span> <span class="fu">apply</span>(pairComb,<span class="dv">2</span>,<span class="cf">function</span>(x) </span>
<span id="cb66-6"><a href="#cb66-6" tabindex="-1"></a>  <span class="fu">ma2Samp</span>(<span class="fu">assays</span>(se)<span class="sc">$</span>counts[,x], effLibSize[x]) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste</span>(<span class="st">&quot;samples&quot;</span>,x[<span class="dv">2</span>],<span class="st">&quot;vs&quot;</span>, x[<span class="dv">1</span>])))</span>
<span id="cb66-7"><a href="#cb66-7" tabindex="-1"></a><span class="fu">do.call</span>(<span class="st">&quot;grid.arrange&quot;</span>,<span class="fu">c</span>(plotsNorm,<span class="at">ncol=</span><span class="dv">3</span>))</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
</div>
<div id="median-of-ratios-method-default-of-deseq2"
class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Median-of-Ratios
Method (default of <code>DESeq2</code>)</h2>
<p>The median-of-ratios method is used in <code>DESeq2</code> as
described in <a
href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Love
<em>et al.</em> (2014)</a>. It assumes that the expected value <span
class="math inline">\(\mu_{gi} = E(Y_{gi})\)</span> is proportional to
the true expression of the gene, <span
class="math inline">\(q_{gi}\)</span>, scaled by a normalization factor
<span class="math inline">\(s_{i}\)</span> for each sample, <span
class="math display">\[ \mu_{gi} = s_{i}q_{gi}. \]</span></p>
<p>The normalization factor <span class="math inline">\(s_{i}\)</span>
is then estimated using the median-of-ratios method compared to a
synthetic reference sample <span class="math inline">\(r\)</span>
defined based on geometric means of counts across samples <span
class="math display">\[
s_i = \text{median}_{\{{g:Y^{*}_{gr} \ne 0}\}}
\frac{Y_{gi}}{Y^{*}_{gr}},
\]</span> with <span class="math display">\[ Y^{*}_{gr} = \left(
\prod_{i=1}^n Y_{gi} \right)^{1/n}. \]</span></p>
<p>We can then use the size factors <span
class="math inline">\(s_i\)</span> as offsets to the GLM.</p>
<p>Median-of-ratios normalization is implemented in the
<code>DESeq2</code> package:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">assays</span>(se)<span class="sc">$</span>counts,</span>
<span id="cb67-2"><a href="#cb67-2" tabindex="-1"></a>                                      <span class="at">colData =</span> <span class="fu">colData</span>(se),</span>
<span id="cb67-3"><a href="#cb67-3" tabindex="-1"></a>                                      <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>) <span class="co">#just add intercept to showcase normalization</span></span></code></pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb69-2"><a href="#cb69-2" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)</span></code></pre></div>
<pre><code>##  [1] 0.9187624 1.0644717 0.5232069 0.9826698 0.5676212 0.7807662 0.8284089
##  [8] 0.6600848 2.3955069 0.7906917 2.2784591 0.7860386 0.7807842 0.8445577
## [15] 1.3345483 1.7011973 2.5001652 0.7889061 0.8138720 0.7632306 1.4518320
## [22] 0.6553677 1.6696064</code></pre>
<p>You may also want to check out the <a
href="https://www.youtube.com/watch?v=UFB993xufUU">StatQuest video on
DESeq2 normalization</a>.</p>
<div id="comparing-tmm-with-deseq2-normalization" class="section level3"
number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Comparing TMM with
DESeq2 Normalization</h3>
<p>We can compare the size factors for both normalizations to verify if
they agree on the normalization factors. Note we need to scale the
effective library sizes from <code>edgeR</code> to enforce a similar
scale as the size factors from <code>DESeq2</code>. While below we are
using an arithmetic mean, a geometric mean may be used as well, which
will be more robust to outlying effective library sizes.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" tabindex="-1"></a><span class="fu">plot</span>(effLibSize <span class="sc">/</span> <span class="fu">mean</span>(effLibSize), <span class="fu">sizeFactors</span>(dds),</span>
<span id="cb71-2"><a href="#cb71-2" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;edgeR size factor&quot;</span>,</span>
<span id="cb71-3"><a href="#cb71-3" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;DESeq2 size factor&quot;</span>)</span></code></pre></div>
<p><img src="sequencing_technicalDE_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="aliasing" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Aliasing</h1>
<p>Suppose we are working with the following experimental design on
colon cancer. Studying the effect of a drug on gene expression,
researchers gather RNA-seq data from four colon cancer patients and four
healthy individuals. For each individual, they obtain RNA-seq data from
a blood sample before as well as two weeks after taking a daily dose of
the drug. The research question relates to differential expression after
vs. before taking the drug, in particular whether this is different for
the diseased versus healthy group (i.e., the interaction between time
(before/after taking the drug) and disease status (healthy/colon
cancer)).</p>
<p>In terms of the model matrix, we could imagine a design such as
<code>~ patient + disease*time</code>, where</p>
<ul>
<li><code>disease</code> is a binary indicator referring to colon cancer
versus control sample.</li>
<li><code>time</code> defines if the sample is taken before or after
taking the drug.</li>
<li><code>patient</code> defines the individual donor the sample comes
from.</li>
</ul>
<p>The research question could then amount to testing the
<code>disease * time</code> interaction.</p>
<p>Let’s try this, by simulating random data for one gene.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2</span>)</span>
<span id="cb72-2"><a href="#cb72-2" tabindex="-1"></a><span class="co"># 2 samples per patient for 8 patients</span></span>
<span id="cb72-3"><a href="#cb72-3" tabindex="-1"></a>patient <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">rep</span>(letters[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>], <span class="at">each=</span><span class="dv">2</span>)) </span>
<span id="cb72-4"><a href="#cb72-4" tabindex="-1"></a><span class="co"># first four are healthy, next four are diseased</span></span>
<span id="cb72-5"><a href="#cb72-5" tabindex="-1"></a>disease <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;healthy&quot;</span>,<span class="dv">8</span>), <span class="fu">rep</span>(<span class="st">&quot;cancer&quot;</span>,<span class="dv">8</span>)), <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;healthy&quot;</span>, <span class="st">&quot;cancer&quot;</span>)) </span>
<span id="cb72-6"><a href="#cb72-6" tabindex="-1"></a><span class="co"># one before and one after sample for each</span></span>
<span id="cb72-7"><a href="#cb72-7" tabindex="-1"></a>time <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;before&quot;</span>, <span class="st">&quot;after&quot;</span>), <span class="dv">8</span>), <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;before&quot;</span>, <span class="st">&quot;after&quot;</span>)) </span>
<span id="cb72-8"><a href="#cb72-8" tabindex="-1"></a></span>
<span id="cb72-9"><a href="#cb72-9" tabindex="-1"></a><span class="fu">table</span>(patient, disease, time)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">patient</th>
<th align="left">disease</th>
<th align="left">time</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">a</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">b</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">c</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">d</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">e</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">f</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">g</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">h</td>
<td align="left">cancer</td>
<td align="left">after</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">healthy</td>
<td align="left">after</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left"></td>
<td align="left">before</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" tabindex="-1"></a><span class="do">## simulate data for one gene</span></span>
<span id="cb73-2"><a href="#cb73-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="dv">16</span></span>
<span id="cb73-3"><a href="#cb73-3" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n =</span> n, <span class="at">lambda =</span> <span class="dv">50</span>)</span>
<span id="cb73-4"><a href="#cb73-4" tabindex="-1"></a></span>
<span id="cb73-5"><a href="#cb73-5" tabindex="-1"></a><span class="do">## fit a Poisson model</span></span>
<span id="cb73-6"><a href="#cb73-6" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> patient <span class="sc">+</span> disease<span class="sc">*</span>time,</span>
<span id="cb73-7"><a href="#cb73-7" tabindex="-1"></a>         <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>)</span>
<span id="cb73-8"><a href="#cb73-8" tabindex="-1"></a><span class="fu">summary</span>(m)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = y ~ patient + disease * time, family = &quot;poisson&quot;)
## 
## Coefficients: (1 not defined because of singularities)
##                         Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)              3.76900    0.11916  31.631   &lt;2e-16 ***
## patientb                 0.06744    0.14999   0.450   0.6530    
## patientc                 0.06744    0.14999   0.450   0.6530    
## patientd                 0.27304    0.14310   1.908   0.0564 .  
## patiente                 0.16449    0.16224   1.014   0.3107    
## patientf                 0.02565    0.16644   0.154   0.8775    
## patientg                -0.01784    0.16785  -0.106   0.9154    
## patienth                 0.05706    0.16544   0.345   0.7302    
## diseasecancer                 NA         NA      NA       NA    
## timeafter               -0.01567    0.10220  -0.153   0.8782    
## diseasecancer:timeafter  0.12374    0.14407   0.859   0.3904    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 16.1200  on 15  degrees of freedom
## Residual deviance:  8.8417  on  6  degrees of freedom
## AIC: 120.16
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<hr />
<p>We find that one of the coefficients is <code>NA</code>! This is
obviously not because we’re dealing with <code>NA</code> values in the
data as we’ve just simulated the response variable ourselves. What’s
going on?</p>
<p>One of the parameters, in this case the parameter distinguishing
cancer from healthy patients <strong>cannot be estimated as it is a
linear combination of other parameters</strong>. In our case, estimating
the diseased effect would use information that is already used to
estimate the patient-level intercepts. In other words, <strong>once you
know the patient, you immediately also know the disease status</strong>,
so estimating the diseased vs healthy effect on top of the patient
effect provides no additional information if we have already estimated
the patient-level effects. This concept is called aliasing, and is a
common technical issue in ’omics experiments with complex experimental
designs.</p>
<hr />
<p>While to understand the origin of the aliasing it is crucial to
understand the relationship between the variables in the experimental
design, we can also investigate it in detail using the
<code>alias</code> function, to give us an idea.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" tabindex="-1"></a><span class="fu">alias</span>(m)</span></code></pre></div>
<pre><code>## Model :
## y ~ patient + disease * time
## 
## Complete :
##               (Intercept) patientb patientc patientd patiente patientf patientg
## diseasecancer 0           0        0        0        1        1        1       
##               patienth timeafter diseasecancer:timeafter
## diseasecancer 1        0         0</code></pre>
<p>We see that the effect <code>diseasecancer</code> is a linear
combination of the patient-specific effects of the cancer patients. This
makes sense!</p>
<hr />
<p>For clarity, let’s reproduce this using our design matrix.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> patient <span class="sc">+</span> disease<span class="sc">*</span>time) <span class="co"># this is the design used in glm()</span></span>
<span id="cb77-2"><a href="#cb77-2" tabindex="-1"></a></span>
<span id="cb77-3"><a href="#cb77-3" tabindex="-1"></a><span class="do">## these are indeed identical.</span></span>
<span id="cb77-4"><a href="#cb77-4" tabindex="-1"></a>X[,<span class="st">&quot;diseasecancer&quot;</span>]</span></code></pre></div>
<pre><code>##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 
##  0  0  0  0  0  0  0  0  1  1  1  1  1  1  1  1</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" tabindex="-1"></a>X[,<span class="st">&quot;patiente&quot;</span>] <span class="sc">+</span> X[,<span class="st">&quot;patientf&quot;</span>] <span class="sc">+</span> X[,<span class="st">&quot;patientg&quot;</span>] <span class="sc">+</span> X[,<span class="st">&quot;patienth&quot;</span>]</span></code></pre></div>
<pre><code>##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 
##  0  0  0  0  0  0  0  0  1  1  1  1  1  1  1  1</code></pre>
<p>Since one of our parameters is a linear combination of other
parameters, it cannot be estimated simultaneously with the other
parameters. In this case, we can actually drop the <code>disease</code>
main effect from the model, since we know that it is already included in
the <code>patient</code> effect.</p>
<hr />
<p>We will have to carefully construct our design matrix in order to
account for all important sources of variation while still allowing us
to answer the research question of interest. The aliasing exploration
above has made it clear we may drop the <code>disease</code> main
effect, so let’s start by constructing this design matrix.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> patient <span class="sc">+</span> time <span class="sc">+</span> disease<span class="sc">:</span>time)</span>
<span id="cb81-2"><a href="#cb81-2" tabindex="-1"></a></span>
<span id="cb81-3"><a href="#cb81-3" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> X,</span>
<span id="cb81-4"><a href="#cb81-4" tabindex="-1"></a>         <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>)</span>
<span id="cb81-5"><a href="#cb81-5" tabindex="-1"></a><span class="fu">summary</span>(m2)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = y ~ -1 + X, family = &quot;poisson&quot;)
## 
## Coefficients: (1 not defined because of singularities)
##                           Estimate Std. Error z value Pr(&gt;|z|)    
## X(Intercept)               3.76900    0.11916  31.631   &lt;2e-16 ***
## Xpatientb                  0.06744    0.14999   0.450   0.6530    
## Xpatientc                  0.06744    0.14999   0.450   0.6530    
## Xpatientd                  0.27304    0.14310   1.908   0.0564 .  
## Xpatiente                  0.28823    0.16077   1.793   0.0730 .  
## Xpatientf                  0.14939    0.16500   0.905   0.3653    
## Xpatientg                  0.10590    0.16643   0.636   0.5246    
## Xpatienth                  0.18081    0.16400   1.102   0.2703    
## Xtimeafter                -0.01567    0.10220  -0.153   0.8782    
## Xtimebefore:diseasecancer -0.12374    0.14407  -0.859   0.3904    
## Xtimeafter:diseasecancer        NA         NA      NA       NA    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 4489.2752  on 16  degrees of freedom
## Residual deviance:    8.8417  on  6  degrees of freedom
## AIC: 120.16
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" tabindex="-1"></a><span class="fu">alias</span>(m2)</span></code></pre></div>
<pre><code>## Model :
## y ~ -1 + X
## 
## Complete :
##                          X(Intercept) Xpatientb Xpatientc Xpatientd Xpatiente
## Xtimeafter:diseasecancer  0            0         0         0         1       
##                          Xpatientf Xpatientg Xpatienth Xtimeafter
## Xtimeafter:diseasecancer  1         1         1         0        
##                          Xtimebefore:diseasecancer
## Xtimeafter:diseasecancer -1</code></pre>
<p>We are still confronted with aliasing as the model matrix contains an
interaction effect <code>timebefore:diseasecancer</code> as well as
<code>timeafter:diseasecancer</code>, while only the latter is relevant.
Indeed, we know that we can derive the
<code>timebefore:diseasecancer</code> effect by averaging the patient
effects of the cancer patients.</p>
<hr />
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" tabindex="-1"></a>X <span class="ot">&lt;-</span> X[,<span class="sc">!</span><span class="fu">colnames</span>(X) <span class="sc">%in%</span> <span class="st">&quot;timebefore:diseasecancer&quot;</span>]</span>
<span id="cb85-2"><a href="#cb85-2" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" tabindex="-1"></a></span>
<span id="cb85-4"><a href="#cb85-4" tabindex="-1"></a><span class="do">## fit a Poisson model</span></span>
<span id="cb85-5"><a href="#cb85-5" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(y <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> X,</span>
<span id="cb85-6"><a href="#cb85-6" tabindex="-1"></a>         <span class="at">family =</span> <span class="st">&quot;poisson&quot;</span>)</span>
<span id="cb85-7"><a href="#cb85-7" tabindex="-1"></a><span class="fu">summary</span>(m2)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = y ~ -1 + X, family = &quot;poisson&quot;)
## 
## Coefficients:
##                          Estimate Std. Error z value Pr(&gt;|z|)    
## X(Intercept)              3.76900    0.11916  31.631   &lt;2e-16 ***
## Xpatientb                 0.06744    0.14999   0.450   0.6530    
## Xpatientc                 0.06744    0.14999   0.450   0.6530    
## Xpatientd                 0.27304    0.14310   1.908   0.0564 .  
## Xpatiente                 0.16449    0.16224   1.014   0.3107    
## Xpatientf                 0.02565    0.16644   0.154   0.8775    
## Xpatientg                -0.01784    0.16785  -0.106   0.9154    
## Xpatienth                 0.05706    0.16544   0.345   0.7302    
## Xtimeafter               -0.01567    0.10220  -0.153   0.8782    
## Xtimeafter:diseasecancer  0.12374    0.14407   0.859   0.3904    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for poisson family taken to be 1)
## 
##     Null deviance: 4489.2752  on 16  degrees of freedom
## Residual deviance:    8.8417  on  6  degrees of freedom
## AIC: 120.16
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>We see that all coefficients can now be estimated. The
<code>timeafter</code> effect may be interpreted as the time effect for
healthy patients, while the <code>timeafter:diseasecancer</code> effect
may be interpreted as the difference in the time effect for cancer
patients as compared to healthy patients, i.e., it is the relevant
interaction effect we are interested in.</p>
</div>

<div id="rmd-source-code">---
title: 'Sequencing: Selected technical topics'
author: "Lieven Clement & Koen Van den Berge"
date: "Last edited on `r format(Sys.time(), '%d %B, %Y')`"
output: 
  html_document:
    toc: true
    toc_float: true
  pdf_document:
    toc: true
    number_sections: true
    latex_engine: xelatex
always_allow_html: true
linkcolor: blue
urlcolor: blue 
citecolor: blue
link-citations: yes

---

```{r functions, include=FALSE}
# A function for captioning and referencing images
fig <- local({
    i <- 0
    ref <- list()
    list(
        cap=function(refName, text) {
            i <<- i + 1
            ref[[refName]] <<- i
            paste("Figure ", i, ": ", text, sep="")
        },
        ref=function(refName) {
            ref[[refName]]
        })
})
```

```{r, echo=FALSE, message=FALSE, eval=TRUE}
suppressPackageStartupMessages({
  library(knitr)
  library(rmarkdown)
  library(ggplot2)
  library(printr)
  library(gridExtra)
  library(tidyverse)
  library(plotly)
})
```

# Parameter Estimation and Inference in Generalized linear models 

## Simulate Poisson Data

- We simulate data for 100 observations.
- Covariates x are simulated from normal distribution
- The $\beta$ are chosen at $\beta_0=2$, $\beta_1=0.8$, $\beta_2=1.2$

```{r}
set.seed(300)
xhlp<-cbind(1,rnorm(100),rnorm(100))
betasTrue<-c(2,0.8,1.2)
etaTrue<-xhlp%*%betasTrue
y<-rpois(100,exp(etaTrue))
data.frame(coef =0:2,betasTrue=betasTrue) %>% 
  ggplot(aes(x=coef,y=betasTrue)) + 
  geom_point() +
  geom_line() +
  ylab("parameter value") +
  xlab("beta") +
  ylim(0,4) +
  theme_bw() +
  scale_x_continuous(breaks=c(0,1,2))


```

```{r echo=FALSE, warning=FALSE}
x1 <- xhlp[,2]
x2 <- xhlp[,3]
p1 <- plot_ly(
    x = ~x1,
    y = ~x2,
    z= ~y,
    mode="markers",
    size=.5) %>%
  add_markers(type="scatter3d") %>%
  layout(
    scene = list(
      aspectmode="cube",
      xaxis = list(range=range(xhlp[,2]))), yaxis = list(range=range(xhlp[,3])), zaxis = list(range=range(y))
      )
p1
```

```{r echo=FALSE, warning=FALSE}
x1 <- xhlp[,2]
x2 <- xhlp[,3]
p1 <- plot_ly(
    x = ~x1,
    y = ~x2,
    z= ~ log2(y+.5),
    mode="markers",
    size=.5) %>%
  add_markers(type="scatter3d") %>%
  layout(
    scene = list(
      aspectmode="cube",
      xaxis = list(range=range(xhlp[,2]))), yaxis = list(range=range(xhlp[,3])), zaxis = list(range=range(y))
      )
p1
```


## Exponential Family

$$
f(y_i\vert \theta_i,\phi)=\exp\left\{ \frac{y_i\theta_i- b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}
$$

with

- $\theta_i$: canonical parameters
- $\phi$: dispersion parameter
- $a(.)$, $b(.)$, $c(.)$: specific functions that depend on the distribution, 

e.g. for normal distribution 

- $\phi=\sigma^2$, 
- $\theta=\mu$, 
- $a(\phi)=\phi=\sigma^2$, 
- $b(\theta_i)=\theta_i^2/2$, 
- $c(y_i,\phi)=-\frac{1}{2}[y^2/\phi+\log(2\pi\phi)]$

### Poisson Distribution

$$
y_i \sim \frac{\mu_i^{y_i}e^{-\mu_i}}{y_i!}
$$

In format of exponential family: 

$$
y_i \sim \exp\left\{y_i\log(\mu_i) - \mu_i - \log(y_i!)\right\}
$$

## Components of Generalized Linear Model

$$
\left\{\begin{array}{ccc}
y_i\vert x_i&\sim&f(y_i\vert{\theta}_i,\phi)\\\\
\text{E}\left[ y_i\vert \mathbf{x}_i\right]&=&\mu_i\\\\
g(\mu_i)&=&\eta(\mathbf{x}_i)\\\\
\eta(\mathbf{x}_i)&=&\mathbf{x}_i^T\boldsymbol{\beta}
\end{array}\right.,
$$
with $g(.)$ the link function, e.g. 

- $g(.)=.$ : identity link for Normal distribution
- $g(.)=\log(.)$ : canonical link for Poisson distribution
- $g(.)=\text{logit}(.)=\log\left[\frac{(.)}{(1-.)}\right]$ : canonical link for Bernouilli distribution.

### Poisson GLM

$$\left\{\begin{array}{lcr}
y_i &\sim& Poisson(\mu_i)\\
E[y_i]&=&\mu_i\\
\log(\mu_i)&=&\eta_i\\
\eta_i&=&\mathbf{x}_i\boldsymbol{\beta}
\end{array}\right.$$

## Likelihood

We start from a sample, and consider it as fixed and known. 

- In particular we do NOT consider the sample observations as random variables. 

- Therefore we write the observed sample as $y_i, . . . , y_n$

- The theory is based on the likelihood function, which  can be interpreted as a measure for the probability that the given sample is observed as a realisation of a sequence of random variables $Y_1, \ldots Y_n$ 

- The random variables $Y_i$ are characterized by a distribution or density function which has typically unknown parameters, e.g. a Poisson distribution $f(Y_i)\sim \text{Poisson}(\theta_i)$.

- When the subjects are mutually independent the joint likelihood to observe $y_1, \ldots, y_n$ equals 
$$\prod\limits_{i=1}^n f(y_i,\theta_i,\phi)$$
- The densities are actually also a function of the parameters $\theta_i, \phi$. To stress this, we indicated that in the density formulation.

- The likelihood function is a function of all parameters 

$$
L(\boldsymbol{\theta},\phi\vert \boldsymbol{y})=\prod\limits_{i=1}^n f(y_i,\theta_i,\phi)
$$

- The log-likelihood function is often used, which is defined as
$$
l(\boldsymbol{\theta},\phi\vert \boldsymbol{y}) = \log L(\boldsymbol{\theta},\phi\vert \boldsymbol{y}) =  \sum\limits_{i=1}^n \log  f(y_i,\theta_i,\phi)
$$
 
### Poisson Example 

The log-likelihood for our simulated dataset given the real model parameters is: 

For one observation:
$$l(\mu_i \vert y_i) = y_i \log \mu_i - \mu_i - \log y_i!$$

- Verify in R.

```{r}
muTrue <- exp(etaTrue)
loglikPois <- dpois(y,muTrue,log = TRUE)
logLikSelf <- y*log(muTrue) - muTrue - lfactorial(y)
qplot(loglikPois,logLikSelf) + theme_bw() + geom_abline(intercept = 0, slope=1)
```

The log-likelihood can also be written in terms of the canonical model parameters $\theta$

$$l(\mu_i \vert y_i) = y_i \theta_i - e^{\theta_i} - \log y_i!$$

- Note that $\theta_i = \eta_i$. The canonical parameter for the poisson equals the linear predictor!
  $$\theta_i=\eta_i=\mathbf{x}_i^t\boldsymbol{\beta}$$

Log-likelihood for all observations, given that they are independent:
$$l(\boldsymbol{\mu} \vert \mathbf{y}) = \sum\limits_{i=1}^n \left\{ y_i \theta_i - e^{\theta_i} - \log y_i!\right\}$$

- Calculate in R: 

```{r}
dpois(y,lambda = muTrue, log = TRUE) %>% 
  sum()
```

### Properties of the Log-Likelihood

$$
l(\theta_i,\phi\vert y_i)=\left\{ \frac{y_i\theta_i- b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}
$$

- $E[y_i]=\mu_i=b^\prime(\theta_i)$
- $\text{var}[y_i]=b^{\prime\prime}(\theta_i) a(\phi)$

Note that, 

- Variance $\text{var}[y_i]$ depends on mean! 
- Often there is no dispersion parameter e.g. Bernouilli: $\text{var}[y_i]=\mu_i(1-\mu_i)$, 

### Poisson Example 

- Canonical model parameter $\theta_i=\log{\mu_i}$.
- $b(\theta_i) = \exp(\theta_i)$
- $c(y_i,\phi) = - \log(y_i!)$
- $\phi = 1$
- $a(\phi)= 1$
- $\mu_i =\frac{\partial b(\theta_i)}{\partial \theta_i}=  \frac{\partial \exp(\theta_i)}{\partial \theta_i}=\exp(\theta_i)$
- $\text{Var}\left[y_i \right]= a(\phi)\frac{\partial^2 b(\theta_i)}{(\partial \theta_i)^2}=  \frac{\partial^2 \exp(\theta_i)}{\partial \theta_i^2}=\exp(\theta_i)$.
- Mean is equal to variance for Poisson!

## Parameter Estimation: Maximum Likelihood

Choose the parameters $\boldsymbol{\beta}$ so that the likelihood to observe the sample under the statistical model is maximized.

It is easier and equivalent to maximize the log-likihood

$$\text{argmax}_{\boldsymbol{\beta}} l(\boldsymbol{\mu} \vert \mathbf{y})$$

$$
\frac{\partial  l(\boldsymbol{\mu} \vert \mathbf{y})}{
\partial{\boldsymbol{\beta}}}=0
$$

$\frac{\partial  l(\boldsymbol{\mu} \vert \mathbf{y})}{
\partial{\boldsymbol{\beta}}}$ is also referred to as the score function.

### Score function

$$
S_i(\theta_i)= \frac{\partial l(\theta_i,\phi\vert y_i)}{\partial \theta_i}
$$

$$
S_i(\theta_i)= \frac{\partial l(\theta_i,\phi\vert y_i)}{\partial \theta_i}=\frac{y_i - \mu_i}{a(\phi)}
$$

when canonical link function is used:

- $\mu_i=b^\prime(\theta_i)$ 


Regression (chain rule and $i=1,\ldots,n$ i.i.d observations)

\begin{eqnarray*}
S(\boldsymbol{\beta})&=&\frac{\partial  \sum\limits_{i=1}^n \left\{ \frac{y_i \theta_i - b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}}{\partial \beta}\\
&=&\sum\limits_{i=1}^n\frac{\partial  \left\{ \frac{y_i \theta_i - b(\theta_i)}{a(\phi)}+c(y_i,\phi)\right\}}{\partial \theta}\frac{\partial \theta}{\partial\mu}\frac{\partial\mu}{\partial\eta}\frac{\partial\eta}{\partial\beta}\\
&=&\sum\limits_{i=1}^n\frac{y_i-\mu_i}{a(\phi)}\frac{1}{b^{\prime\prime}(\theta)}\frac{\partial\mu}{\partial\eta}\mathbf{x}^t\\
&=&\mathbf{X}^T\mathbf{A}\left(\mathbf{y}-\boldsymbol{\mu}\right)
\end{eqnarray*}


- $\boldsymbol{A}$ is a diagonal matrix: $a_{ii}=\left(\text{var}[y_i]\frac{\partial \eta_i}{\partial \mu_i}\right)^{-1}$, $\boldsymbol{y}=[y_1,\ldots, y_n]^T$, $\boldsymbol{\mu}=[\mu_1,\ldots,\mu_n]^T$ 

#### Poisson Example 


For poisson data: $a(\phi)b^{\prime\prime}(\theta)=\mu$ and $\frac{\partial \mu}{\partial \eta}=\mu$. So $\mathbf{A}=\mathbf{I}$ and

$$ S(\beta)=\mathbf{X}^T \left\{\mathbf{Y}-\boldsymbol{\mu}\right\}$$

### Solve Score Equations

Find  parameter estimator  $\hat{\boldsymbol{\beta}}$ so that
$$ S(\boldsymbol{\beta}) = \mathbf{0}$$
$$\mathbf{X}^T\mathbf{A}\left(\mathbf{y}-\boldsymbol{\mu}\right) =\mathbf{0}$$
Problem! Non linear in $\beta$ due to 

- link function: $\boldsymbol{\mu}=h^{-1}(\eta)$ 
- $a_{ii}=\left(\text{var}[y_i]\frac{\partial \eta_i}{\partial \mu_i}\right)^{-1}$

$\rightarrow$ Find roots of score equation by using Newton-Raphson method.

### Newton-Raphson

```{r echo=FALSE, fig.cap= "Newton-Raphson algorithm: The algorithm starts at an initial guess bk for the root of the score function S. It walks along the tangent line of the score function in S(bk)  to move into the direction where the score function becomes zero (and where the log-likelihood function is maximal). The new estimate of the root is taken at bk+1 where the tangent line becomes zero. It then calculates the score for the updated parameter estimator and the whole process is repeated until the root is found."}
S <- function(beta) (beta - 3)^2 
dS <- function(beta) 2*(beta-3)
dSline <- function(beta, betak) dS(betak) *beta + S(betak) - dS(betak) * betak

beta <- seq(-21,15,length=100)
betak <- -20
betak1 <- betak - 1/dS(betak)*S(betak)
betaSeqDk <- seq(betak,betak1,length=100)
pNR <- qplot(beta, S(beta), geom="line") +
  annotate("line",x=betaSeqDk,y=dSline(betaSeqDk,betak),linetype = "dashed",col="red") +
    annotate("line",x=rep(betak1,2),y=c(0,S(betak1)),linetype = "dashed",col="red") +
    annotate("point",x=betak,y=S(betak),col="red") +
  theme_bw()
betaNs <- betak1
for (i in 1:5)
{
betak <- betak1
betak1 <- betak - 1/dS(betak)*S(betak)
betaNs <- c(betaNs,betak1)
betaSeqDk <- seq(betak,betak1,length=100)
pNR <- pNR +
    annotate("line",x=betaSeqDk,y=dSline(betaSeqDk,betak),linetype = "dashed",col="red") +
    annotate("line",x=rep(betak1,2),y=c(0,S(betak1)),linetype = "dashed",col="red") 
}
pNR + 
  annotate("text",x=-20,y=S(-20)+40,label=paste0("S(beta[k])"),col="red",parse=TRUE) +
  annotate("text",x=betaNs[1:2],y=S(betaNs[1:2])+40,label=paste0("S(beta[k+",1:2,"])"),col="red",parse=TRUE)   +
  annotate("text",x=betaNs[1:3],y=rep(-40,3),label=paste0("beta[k+",1:3,"]"),col="red",parse=TRUE) +
  annotate("point",x=betaNs,y=S(betaNs),col="red") +
  annotate("text",x=-20,y=400, label="frac(partialdiff*S, partialdiff*beta)*' |'*beta[k]", col="red",parse=TRUE) +
  xlab(expression(beta)) +
  ylab(expression(S(beta)))
```

Newton Raphson algorithm to find the root of the score function. 

1. Choose initial parameter estimate $\boldsymbol{\beta}^k=\boldsymbol{\beta}^0$
2. Calculate score $S(\boldsymbol{\beta})\vert_{\boldsymbol{\beta}=\boldsymbol{\beta}^k}$
3. Calculate derivative of the function for which you want to calculate the roots
4. Walk along first derivative until line (plane) of the derivative crosses zero
5. Update the betas $\boldsymbol{\beta}^{k+1}$
6. Iterate from step 2 - 5 until convergence.


#### Derivative of Score 

We have to implement an interative algorithm for optimisation. To make things tractable we will act as if $\mathbf{A}$ is known and fix it using the current values of $\boldsymbol{\beta}^k$. Note, that for Poisson regression $\mathbf{A}=\mathbf{I}$.

\begin{eqnarray*}
\frac{\partial S(\boldsymbol{\beta})}{\partial \boldsymbol{\beta}} &=& \frac{\partial \mathbf{X}^T\mathbf{A} \left\{\mathbf{Y}-\boldsymbol{\mu}\right\}}{\partial \boldsymbol{\beta}}\\
&\approx& - \mathbf{X}^T \mathbf{A}\left[
\begin{array}{cccc} \frac{\partial \mu_1}{\partial \eta_1} &0&\ldots&0\\
 0&\frac{\partial \mu_2}{\partial \eta_2} &\ldots&0\\
\vdots&\vdots&\vdots&\vdots\\
0&0&\ldots& \frac{\partial \mu_n}{\partial \eta_n}\\
\end{array}\right] \frac{\partial \boldsymbol{\eta}}{\partial \boldsymbol{\beta}}\\
&\approx&-\mathbf{X}^T\mathbf{WX}
\end{eqnarray*}

#### Define equation of Tangent Line (Plane)

- We know two points of the tangent plane $(\boldsymbol{\beta}^k,S(\boldsymbol{\beta}^k))$ and $(\boldsymbol{\beta}^{k+1},0)$
- We know the direction of the plane $S^\prime(\boldsymbol{\beta})=\frac{\partial S(\boldsymbol{\beta})}{\partial \boldsymbol{\beta}}$
- Equation of Plane:
$$S(\boldsymbol{\beta})={\alpha}_0+S^\prime\vert_{\boldsymbol{\beta}^k} \boldsymbol{\beta}$$

- Get $\boldsymbol{\beta}_{k+1}$
\begin{eqnarray*}
\mathbf{0}&=&{\alpha}_0+S^\prime\vert_{\boldsymbol{\beta}^{k}} \boldsymbol{\beta}^{k+1}\\
\boldsymbol{\beta}^{k+1}&=&-\left(S^{\prime}\vert_{\boldsymbol{\beta}^{k}}\right)^{-1}{\alpha}_0\\
\end{eqnarray*}

- Get ${\alpha}_0$
\begin{eqnarray*}
S(\boldsymbol{\beta}^k)&=&\boldsymbol{\alpha}_0+S^\prime\vert_{\boldsymbol{\beta}^k} \boldsymbol{\beta}^k\\
{\alpha}_0&=&-S^\prime\vert_{\boldsymbol{\beta}^k} \boldsymbol{\beta}^k + S(\boldsymbol{\beta}^k)\\
\end{eqnarray*}

- Get $\boldsymbol{\beta}_{k+1}$

\begin{eqnarray*}
\boldsymbol{\beta}^{k+1}&=&\boldsymbol{\beta}^k-\left(S^{\prime}\vert_{\boldsymbol{\beta}^{k}}\right)^{-1}S(\boldsymbol{\beta}^k)\\
\boldsymbol{\beta}^{k+1}&=&\boldsymbol{\beta}^k+ \left(\mathbf{X}^T\mathbf{WX}\right)^{-1} S(\boldsymbol{\beta}^k)
\end{eqnarray*}

With $J(\boldsymbol{\beta})=I(\boldsymbol{\beta})=\mathbf{X}^T\mathbf{WX}$  the Fisher information matrix.

### Fisher Scoring

Because we use the canonical model parameters the observed  Fisher information matrix equals the expected  Fisher information matrix $J(\boldsymbol{\beta})=I(\boldsymbol{\beta})$.
Indeed, the observed Fisher information matrix is not depending on the observations, but only on the design and the variance of the data (via the weights).

Hence, Newton-Raphson is equivalent to Fisher scoring when the canonical link function is used.

Note, that the Fisher matrix, minus second derivative (or hessian) of the likelihood to the model parameters, is also the inverse of the variance covariance matrix of the model parameters. It is thus related to the precision.


### Iteratively Reweighted Least Squares (IRLS).

We can rewrite Newton Raphson or Fisher scoring as IRLS.

\begin{eqnarray*}
\boldsymbol{\beta}^{k+1}&=&\boldsymbol{\beta}^k+ \left(\mathbf{X}^T\mathbf{WX}\right)^{-1} S(\boldsymbol{\beta}^k)\\
\boldsymbol{\beta}^{k+1}&=&\boldsymbol{\beta}^k+ \left(\mathbf{X}^T\mathbf{WX}\right)^{-1} \mathbf{X}^T\mathbf{A} \left(\mathbf{Y}-\boldsymbol{\mu}\right)\\
\boldsymbol{\beta}^{k+1}&=& \left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\mathbf{X}^T\mathbf{WX}\boldsymbol{\beta}^k+ \left(\boldsymbol{X}^T\mathbf{WX}\right)^{-1} \mathbf{X}^T \mathbf{W}\frac{\partial \eta}{\partial \mu}  \left(\mathbf{Y}-\boldsymbol{\mu}\right)\\
\boldsymbol{\beta}^{k+1}&=& \left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\mathbf{X}^T\mathbf{W} \left[\mathbf{X}\boldsymbol{\beta}^k + \frac{\partial \eta}{\partial \mu}  \left(\mathbf{Y}-\boldsymbol{\mu}\right)
\right]\\
\boldsymbol{\beta}^{k+1}&=& \left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\mathbf{X}^T\mathbf{Wz}
\end{eqnarray*}

with $\mathbf{z}=\left[\mathbf{X}\boldsymbol{\beta}^k + \frac{\partial \eta}{\partial \mu}  \left(\mathbf{Y}-\boldsymbol{\mu}\right)\right]$

So we can fit the model by performing iterative regressions of the pseudo data $\mathbf{z}$ on $\mathbf{X}$.
In each iteration we will update $\mathbf{z}$, the weights $\mathbf{W}$ and the model parameters.

For Poisson data 

- $\frac{\partial \eta}{\partial \mu}=\frac{\partial\log \mu}{\partial\mu}=\frac{1}{\mu}=\exp(-\eta)$  
- $\mathbf{W}=\mathbf{A}\frac{\partial{\mu}}{{\partial \eta}}$ is a diagonal matrix with $[\frac{\partial{\mu_i}}{{\partial \eta_i}}]_{ii}=[\mu_i]_{ii}=[\exp(\eta_i)]_{ii}$ on its diagonal elements.


### Variance-Covariance Matrix of Mean Model Parameters?

In the IRWLS algorithm, the data is weighted according to the variance of $\mathbf{Y}$. We correct for the fact that the data are heteroscedastic.

Count data have a mean variance relation (e.g. in Poisson case $\text{E}\left[Y \right]=\text{var}\left[Y \right]=\mu$).
The IRWLS also corrects for the scale parameter $\phi$ in $\mathbf{W}$. (Note that the scale parameter for Poisson is $\phi=1$).

So IRWLS the variance-covariance matrix for the model parameter equals
$$\mathbf{\Sigma}_{\hat\beta}=\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}.$$

Note, that the Fisher Information Matrix equals the inverse of the variance-covariance matrix of the experiment.
The larger the Fisher Information Matrix the more information we have on the experiment to estimate the model parameters.
FIM $\uparrow$, precision $\uparrow$, $\text{SE}\downarrow$

## Poisson Example 


### Initial Estimate

This is a very poor initial estimate used to illustrate the algorithm.
Otherwise convergence for this simple example is way too quick

```{r}
iteration=0
betas<-c(log(mean(y)),0,0)
plot(betasTrue,
     ylab=expression(beta),
     ylim=c(0,4),
     pch=19,
     type="b", 
     main=paste0("likelihood real beta=",
                 round(sum(dpois(y,exp(etaTrue),log=TRUE)),1),"\nlikelihood fit=", round(sum(dpois(y,exp(xhlp%*%betas),log=TRUE)),1))
)
lines(betas,type="b",lty=2 )
```

### Iteratively Reweighted Least Squares

#### Pseudo Data

$$z_i= \eta_i + \frac{\partial \eta_i}{\partial \mu_i}(y_i -\mu_i)$$
$$z_i= \eta_i + e^{-\eta_i} y_i -1$$

#### Weight Matrix?

$$[w_{ii}]= var_{y_i}^{-1} \left(\frac{\partial \mu}{\partial \eta}\right)^2$$
$$[w_{ii}]= e^{\eta_i}$$

#### Run Update Step Multiple Times

First 3 times (colors are black 0, red iteration 1, green iteration 2, blue iteration 3)

```{r}
plot(betasTrue,ylab=expression(beta),ylim=c(0,4),pch=19,type="b")
lines(betas,type="b",lty=2)

cat("\nlikelihood TRUE=", round(sum(dpois(y,exp(xhlp%*%betasTrue),log=TRUE)),1))
cat("\nlikelihood initial fit=", round(sum(dpois(y,exp(xhlp%*%betas),log=TRUE)),1))

#Calculate current eta
eta<-xhlp%*%betas

iteration=0
for (i in 1:3)
{
#start IRLS UPDATE STEP
iteration=iteration+1
#calculate pseudo data based on current betas
z=eta+exp(-eta)*(y-exp(eta))
#calculate new weights: diagonal elements
w<-c(exp(eta))

#update betas
lmUpdate<-lm(z~-1+xhlp,weight=w)
#eta<-xhlp%*%betas
eta<-lmUpdate$fitted
betas<-lmUpdate$coef
lines(betas,type="b",col=iteration+1,pch=iteration,lty=2)
cat("\nlikelihood current fit=", round(sum(dpois(y,exp(xhlp%*%betas),log=TRUE)),1))
}
```


## Comparison with GLM Function

### Smarter Initialisation
```{r}
z<-log(y+.5)
betas<-lm(z~-1+xhlp)$coef
plot(betasTrue,ylab=expression(beta),ylim=c(0,4),pch=19,type="b")
lines(betas,col=2,type="b",lty=2)
#calculate current eta
eta<-xhlp%*%betas

cat("\nlikelihood TRUE=", round(sum(dpois(y,exp(xhlp%*%betasTrue),log=TRUE)),1))
cat("\nlikelihood initial fit=", round(sum(dpois(y,exp(xhlp%*%betas),log=TRUE)),1))
```

### Evaluation Stopping Criterion

- Residual deviance: Is 2 log of LR between best possible fit and current fit
$$LR=\frac{L_\text{best}}{L_\text{current}}$$
$$D=2 (\log L_\text{best}- \log L_\text{current})$$
$$D=2 (l_\text{best}-l_\text{current})$$
- Best fit: $\mu=y$
- Optimal poisson:
$$ l_\text{best}=\sum\left[y_i \log(y_i) - y_i - \log\left(y_i!\right)\right]$$
- Current fit
$$ l_\text{current}=\sum \left[y_i \eta_i -e^{\eta_i} - log\left(y_i!\right)\right]$$
- Deviance D:
$$D = 2 \sum \left[ y_i log(y_i) - y_i \eta_i - (y_i -e^{\eta_i})\right]$$
- Problem to calculate it if y=0 but by apply l'Hopital's rule we know
$$\lim_{y_i \to 0} y_i \log(y_i) =0$$

```{r}
ylogy<-function(y)
{
return(ifelse(y==0,rep(0,length(y)),y*log(y)))
}

deviance<-2*sum(ylogy(y)-y*eta-(y-exp(eta)))

devianceOld<-1e30
```


### Run Update Step until Convergence

```{r}
plot(betasTrue,ylab=expression(beta),ylim=c(0,4),pch=19,type="b")
lines(betas,type="b",lty=2)

tol<-1e-6
iteration=0
while(((devianceOld-deviance)/devianceOld)>tol)
{
#start IRLS UPDATE STEP
iteration=iteration+1
#calculate pseudo data based on current betas
z=eta+exp(-eta)*(y-exp(eta))
#calculate new weights: diagonal elements
w<-c(exp(eta))

#update betas
lmUpdate<-lm(z~-1+xhlp,weight=w)
#eta<-xhlp%*%betas
eta<-lmUpdate$fitted
betas<-lmUpdate$coef
lines(betas,type="b",col=iteration+1,pch=iteration,lty=2)

#criterion for convergence
devianceOld<-deviance
deviance<-2*sum(ylogy(y)-y*eta-(y-exp(eta)))
cat("iteration",iteration,"Deviance Old",devianceOld,"Deviance", deviance,"\n")
}
```

### Variance $\beta$?

$$\Sigma_{\beta}=\left(\mathbf{X}^T\mathbf{W} \mathbf{X}\right)^{-1}$$

```{r}
varBeta=solve(t(xhlp)%*%diag(w)%*%xhlp)
```

### Comparison with GLM fit

Use -1 because intercept is already in xhlp
```{r}
glmfit=glm(y~-1+xhlp,family=poisson)
comp=data.frame(glmfit=c(glmfit$deviance,glmfit$coef,summary(glmfit)$coef[,2]),ourFit=c(deviance,betas,sqrt(diag(varBeta))))
row.names(comp)=c("deviance",paste("beta",1:3,sep=""),paste("se",1:3,sep=""))
comp
```


## Hypothesis testing: Large sample theory

### Wald test

- Follows immediately from the information matrix for generalized linear models
 \[I(\boldsymbol{\beta}) = \mathbf{X}^T\mathbf{WX}\] 
 
 so large sample distribution of the maximum likelihood estimator $\hat{\boldsymbol{\beta}}$ is multivariate normal
\[
\hat{\boldsymbol{\beta}} \sim MVN\left[\boldsymbol{\beta},\left(\mathbf{X}^T\mathbf{WX}\right)^{-1}\right]
\]

We can perform a Wald Test for a single model parameter 

$$
W = \frac{\hat\beta_m}{\hat{\text{se}}_{\hat\beta_m}} \approx N(0,1)\vert H_0
$$
to test for 

$$
H_0: \beta_p=0 \leftrightarrow H_1:\beta_p\neq0
$$


Again, we can also assess contrasts! Indeed, linear combinations of model parameter estimators also follow a normal distribution. 

$$
\mathbf{L}^T\hat{\boldsymbol{\beta}} \sim N\left[\mathbf{L}^T\boldsymbol{\beta},\mathbf{L}^T\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}\right]
$$

With $\mathbf{L}$ a vector for a single contrast. 

$$
W = \frac{\mathbf{L}^T\hat{\boldsymbol{\beta}}}{\hat{\text{se}}_{\mathbf{L}^T\hat{\boldsymbol{\beta}}}} \approx N(0,1)\vert H_0
$$
testing for 
$$
H_0: \mathbf{L}^T\boldsymbol{\beta}=0 \leftrightarrow H_1:\mathbf{L}^T\boldsymbol{\beta}\neq0
$$

We can also test for multiple contrasts simultaneously, e.g. by assuming that multiple model parameters are zero. Suppose that $\mathbf{L}$ is the contrast matrix that corresponds testing for $c$ model parameters, simultaneously.  Then 

$$
\mathbf{L}^T\hat{\boldsymbol{\beta}} \sim MVN\left[\mathbf{L}^T\boldsymbol{\beta},\mathbf{L}^T\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}\right]
$$

and 

$$ 
W = \mathbf{L}^T\hat{\boldsymbol{\beta}}\left(\mathbf{L}^T\hat{\boldsymbol{\Sigma}}_{\hat{\boldsymbol{\beta}}}\mathbf{L}\right)^{-1}\hat{\boldsymbol{\beta}}\mathbf{L} \sim \chi^2_c\vert H_0
$$
to test for 

$$
H_0: \mathbf{L}^T\boldsymbol{\beta}=\mathbf{0} \leftrightarrow H_1:\mathbf{L}^T\boldsymbol{\beta}=\mathbf{0}\neq0
$$

In general, when we test for $c\geq1$ contrasts, then the test statistic $W∼\chi^2_r|H_0$, with $r$ the rank of the contrast matrix.


###  Likelihood ratio test

The likelihood ratio test (LRT) measures the discrepancy in log-likelihood between our current model (sometimes also referred to as full model) and a reduced model (sometimes also referred to as null or alternative model). 

The reduced model must be nested in (and therefore of lower dimension as compared to) the full model. 

While adding more covariates will always explain more variability in our response variable, the LRT tests whether this is actually significant. 

For example, in the example of gene differential expression between healthy versus tumoral tissue, the full model could be a GLM where the mean is modeled according to an intercept and a tissue indicator variable (healthy / tumor), while the alternative model could be a GLM with just an intercept. Indeed, if the gene is similarly expressed between healthy and tumor tissue, the log-likelihood of the alternative model will decrease only a little as compared to the full model. 

As the name suggests, the likelihood ratio test assesses whether the ratio of the log-likelihoods provides sufficient evidence for a worse fit of the alternative versus full model

 $$
 \lambda=2\left[l(\hat{\boldsymbol{\beta}}_\text{full})-2l(\hat{\boldsymbol{\beta}}_\text{0})\right]
 $$ 
 
    Asymptotically, under the null hypothesis it can be shown that
   $$ L \sim \chi_c^2 | H_0, $$
   with $c$ the number of parameters dropped in the alternative model versus the full model. 
   
Let $\mathbf{C}$ denote the $c \times p$ contrast matrix denoting the contrast for the parameters being dropped, the null and alternative hypothesis are as in the Wald test setting:
   $$ H_0: \mathbf{C} \beta = 0$$
   $$ H_1: \mathbf{C} \beta \ne 0$$
   

 - It is important to keep in mind that standard statistical inference theory in GLMs works **asymptotically in terms of the sample size**.

Thus we need many data points in order for the theory to hold in practice. In order for the $p$-values to be correct, our parametric (distributional) assumptions as well as the independence assumption, must also hold.

 - In bulk RNA-seq, we are often working with a limited number of samples and so we typically do not expect asymptotic theory to hold yet. In single-cell RNA-seq, we often perform several preprocessing steps before calculating $p$-values for each gene and so we may be 'using the data multiple times'. Rather than attaching strong probabilistic interpretations to the $p$-values, we therefore advice to view the $p$-values simply as useful numerical summaries for ranking the genes for further inspection in genomics applications.
 
```{r echo=FALSE, fig.width=0.5, fig.caption="Difference between Wald and LRT test",fig.align='center'}
include_graphics("./images_sequencing/likTests.png")
```
 

# EdgeR

[McCarthy and Smyth, 2012](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3378882/)

```{r fig.width=.99, echo=FALSE}
knitr::include_graphics("./figs/McCarthySmyth2012.png")
```


```{r fig.width=.99, echo=FALSE}
knitr::include_graphics("./figs/McCarthySmyth2012_testing.png")
```

# EdgeR - Quasi-Likelihood

[Lund et al. 2012](https://publications.wehi.edu.au/documentaspdf/256.pdf)

For quasi-likelihood we do not specify the full distribution, only the first two moments: the mean and variance.

$$
\left\{
\begin{array}{lcl}
E[y_{ig}\vert \mathbf{x}_{ig}]&=&\mu_{ig}\\
log(\mu_{ig})&=&\eta_{ig}\\
\eta_{ig}&=& \mathbf{x}_{i}^T\boldsymbol{\beta}+ \log N_i\\
\text{Var}[y_{ig}\vert \mathbf{x}_{ig}]&=&\sigma^2_g\left(\mu_{ig}+\phi\mu_{ig}^2\right)
\end{array}\right.
$$

We will look-up the details in the paper. 




# Limma - Voom

[Law et al. (2013). Genome Biology](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29)

- Count models vs transformation: Poisson counts, $\sqrt(y)$  stabilises the variance, insufficient for negative binomial. 
Log transformation: the transformed data are still heteroscedastic.$\rightarrow$ limma-voom
- Use normalized log-cpm Limma pipeline for sequencing

Problem: counts have a mean variance relationship: heteroscedastic

How do we deal with heteroscedasticity in traditional linear models?

Two stage approach:

1. Stage I

  - OLS
  - Estimate variances at each data point
  - Use variances as weights: $W=\text{diag}[1/\hat\sigma_i^2]$

2. Stage II WLS $\text{argmin}_{\boldsymbol{\beta}} \{ (\mathbf{y}-\mathbf{X}\boldsymbol{\beta})^T\mathbf{W} (\mathbf{y}-\mathbf{X}\boldsymbol{\beta})\}$


Port this idea to RNA-seq pipeline

```{r fig.width=.99, echo=FALSE}
knitr::include_graphics("./figs/limmaVoomPaperChunk1.png")
```

```{r fig.width=.99, echo=FALSE}
knitr::include_graphics("./figs/limmaVoomPaperChunk2.png")
```

```{r fig.width=.99, echo=FALSE}
knitr::include_graphics("./figs/limmaVoomPaperFig2.png")
```



# Independent Filtering

Independent filtering is a strategy to remove features (in this case, genes) prior to the analysis. Removal of these features may lower the multiple testing correction for other genes that pass the filter. We try to remove genes that have a low power to be found statistically significant, and/or that are biologically less relevant. 
A common filtering strategy is to remove genes with a generally low expression, as low counts have lower relative uncertainty (hence lower statistical power), and may be considered biologically less relevant.

Implementation in edgeR. 

```{r}
?edgeR::filterByExpr
```

```{r warning=FALSE, message=FALSE}
suppressPackageStartupMessages({
  library(limma)
  library(edgeR)
  library(DESeq2)
})

dds <- makeExampleDESeqDataSet()
simCounts <-counts(dds)
group <- dds$condition
dge <- edgeR::DGEList(simCounts)
design <- model.matrix(~group)
keep <- filterByExpr(dge, design)
table(keep)
```

```{r}
lib.size <- dge$samples$lib.size * dge$samples$norm.factors
cpmMinCount <- 10/median(lib.size)*1e6
summary(group)
minSampSize <- min(summary(group))
minSampSize
keep <- rowSums(cpm(dge) > cpmMinCount) >= minSampSize
table(keep)
```

```{r}
leverage <- design%*% solve(t(design)%*%design)%*%t(design) %>%diag()
1/leverage
min(1/leverage)
```

Independent filtering has been formalized by [Bourgon *et al.* (2010)](https://www.pnas.org/content/107/21/9546).

```{r, echo=FALSE, fig.cap=paste("Figure 1 from Bourgon *et al.* (2010)."), fig.width=0.99}
# All defaults
include_graphics("./images_sequencing/independentFiltering.png")
```

 ---

The concept of independent filtering can be summarized as follows:

 - For each feature we calculate two statistics, $S_F$ and $S_T$, respectively used for two stages: filtering and testing (e.g., differential expression).
 - In order for a feature to be deemed significant, both of its statistics must be greater than some cut-off.
 - We want to control the type I error rate of the second stage (testing). But note that **the second stage is conditional on the first stage**, as we only test features passing the filter, and basically ignore the fact that filtering was performed. Indeed, one criticism is that computing and correcting the $p$-values as if filtering had not been performed may lead to overoptimistic adjusted $p$-values.
 - [Bourgon *et al.* (2010)](https://www.pnas.org/content/107/21/9546) show that filtering is only appropriate (i.e., does not inflate type I error rate) if the conditional null distribution of test statistics for features passing the filter is the same as the unconditional null distribution. Therefore, **filtering is appropriate if the statistic used for filtering is independent of the statistic used for testing under the null hypothesis**.
 
```{r, echo=FALSE, fig.cap=paste("Figure 2 from Bourgon *et al.* (2010)."), fig.width=0.99,fig.align='center'}
# All defaults
include_graphics("./images_sequencing/independentFiltering2.png")
```
  ---
  
Let's try a couple of examples to get some intuition using simulated data.

```{r warning=FALSE, message=FALSE}
suppressPackageStartupMessages(library(DESeq2))
set.seed(24)
dds <- DESeq2::makeExampleDESeqDataSet()
simCounts <- counts(dds)
group <- dds$condition
```

## A Dependent Test Statistic

```{r}
filterStatEffectSize <- abs(rowMeans(simCounts[,group == "A"]) - rowMeans(simCounts[,group == "B"]))
testStat <- genefilter::rowttests(simCounts, group)

## unconditional distribution
plot(density(testStat$statistic, na.rm=TRUE),
     xlab = "Test statistic",
     main = "Unconditional distribution")

## conditional distribution: very different!
mean(filterStatEffectSize > 1)
hist(filterStatEffectSize, breaks=40)
abline(v=1, col="red")
keepEffectSize <- filterStatEffectSize > 1
plot(density(testStat$statistic[keepEffectSize], na.rm=TRUE),
     xlab = "Test statistic",
     main = "Conditional distribution")
```


## An Independent Test Statistic

```{r}
filterStatGlobalMean <- rowMeans(simCounts)
mean(filterStatGlobalMean > 5) # we remove a similar fraction
keepGlobalMean <- filterStatGlobalMean > 5

## unconditional distribution
plot(density(testStat$statistic, na.rm=TRUE),
     xlab = "Test statistic",
     main = "Unconditional distribution")

## conditional distribution: the same.
plot(density(testStat$statistic[keepGlobalMean], na.rm=TRUE),
     xlab = "Test statistic",
     main = "Conditional distribution")
```


# Normalization

Normalization is necessary to correct for several sources of technical variation:

 - **Differences in sequencing depth** between samples. Some samples get sequenced deeper in the sense that they consist of more (mapped) reads and therefore can be considered to contain a higher amount of information, which we should be taking into account. In addition, if a sample is sequenced deeper, it is natural that the counts for each gene will be higher, jeopardizing a direct comparison of the expression counts.
 - **Differences in RNA population composition** between samples. As an extreme example, suppose that two samples have been sequenced to the exact same depth. One sample is contaminated and has a very high concentration of the contaminant cDNA being sequenced, but otherwise the two samples are identical. Since the contaminant will be taking up a significant proportion of the reads being sequenced, the counts will not be directly comparable between the samples. Hence, we may also want to correct for differences in the composition of the RNA population of the samples.
 - **Other technical variation** such as sample-specific GC-content or transcript length effects may also be accounted for.
 
 
```{r}
data("parathyroidGenesSE", package="parathyroidSE")
se1 <- parathyroidGenesSE
rm(parathyroidGenesSE)

colData(se1) %>% 
  as.data.frame() %>% 
  filter(duplicated(experiment)) 
```

There are technical repeats in the data. 

We mentioned previous lectures that we can sum over technical repeats, because techical repeats are Poisson and the sum of two Poisson variables is again Poisson. 

```{r}
dupExps <- colData(se1) %>% 
  as.data.frame() %>% 
  filter(duplicated(experiment))  %>% 
  pull(experiment)

counts <- assays(se1)$counts
newCounts <- counts
cd <- colData(se1)
for(ss in 1:length(dupExps)){
  # check which samples are duplicates
  relevantId <- which(colData(se1)$experiment == dupExps[ss])
  # sum counts
  newCounts[,relevantId[1]] <- rowSums(counts[,relevantId])
  # keep which columns / rows to remove.
  if(ss == 1){
    toRemove <- relevantId[2]
  } else {
    toRemove <- c(toRemove, relevantId[2])
  }
}

# remove after summing counts (otherwise IDs get mixed up)
newCounts <- newCounts[,-toRemove]
newCD <- cd[-toRemove,]

# Create new SummarizedExperiment
se <- SummarizedExperiment(assays = list("counts" = newCounts),
                            colData = newCD,
                            metadata = metadata(se1))

treatment <- colData(se)$treatment
table(treatment)
```

```{r}
qplot(colSums(assays(se)$counts)/1e6, geom="histogram", bins=10,col="black") +
  theme(legend.position = "none") +
  xlab("libsize (million reads)")

qplot(
  colData(se)$treatment:colData(se)$time,
  colSums(assays(se)$counts)/1e6,geom="boxplot"
  ) +
  xlab("treatment")+
  ylab("libsize (million reads)")

qplot(
  colData(se)$patient,
  colSums(assays(se)$counts)/1e6,geom="boxplot"
  ) +
  xlab("Patient")+
  ylab("libsize (million reads)")
```

```{r}
ma2Samp <- function(countMx,libSize=NULL) {
stopifnot("`countMx` is not a matrix with two columns" = ncol(countMx) == 2)
A <- countMx %>% log2 %>% rowMeans
if(is.null(libSize)) 
  M <- countMx %>% log2 %>% apply(.,1,diff) 
else 
  M <- countMx %>% log2 %>% apply(.,1,diff) - libSize %>% log2 %>% diff
w <- countMx[,1]==min(countMx[,1]) | countMx[,2]==min(countMx[,2])
if (any(w)) {
            A[w] <- runif(sum(w), min = -1, max = .1)
            M[w] <- log2(countMx[w,2] + 1) - log2(countMx[w,1] + 1)
}
MAplot <- qplot(A, M, col=w) +
  theme(legend.position = "none") + 
  scale_color_manual(values = c("black","orange")) +
  xlab("A: log2 Average") +
  ylab("M: log2 Fold Change")

MAplot +
  geom_abline(intercept=0,slope=0,col="blue") +
  geom_abline(intercept=median(M[!w],na.rm=TRUE),slope=0,col="red")
}
```

Let’s take a look at how comparable different replicates are in the Control condition at 48h in our dataset. We will investigate this using MD-plots (mean-difference plots as introduced by Dudoit et al. (2002)), also sometimes referred to as MA-plots.

```{r}
ids <- which(colData(se)$treatment =="Control" & colData(se)$time == "48h")
ids
colSums(assays(se)$counts[,ids]) / 1e6
```

```{r}
pairComb <- combn(
  ids, 
  m=2)
plots <- apply(pairComb,2,function(x) ma2Samp(assay(se)[,x]) + ggtitle(paste("samples",x[2],"vs", x[1])))
do.call("grid.arrange",c(plots,ncol=3))
```

We see clear bias for some pairwise comparisons. For example, in the first plot comparing sample 8 versus sample 2, the log fold-changes are biased downwards. This means that, on average, a gene is lower expressed in sample 8 versus sample 2. Looking at the library sizes, we can indeed see that the library size for sample 2 is about $11×10^6$ while it is only about $7×10^6$ for sample 8! This is a clear library size effect that we should take into account.


We can solve these issues by introducing offsets in our model.

$$
  \left\{
  \begin{array}{ccc}
  Y_{gi} & \sim & Poi(\mu_{gi}) \\
  \log \mu_{gi} & = & \eta_{gi} \\
  \eta_{gi} & = & \mathbf{X}^T_i \beta_g + log(O_{gi}) \\
  \end{array}
  \right.
  $$
  

```{r}
libSize <- colSums(assay(se))
plots2 <- apply(pairComb,2,function(x) ma2Samp(assay(se)[,x],libSize = libSize[x]) + ggtitle(paste("samples",x[2],"vs", x[1])))
do.call("grid.arrange",c(plots2,ncol=3))
```

## TMM method (default of `edgeR`)

[Robinson and Oshlack (2010). Genome Biology](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25)

```{r fig.width=0.5, fig.align='center', fig.cap==" Robinson and Oshlack (2010). Genome Biology."}
knitr::include_graphics("./figs/edgeRNormIntro.png")
```

- On the plot we see a clear effect on all genes
- Correcting for library size tends to over correct. 
- Some DE genes are highly abundant and determine the library size to a large extend

The trimmed mean of M-values (TMM) method introduced by [Robinson & Oshlack (2010)](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25) is a normalization procedure that calculates a single normalization factor for each sample. As the name suggests, it is based on a trimmed mean of fold-changes ($M$-values) as the scaling factor. A trimmed mean is an average after removing a set of ``extreme'' values. 
Specifically, TMM calculates a normalization factor $F_i^{(r)}$ across genes $g$ for each sample $i$ as compared to a reference sample $r$,
$$
\log_2(F_i^{(r)}) = \frac{\sum_{g \in {\cal G}^*} w_{gi}^r M_{gi}^r}{\sum_{g \in {\cal G}^*} w_{gi}^r},
$$
where $M_{gi}^r$ represents the $\log_2$-fold-change of the gene expression fraction as compared to a reference sample $r$, i.e.,
$$ M_{gi}^r = \log_2\left( \frac{Y_{gi} / N_i}{ Y_{gr} / N_r} \right), $$
and $w_{gi}^r$ represents a precision weight calculated as
$$
 w_{gi}^r = \frac{N_i - Y_{gi}}{N_i Y_{gi}} + \frac{N_r - Y_{gr}}{N_r Y_{gr}},
$$
and ${\cal G}^*$ represents the set of genes after trimming those with the most extreme average expression. The weights serve to account for the fact that fold-changes for genes with lower read counts are more variable.

The procedure only takes genes into account where both $Y_{gi}>0$ and $Y_{gr}>0$. By default, TMM trims genes with the $30\%$ most extreme $M$-values and $5\%$ most extreme average gene expression, and chooses as reference $r$ the sample whose upper-quartile is closest to the across-sample average upper-quartile. 

TMM normalization may be performed from the `calcNormFactors` function implemented in `edgeR`:

```{r}
dge <- edgeR::calcNormFactors(se)
dge$samples #normalization factors added to colData
```

Let's check how our MD-plots look like after normalization. Note that, we can rewrite the GLM as
$$ \log\left( \frac{\mu_{gi}}{N_i^s} \right) = \mathbf{X}_i^T \beta_g $$
and so $\frac{\mu_{gi}}{N_i^s}$ can be considered as an 'offset-corrected count'.

We see that all MD-plots are now nicely centered around a log-fold-change of zero!

```{r}
## normalize
effLibSize <- dge$samples$lib.size * dge$samples$norm.factors
#normCountTMM <- sweep(assays(se)$counts, 2, FUN="/", effLibSize)

plotsNorm <- apply(pairComb,2,function(x) 
  ma2Samp(assays(se)$counts[,x], effLibSize[x]) + ggtitle(paste("samples",x[2],"vs", x[1])))
do.call("grid.arrange",c(plotsNorm,ncol=3))
```


## Median-of-Ratios Method (default of `DESeq2`)

The median-of-ratios method is used in `DESeq2` as described in [Love *et al.* (2014)](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8).
It assumes that the expected value $\mu_{gi} = E(Y_{gi})$ is proportional to the true expression of the gene, $q_{gi}$, scaled by a normalization factor $s_{i}$ for each sample,
$$ \mu_{gi} = s_{i}q_{gi}. $$

The normalization factor $s_{i}$ is then estimated using the median-of-ratios method compared to a synthetic reference sample $r$ defined based on geometric means of counts across samples
$$
s_i = \text{median}_{\{{g:Y^{*}_{gr} \ne 0}\}} \frac{Y_{gi}}{Y^{*}_{gr}},
$$
with 
$$ Y^{*}_{gr} = \left( \prod_{i=1}^n Y_{gi} \right)^{1/n}. $$

We can then use the size factors $s_i$ as offsets to the GLM.

Median-of-ratios normalization is implemented in the `DESeq2` package:

```{r}
dds <- DESeq2::DESeqDataSetFromMatrix(countData = assays(se)$counts,
                                      colData = colData(se),
                                      design = ~ 1) #just add intercept to showcase normalization
dds <- DESeq2::estimateSizeFactors(dds)
sizeFactors(dds)
```

You may also want to check out the [StatQuest video on DESeq2 normalization](https://www.youtube.com/watch?v=UFB993xufUU).

### Comparing TMM with DESeq2 Normalization

We can compare the size factors for both normalizations to verify if they agree on the normalization factors. Note we need to scale the effective library sizes from `edgeR` to enforce a similar scale as the size factors from `DESeq2`. While below we are using an arithmetic mean, a geometric mean may be used as well, which will be more robust to outlying effective library sizes.

```{r}
plot(effLibSize / mean(effLibSize), sizeFactors(dds),
     xlab = "edgeR size factor",
     ylab = "DESeq2 size factor")
```

# Aliasing

Suppose we are working with the following experimental design on colon cancer. Studying the effect of a drug on gene expression, researchers gather RNA-seq data from four colon cancer patients and four healthy individuals. For each individual, they obtain RNA-seq data from a blood sample before as well as two weeks after taking a daily dose of the drug. The research question relates to differential expression after vs. before taking the drug, in particular whether this is different for the diseased versus healthy group (i.e., the interaction between time (before/after taking the drug) and disease status (healthy/colon cancer)).

In terms of the model matrix, we could imagine a design such as ` ~ patient + disease*time`, where 

 - `disease` is a binary indicator referring to colon cancer versus control sample.
 - `time` defines if the sample is taken before or after taking the drug.
 - `patient` defines the individual donor the sample comes from.

The research question could then amount to testing the `disease * time` interaction.

Let's try this, by simulating random data for one gene.

```{r}
set.seed(2)
# 2 samples per patient for 8 patients
patient <- factor(rep(letters[1:8], each=2)) 
# first four are healthy, next four are diseased
disease <- factor(c(rep("healthy",8), rep("cancer",8)), levels=c("healthy", "cancer")) 
# one before and one after sample for each
time <- factor(rep(c("before", "after"), 8), levels=c("before", "after")) 

table(patient, disease, time)

## simulate data for one gene
n <- 16
y <- rpois(n = n, lambda = 50)

## fit a Poisson model
m <- glm(y ~ patient + disease*time,
         family = "poisson")
summary(m)
```

 ---
 
We find that one of the coefficients is `NA`! This is obviously not because we're dealing with `NA` values in the data as we've just simulated the response variable ourselves. What's going on?

One of the parameters, in this case the parameter distinguishing cancer from healthy patients **cannot be estimated as it is a linear combination of other parameters**. In our case, estimating the diseased effect would use information that is already used to estimate the patient-level intercepts. In other words, **once you know the patient, you immediately also know the disease status**, so estimating the diseased vs healthy effect on top of the patient effect provides no additional information if we have already estimated the patient-level effects. This concept is called aliasing, and is a common technical issue in 'omics experiments with complex experimental designs. 

 ---

While to understand the origin of the aliasing it is crucial to understand the relationship between the variables in the experimental design, we can also investigate it in detail using the `alias` function, to give us an idea.

```{r}
alias(m)
```

We see that the effect `diseasecancer` is a linear combination of the patient-specific effects of the cancer patients. This makes sense!

 --- 

For clarity, let's reproduce this using our design matrix.

```{r}
X <- model.matrix(~ patient + disease*time) # this is the design used in glm()

## these are indeed identical.
X[,"diseasecancer"]
X[,"patiente"] + X[,"patientf"] + X[,"patientg"] + X[,"patienth"]
```

Since one of our parameters is a linear combination of other parameters, it cannot be estimated simultaneously with the other parameters. In this case, we can actually drop the `disease` main effect from the model, since we know that it is already included in the `patient` effect.

 ---

We will have to carefully construct our design matrix in order to account for all important sources of variation while still allowing us to answer the research question of interest. The aliasing exploration above has made it clear we may drop the `disease` main effect, so let's start by constructing this design matrix.

```{r}
X <- model.matrix(~ patient + time + disease:time)

m2 <- glm(y ~ -1 + X,
         family = "poisson")
summary(m2)
alias(m2)
```

We are still confronted with aliasing as the model matrix contains an interaction effect `timebefore:diseasecancer` as well as `timeafter:diseasecancer`, while only the latter is relevant. Indeed, we know that we can derive the `timebefore:diseasecancer` effect by averaging the patient effects of the cancer patients.

 ---

```{r}
X <- X[,!colnames(X) %in% "timebefore:diseasecancer"]


## fit a Poisson model
m2 <- glm(y ~ -1 + X,
         family = "poisson")
summary(m2)
```

We see that all coefficients can now be estimated. The `timeafter` effect may be interpreted as the time effect for healthy patients, while the `timeafter:diseasecancer` effect may be interpreted as the difference in the time effect for cancer patients as compared to healthy patients, i.e., it is the relevant interaction effect we are interested in. 

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("sequencing_technicalDE.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
